<!doctype html>
<html lang="en"><head><meta charset="utf-8"><meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1"><meta><title>ATAC-seq Analysis - Yuqi Gong&#039;s blog</title><link rel="manifest" href="/manifest.json"><meta name="application-name" content="Yuqi Gong&#039;s blog"><meta name="msapplication-TileImage" content="/img/ava.jpg"><meta name="apple-mobile-web-app-capable" content="yes"><meta name="apple-mobile-web-app-title" content="Yuqi Gong&#039;s blog"><meta name="apple-mobile-web-app-status-bar-style" content="default"><meta name="description" content="ATAC-seq旧版处理流程"><meta property="og:type" content="blog"><meta property="og:title" content="ATAC-seq Analysis"><meta property="og:url" content="http://gyqgyq.com/ATAC-seq%20Analysis/"><meta property="og:site_name" content="Yuqi Gong&#039;s blog"><meta property="og:description" content="ATAC-seq旧版处理流程"><meta property="og:locale" content="en_US"><meta property="og:image" content="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/rawdata.png"><meta property="og:image" content="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/rename_rawdata.png"><meta property="og:image" content="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/ubuntu-atac-seq-image.png"><meta property="og:image" content="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/raw-ls.png"><meta property="og:image" content="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/ATAC-environment.png"><meta property="og:image" content="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/clean-data.png"><meta property="og:image" content="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/rawqc.png"><meta property="og:image" content="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/trimqc.png"><meta property="og:image" content="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/raw_multiqc.png"><meta property="og:image" content="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/trim_multiqc.png"><meta property="og:image" content="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/multiqc-raw-heatmap.png"><meta property="og:image" content="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/multiqc-trim-heatmap.png"><meta property="og:image" content="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/raw-sequence-display.png"><meta property="og:image" content="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/trim-sequence-display.png"><meta property="og:image" content="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/genome.png"><meta property="og:image" content="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/gtf.png"><meta property="og:image" content="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/mm10_index.png"><meta property="og:image" content="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/align_result.png"><meta property="og:image" content="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/bam.bai.png"><meta property="og:image" content="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/bam.stat.png"><meta property="og:image" content="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/rmdup-bam-bai.png"><meta property="og:image" content="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/rmdup-stat.png"><meta property="og:image" content="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/rmdup-last-all.png"><meta property="og:image" content="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/last-bed.png"><meta property="og:image" content="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/peaks.png"><meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/feature-distribution.JPG"><meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/loci-to-TSS.JPG"><meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/peaks-over-chromsomes.JPG"><meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/peaks-overlap.JPG"><meta property="article:published_time" content="2020-02-08T08:30:23.000Z"><meta property="article:modified_time" content="2022-11-26T12:54:54.867Z"><meta property="article:author" content="Yuqi Gong"><meta property="article:tag" content="NGS"><meta property="twitter:card" content="summary"><meta property="twitter:image:src" content="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/rawdata.png"><script type="application/ld+json">{"@context":"https://schema.org","@type":"BlogPosting","mainEntityOfPage":{"@type":"WebPage","@id":"http://gyqgyq.com/ATAC-seq%20Analysis/"},"headline":"ATAC-seq Analysis","image":["https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/rawdata.png","https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/rename_rawdata.png","https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/ubuntu-atac-seq-image.png","https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/raw-ls.png","https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/ATAC-environment.png","https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/clean-data.png","https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/rawqc.png","https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/trimqc.png","https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/raw_multiqc.png","https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/trim_multiqc.png","https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/multiqc-raw-heatmap.png","https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/multiqc-trim-heatmap.png","https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/raw-sequence-display.png","https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/trim-sequence-display.png","https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/genome.png","https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/gtf.png","https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/mm10_index.png","https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/align_result.png","https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/bam.bai.png","https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/bam.stat.png","https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/rmdup-bam-bai.png","https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/rmdup-stat.png","https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/rmdup-last-all.png","https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/last-bed.png","https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/peaks.png"],"datePublished":"2020-02-08T08:30:23.000Z","dateModified":"2022-11-26T12:54:54.867Z","author":{"@type":"Person","name":"Yuqi Gong"},"publisher":{"@type":"Organization","name":"Yuqi Gong's blog","logo":{"@type":"ImageObject","url":"http://gyqgyq.com/img/ava.jpg"}},"description":"ATAC-seq旧版处理流程"}</script><link rel="canonical" href="http://gyqgyq.com/ATAC-seq%20Analysis/"><link rel="icon" href="/img/ava.jpg"><link rel="stylesheet" href="https://use.fontawesome.com/releases/v6.0.0/css/all.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/highlight.js@9.12.0/styles/atom-one-light.css"><link rel="stylesheet" href="https://fonts.googleapis.com/css2?family=Ubuntu:wght@400;600&amp;family=Source+Code+Pro"><link rel="stylesheet" href="/css/default.css"><style>body>.footer,body>.navbar,body>.section{opacity:0}</style><!--!--><!--!--><!--!--><script src="//busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js" defer></script><!--!--><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/cookieconsent@3.1.1/build/cookieconsent.min.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/lightgallery@1.10.0/dist/css/lightgallery.min.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/justifiedGallery@3.8.1/dist/css/justifiedGallery.min.css"><!--!--><!--!--><style>.pace{-webkit-pointer-events:none;pointer-events:none;-webkit-user-select:none;-moz-user-select:none;user-select:none}.pace-inactive{display:none}.pace .pace-progress{background:#3273dc;position:fixed;z-index:2000;top:0;right:100%;width:100%;height:2px}</style><script src="https://cdn.jsdelivr.net/npm/pace-js@1.2.4/pace.min.js"></script><!--!--><!--!--><!-- hexo injector head_end start --><script>
  (function () {
      function switchTab() {
          if (!location.hash) {
            return;
          }

          const $tabMenu = document.querySelector(`a[href="${location.hash}"]`);
          if (!$tabMenu) {
            return;
          }

          const $tabMenuContainer = $tabMenu.parentElement.parentElement;
          Array.from($tabMenuContainer.children).forEach($menu => $menu.classList.remove('is-active'));
          Array.from($tabMenuContainer.querySelectorAll('a'))
              .map($menu => document.getElementById($menu.getAttribute("href").substring(1)))
              .forEach($content => $content.classList.add('is-hidden'));

          if ($tabMenu) {
              $tabMenu.parentElement.classList.add('is-active');
          }
          const $activeTab = document.querySelector(location.hash);
          if ($activeTab) {
              $activeTab.classList.remove('is-hidden');
          }
      }
      switchTab();
      window.addEventListener('hashchange', switchTab, false);
  })();
  </script><!-- hexo injector head_end end --><meta name="generator" content="Hexo 6.3.0"></head><body class="is-2-column"><nav class="navbar navbar-main"><div class="container navbar-container"><div class="navbar-brand justify-content-center"><a class="navbar-item navbar-logo" href="/"><img src="/img/ava.jpg" alt="Yuqi Gong&#039;s blog" height="28"></a></div><div class="navbar-menu"><div class="navbar-start"><a class="navbar-item" href="/">Home</a><a class="navbar-item" href="/archives">Archives</a><a class="navbar-item" href="/categories">Categories</a><a class="navbar-item" href="/tags">Tags</a><a class="navbar-item" href="/about">About</a></div><div class="navbar-end"><a class="navbar-item" target="_blank" rel="noopener" title="Download on GitHub" href="https://github.com/gongyuqiqi/gongyuqiqi.github.io"><i class="fab fa-github"></i></a><a class="navbar-item is-hidden-tablet catalogue" title="Catalogue" href="javascript:;"><i class="fas fa-list-ul"></i></a><a class="navbar-item search" title="Search" href="javascript:;"><i class="fas fa-search"></i></a></div></div></div></nav><section class="section"><div class="container"><div class="columns"><div class="column order-2 column-main is-8-tablet is-8-desktop is-8-widescreen"><div class="card"><article class="card-content article" role="article"><div class="article-meta is-size-7 is-uppercase level is-mobile"><div class="level-left"><span class="level-item">Posted&nbsp;<time dateTime="2020-02-08T08:30:23.000Z" title="2/8/2020, 4:30:23 PM">2020-02-08</time></span><span class="level-item">Updated&nbsp;<time dateTime="2022-11-26T12:54:54.867Z" title="11/26/2022, 8:54:54 PM">2022-11-26</time></span><span class="level-item">21 minutes read (About 3109 words)</span><span class="level-item" id="busuanzi_container_page_pv"><span id="busuanzi_value_page_pv">0</span>&nbsp;visits</span></div></div><h1 class="title is-3 is-size-4-mobile">ATAC-seq Analysis</h1><div class="content"><p>ATAC-seq旧版处理流程</p>
<span id="more"></span>

<h1 id="ATAC-seq-Analysis"><a href="#ATAC-seq-Analysis" class="headerlink" title="ATAC-seq Analysis"></a>ATAC-seq Analysis</h1><p>表观测序数据中比较重要的就是ATAC-seq、Chip-seq。这两者的原理和分析过程类似。<br>具体的异同详见此链接：<a target="_blank" rel="noopener" href="https://www.jianshu.com/p/87bc2002e82c">https://www.jianshu.com/p/87bc2002e82c</a></p>
<h2 id="一-数据下载"><a href="#一-数据下载" class="headerlink" title="(一)数据下载"></a>(一)数据下载</h2><p><font color=red>数据下载是我走过的最曲折的路</font></p>
<p>强烈不推荐wget、curl——需要向上天再借500年才能等的起。</p>
<p>不推荐prefetch——不知道为什么prefetch在我这里也不起作用！按理说有ascp环境后，prefetch会自动调用ascp进行数据下载呀！可是prefetch依然显示通过https进行数据下载，这就没法下载呀！直接timeout！</p>
<p><strong>正解——调用ascp下载数据</strong></p>
<p>（1）ascp下载NCBI的SRA文件</p>
<p>此方法亲测不行，但是就在2个月前还是可以的。官网给出的解释大致意思就是——不好意思，这段时间出现无法通过ascp下载NCBI数据属于正常现象…blablabla…</p>
<p>（2）ascp下载EBI的fastq文件（强烈推荐，省去了SRA转成fastq的麻烦，很耗资源，很耗时）</p>
<p>在此次数据分析中，下载SRR2927015、SRR2927016、SRR2927018、SRR3545580对应的fastq文件进行ATAC-seq数据分析的实战。</p>
<p>EBI官网输入对应SRR号（以SRR3545580为例），找到相应的fastq文件位置，复制连接地址。</p>
<p><a target="_blank" rel="noopener" href="http://ftp.sra.ebi.ac.uk/vol1/fastq/SRR354/000/SRR3545580/SRR3545580_1.fastq.gz">http://ftp.sra.ebi.ac.uk/vol1/fastq/SRR354/000/SRR3545580/SRR3545580_1.fastq.gz</a></p>
<p><a target="_blank" rel="noopener" href="http://ftp.sra.ebi.ac.uk/vol1/fastq/SRR354/000/SRR3545580/SRR3545580_2.fastq.gz">http://ftp.sra.ebi.ac.uk/vol1/fastq/SRR354/000/SRR3545580/SRR3545580_2.fastq.gz</a></p>
<p>下载方式如下（下载速递可达到每秒M级别，快的时候能到60M+，亲测此方法为目前最可取的方法）</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">ascp -QT -l 300m -P33001 -i /home/gongyuqi/.aspera/connect/etc/asperaweb_id_dsa.openssh era-fasp@fasp.sra.ebi.ac.uk:/vol1/fastq/SRR354/000/SRR3545580/SRR3545580_1.fastq.gz .</span><br><span class="line">ascp -QT -l 300m -P33001 -i /home/gongyuqi/.aspera/connect/etc/asperaweb_id_dsa.openssh era-fasp@fasp.sra.ebi.ac.uk:/vol1/fastq/SRR354/000/SRR3545580/SRR3545580_2.fastq.gz .</span><br></pre></td></tr></table></figure>

<p>最终下载结果</p>
<p><img src="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/rawdata.png"></p>
<p>为了后续分析，将其改名为下</p>
<p>2-cell-1  SRR2927015<br>2-cell-2  SRR2927016<br>2-cell-4  SRR2927018<br>2-cell-5  SRR3545580</p>
<p><img src="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/rename_rawdata.png"></p>
<p>附上测序数据下载方式的相关参考资料</p>
<p><a target="_blank" rel="noopener" href="https://www.jianshu.com/p/df34b99cbd43">https://www.jianshu.com/p/df34b99cbd43</a></p>
<p><a target="_blank" rel="noopener" href="https://www.bioinfo-scrounger.com/archives/171/">https://www.bioinfo-scrounger.com/archives/171/</a></p>
<h2 id="（二）自定义docker容器的启动与数据共享"><a href="#（二）自定义docker容器的启动与数据共享" class="headerlink" title="（二）自定义docker容器的启动与数据共享"></a>（二）自定义docker容器的启动与数据共享</h2><p>之前创建了具备ATAC-seq数据分析所需环境的docker容器，并push到阿里云。在此只需要pull该镜像，run即可使用。</p>
<p>但是！！！创建容器时并没有装ascp，因为只有普通用户才可以运行ascp。所以数据下载是在ubuntu主机进行的。为了能够在容器中使用下载的测试数据，运用数据卷，实现主机和容器间的数据共享。</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"># 查看镜像</span><br><span class="line">docker images</span><br></pre></td></tr></table></figure>
<p><img src="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/ubuntu-atac-seq-image.png"></p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"># 启动容器实现数据共享</span><br><span class="line">docker run -it -v /home/gongyuqi/Downloads/ATACdata/:/root/project/atac/raw 608eafa789c8</span><br></pre></td></tr></table></figure>

<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line"># 查看容器内是否已经存在fastq数据</span><br><span class="line">cd root/project/atac/raw</span><br><span class="line">ls</span><br></pre></td></tr></table></figure>

<p><img src="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/raw-ls.png"></p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line"># 激活ATAC环境（该环境配置有处理ATAC-seq数据的所需软件）</span><br><span class="line">conda activate ATAC</span><br><span class="line"># (base)变成了(ATAC)，即由最初的conda环境变成了ATAC环境</span><br></pre></td></tr></table></figure>

<p><img src="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/ATAC-environment.png"></p>
<h2 id="（三）、数据过滤"><a href="#（三）、数据过滤" class="headerlink" title="（三）、数据过滤"></a>（三）、数据过滤</h2><p><strong>运用软件 Trim Galore</strong></p>
<p><strong>去除接头和质量不合格的reads</strong></p>
<blockquote>
<p>Trim Galore是对FastQC和Cutadapt的包装。适用于所有高通量测序，包括RRBS(Reduced Representation Bisulfite-Seq ), Illumina、Nextera 和smallRNA测序平台的双端和单端数据。</p>
</blockquote>
<blockquote>
<p>Trimmomatic是针对Illumina高通量测序平台设计的接头去除和低质量reads清洗软件。</p>
</blockquote>
<p>附上trim-galore相关参考资料: <a target="_blank" rel="noopener" href="https://www.jianshu.com/p/7a3de6b8e503">https://www.jianshu.com/p/7a3de6b8e503</a></p>
<p>以下列数据为例</p>
<p>2-cell-1  SRR2927015</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">trim_galore -q 7 --phred33 --length 35 -e 0.1 --stringency 4 --paired -o /root/project/atac/clean/ 2-cell-1_1.fastq.gz  2-cell-1_2.fastq.gz</span><br></pre></td></tr></table></figure>

<blockquote>
<p><strong>trim_galore</strong></p>
<p>–q：设置线程</p>
<p>–phred33：使用ASCII+33质量值作为Phred的分</p>
<p>–length：去除长度小于参数值的reads</p>
<p>-e：允许的最大误差</p>
<p>–stringency：设置与接头重叠的序列</p>
<p>–paired：对于双段测序文件，正反链质控通过才保留</p>
<p>-o：输出目录</p>
</blockquote>
<p>结果如下</p>
<p><img src="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/clean-data.png"></p>
<p>因为本次数据是用本机（8线程，16G）跑的，要给机子休息的时间，同时实验样本少，所以一个一个样本手动跑。<br>但是！当样本量多了，在服务器上跑的时候，循环就要省事很多，比如跑的时候去睡个觉…</p>
<p><strong>拿本次样本写个循环，举个例</strong></p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">ls /root/project/raw/*1.fq.gz &gt;1</span><br><span class="line">ls /root/project/raw/*2.fq.gz &gt;2</span><br><span class="line">paste 1 2 &gt; sample</span><br><span class="line">path=/root/project/atac/clean</span><br><span class="line">cat sample | while read id; do echo $id arr=($id) fq1=$&#123;arr[0]&#125; fq2=$&#123;arr[1]&#125; nohup trim_galore -q 30 --phred33 -e 0.1 --stringency 4 --paired -o $path $fq1 $fq2 &amp; done</span><br></pre></td></tr></table></figure>

<h2 id="四-、质量控制"><a href="#四-、质量控制" class="headerlink" title="(四)、质量控制"></a>(四)、质量控制</h2><p><font color=red>测试数据的质控真的非常重要！！！如果测序质量本身就很好，没有质空这一步也影响不大。如果测序质量不好，不进行质控的化，直接影响后续的比对率。一般情况下，90%+的比对率才算比较正常，太低的比对率往往是质控步骤有问题，会非常严重的影响最终结果。亲测过！！！！</font></p>
<p><strong>运用软件fastqc、multiqc</strong></p>
<p><strong>进行批处理</strong></p>
<ul>
<li>数据质控</li>
</ul>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">cd /root/project/atac/qc &amp;&amp; mkdir rawqc trimqc</span><br><span class="line"># 对原始数据进行质控</span><br><span class="line">cd rawqc</span><br><span class="line">fastqc -t 7 /root/project/atac/raw/*gz -o ./</span><br><span class="line"># 对过滤数据进行质控</span><br><span class="line">cd .. &amp; cd trimqc</span><br><span class="line">fastqc -t 7 /root/project/atac/clean/*gz -o ./</span><br></pre></td></tr></table></figure>
<p>（1）原始数据质控结果</p>
<p><img src="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/rawqc.png"></p>
<p>（2）过滤后数据质量控制结果</p>
<p><img src="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/trimqc.png"></p>
<ul>
<li>质控结果整合</li>
</ul>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line"># 原始数据qc结果整合</span><br><span class="line">cd /root/project/atac/qc/rawqc &amp;&amp; mkdir multiqc_results</span><br><span class="line">multiqc . -o multiqc_results/</span><br><span class="line"># 过滤数据qc结果整合</span><br><span class="line">cd /root/project/atac/qc/trimqc &amp;&amp; mkdir multiqc_results</span><br><span class="line">multiqc . -o multiqc_results/</span><br></pre></td></tr></table></figure>

<p>（1）原始数据整合结果</p>
<p><img src="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/raw_multiqc.png"></p>
<p>（2）过滤数据整合结果</p>
<p><img src="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/trim_multiqc.png"></p>
<p>附上multiqc相关参考资料<br><a target="_blank" rel="noopener" href="https://blog.csdn.net/ada0915/article/details/77201871">https://blog.csdn.net/ada0915/article/details/77201871</a></p>
<ul>
<li>mulitiqc整合结果解读</li>
</ul>
<p>(1)过滤前</p>
<p><img src="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/multiqc-raw-heatmap.png"></p>
<p>（2）过滤后</p>
<p><img src="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/multiqc-trim-heatmap.png"></p>
<p>（3）解读（随便一搜，各种解读教程实在太多了～）</p>
<p>trim-galore完美的去除了adaptor;去除了质量不合格的序列使得2-cell-1_2和2-cell-2_2的质量过关；2-cell-1_2的N含量超标的序列被去除了。</p>
<p>因为去除了adptor和质量不合格的序列，所以sequence length自然是不一致的了。另外，还没有PCR去重，所以sequence duplication是不过关的，这一点会在PCR去重步骤中得到解决。至于Per Base Sequence Content,前10个碱基总是异常的波动，比对序列长度远大于10，处理时也可以设定不比对前10个，所以这个关系不大。</p>
<p>（4）以Sequence Length和Per Base N Content展示过滤前后的区别</p>
<p>过滤前</p>
<p><img src="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/raw-sequence-display.png"></p>
<p>过滤后</p>
<p><img src="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/trim-sequence-display.png"></p>
<h2 id="五-比对参考基因组"><a href="#五-比对参考基因组" class="headerlink" title="(五)比对参考基因组"></a>(五)比对参考基因组</h2><p><strong>运用软件bowtie2、bedtools</strong></p>
<p>一般对于DNA数据，比对软件一般使用：<strong>bowtie2、bwa</strong></p>
<p>对于RNA数据，比对一般使用：<strong>hisat2、star</strong></p>
<p>使用<strong>bedtools</strong>将最终的bam文件转成bed文件</p>
<p><strong>（1）参考基因组及其注释文件的下载</strong></p>
<p>参考基因组（解压后）<br><img src="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/genome.png"></p>
<p>参考基因组注释文件（解压后）<br><img src="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/gtf.png"></p>
<p>本次分析用到的是小鼠的参考基因组及其注释文件</p>
<p>mm10.fa、gencode.vM24.annotation.gtf</p>
<p><strong>（2）参考基因组索引构建</strong>（将构建的参考基因组索引文件cp到容器中的&#x2F;root&#x2F;project&#x2F;atac&#x2F;reference&#x2F;mm10_index目录下）</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">bowtie2-build /home/gongyuqi/biosoftware/reference/genome/mm10.fa mm10_index</span><br></pre></td></tr></table></figure>

<p><img src="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/mm10_index.png"></p>
<p><strong>（3）参考基因组的比对，直接生成bam文件</strong></p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">bowtie2 -p 7 -x /root/project/atac/reference/mm10_index -1 ../clean/2-cell-1_1_val_1.fq.gz -2 ../clean/2-cell-1_2_val_2.fq.gz | samtools sort -@ 7 -O bam -o 2-cell-1.bam -</span><br><span class="line"># 此samtools sort默认按照reads比对到染色体上的坐标排序</span><br></pre></td></tr></table></figure>

<p>bowtie2直接比对的结果会生成sam文件，sam文件很大，比较占空间。在比对这一步就直接利用samtools sort将其转成bam文件。</p>
<p>另外，samtool排序过程中并不会丢弃一部分reads，但是排序后的bam文件相比排序前的体积会更小。</p>
<p>比对相关参考资料</p>
<p><a target="_blank" rel="noopener" href="https://www.jianshu.com/p/6ed1bfbb7b72">https://www.jianshu.com/p/6ed1bfbb7b72</a></p>
<p><strong>比对结果</strong></p>
<p>以2-cell-1_1_val为例进行结果展示</p>
<p><img src="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/align_result.png"></p>
<p><strong>（4）构建原始bam文件的索引文件、状态文件</strong></p>
<p>构建索引文件</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">ls *.bam | while read id; do samtools index -@ 7 $id; done</span><br></pre></td></tr></table></figure>
<p>.bai文件即位生成的索引文件</p>
<p><img src="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/bam.bai.png"></p>
<p>构建状态文件</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">ls *.bam | while read id; do samtools flagstat -@ 7 $id &gt; $id.stat; done</span><br></pre></td></tr></table></figure>

<p><img src="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/bam.stat.png"></p>
<p><strong>（5）去除PCR重复、构建去重后的索引文件、状态文件、</strong></p>
<p>sambamba软件去重，会同时生成了去重后的bam和bai文件</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">ls *.bam | while read id; do sambamba markdup -r -t 7 $id rmdup.$id; done</span><br></pre></td></tr></table></figure>

<p><img src="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/rmdup-bam-bai.png"></p>
<p>构建状态文件</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">ls rmdup*.bam | while read id; do samtools flagstat -@ 7 $id &gt; $id.stat; done</span><br></pre></td></tr></table></figure>

<p><img src="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/rmdup-stat.png"></p>
<p><strong>（6）去除同一样本的两条reads无法落在同一条染色体及测序质量低于30的reads、去除线粒体基因、构建索引文件、构建状态文件</strong></p>
<p>去除不符合要求的reads及线粒体基因</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">ls rm*.bam | while read id; do samtools view -@ 7 -h -f 2 -q 30 $id | grep -v chrM |samtools sort -O bam -@7 -o last.$id; done</span><br></pre></td></tr></table></figure>

<p>构建索引文件</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">ls last*.bam | while read id; do samtools index -@ 7 $id; done</span><br></pre></td></tr></table></figure>

<p>构建状态文件</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">ls last*.bam | while read id; do samtools flagstat -@ 7 $id &gt; $id.stat; done</span><br></pre></td></tr></table></figure>

<p><img src="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/rmdup-last-all.png"></p>
<p><strong>（7）生成bed文件</strong></p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">ls last*.bam | while read id; do bedtools bamtobed -i $id &gt; $id.bed; done</span><br></pre></td></tr></table></figure>

<p><img src="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/last-bed.png"></p>
<h2 id="（六）使用macs2找peaks"><a href="#（六）使用macs2找peaks" class="headerlink" title="（六）使用macs2找peaks"></a>（六）使用macs2找peaks</h2><p><strong>使用软件macs2</strong></p>
<p>以last.rmdup.2-cell-1.bam.bed进行演示</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">macs2 callpeak -t last.rmdup.2-cell-1.bam.bed -g mm --nomodel --shift -100 --extsize 200 -n 2-cell-1 --outdir ../peaks/</span><br></pre></td></tr></table></figure>

<p>最终生成文件<br><img src="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/peaks.png"></p>
<h2 id="peaks注释"><a href="#peaks注释" class="headerlink" title="peaks注释"></a>peaks注释</h2><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br></pre></td><td class="code"><pre><span class="line">setwd<span class="punctuation">(</span><span class="string">&quot;path\\to\\.bed&quot;</span><span class="punctuation">)</span></span><br><span class="line">library<span class="punctuation">(</span>ChIPpeakAnno<span class="punctuation">)</span></span><br><span class="line">library<span class="punctuation">(</span>TxDb.Mmusculus.UCSC.mm10.knownGene<span class="punctuation">)</span></span><br><span class="line">library<span class="punctuation">(</span>org.Mm.eg.db<span class="punctuation">)</span></span><br><span class="line">library<span class="punctuation">(</span>BiocInstaller<span class="punctuation">)</span></span><br><span class="line">library<span class="punctuation">(</span>ChIPseeker<span class="punctuation">)</span></span><br><span class="line">txdb <span class="operator">&lt;-</span> TxDb.Mmusculus.UCSC.mm10.knownGene</span><br><span class="line">promoter <span class="operator">&lt;-</span> getPromoters<span class="punctuation">(</span>TxDb<span class="operator">=</span>txdb<span class="punctuation">,</span> upstream<span class="operator">=</span><span class="number">3000</span><span class="punctuation">,</span> downstream<span class="operator">=</span><span class="number">3000</span><span class="punctuation">)</span></span><br><span class="line">files <span class="operator">=</span> <span class="built_in">list</span><span class="punctuation">(</span>cell_1_summits <span class="operator">=</span> <span class="string">&quot;2-cell-1_summits.bed&quot;</span><span class="punctuation">,</span> cell_2_summits <span class="operator">=</span> <span class="string">&quot;2-cell-2_summits.bed&quot;</span><span class="punctuation">,</span></span><br><span class="line">             cell_4_summits <span class="operator">=</span> <span class="string">&quot;2-cell-4_summits.bed&quot;</span><span class="punctuation">,</span> cell_5_summits <span class="operator">=</span> <span class="string">&quot;2-cell-5_summits.bed&quot;</span><span class="punctuation">)</span></span><br><span class="line">peakAnno <span class="operator">&lt;-</span> annotatePeak<span class="punctuation">(</span>files<span class="punctuation">[[</span><span class="number">1</span><span class="punctuation">]</span><span class="punctuation">]</span><span class="punctuation">,</span> <span class="comment"># 分别改成2或者3或者4即可，分别对应四个文件</span></span><br><span class="line">                         tssRegion<span class="operator">=</span><span class="built_in">c</span><span class="punctuation">(</span><span class="operator">-</span><span class="number">3000</span><span class="punctuation">,</span> <span class="number">3000</span><span class="punctuation">)</span><span class="punctuation">,</span></span><br><span class="line">                         TxDb<span class="operator">=</span>txdb<span class="punctuation">,</span> annoDb<span class="operator">=</span><span class="string">&quot;org.Mm.eg.db&quot;</span><span class="punctuation">)</span></span><br></pre></td></tr></table></figure>

<p><strong>（1）拿到每个样本中peaks对应得基因名</strong></p>
<p>这一步非常重要，拿到基因名就可以根据课题需要进行差异分析等</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">genelist<span class="operator">&lt;-</span>peakAnno<span class="operator">@</span>anno<span class="operator">@</span>elementMetadata<span class="operator">@</span>listData<span class="punctuation">[[</span><span class="string">&quot;SYMBOL&quot;</span><span class="punctuation">]</span><span class="punctuation">]</span></span><br></pre></td></tr></table></figure>

<p><strong>（2）单个文件作图</strong></p>
<p>图片就不做展示了</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">plotAnnoPie<span class="punctuation">(</span>peakAnno<span class="punctuation">)</span></span><br><span class="line">plotAnnoBar<span class="punctuation">(</span>peakAnno<span class="punctuation">)</span></span><br><span class="line">vennpie<span class="punctuation">(</span>peakAnno<span class="punctuation">)</span></span><br><span class="line">upsetplot<span class="punctuation">(</span>peakAnno<span class="punctuation">)</span></span><br><span class="line">upsetplot<span class="punctuation">(</span>peakAnno<span class="punctuation">,</span> vennpie<span class="operator">=</span><span class="literal">TRUE</span><span class="punctuation">)</span></span><br></pre></td></tr></table></figure>

<p><strong>（3）plotAnnoBar和plotDistToTSS这两个柱状图都支持多个数据同时展示，方便比较</strong></p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br></pre></td><td class="code"><pre><span class="line">setwd<span class="punctuation">(</span><span class="string">&quot;path\\to\\.bed&quot;</span><span class="punctuation">)</span></span><br><span class="line">library<span class="punctuation">(</span>ChIPpeakAnno<span class="punctuation">)</span></span><br><span class="line">library<span class="punctuation">(</span>TxDb.Mmusculus.UCSC.mm10.knownGene<span class="punctuation">)</span></span><br><span class="line">library<span class="punctuation">(</span>org.Mm.eg.db<span class="punctuation">)</span></span><br><span class="line">library<span class="punctuation">(</span>BiocInstaller<span class="punctuation">)</span></span><br><span class="line">library<span class="punctuation">(</span>ChIPseeker<span class="punctuation">)</span></span><br><span class="line">txdb <span class="operator">&lt;-</span> TxDb.Mmusculus.UCSC.mm10.knownGene</span><br><span class="line">promoter <span class="operator">&lt;-</span> getPromoters<span class="punctuation">(</span>TxDb<span class="operator">=</span>txdb<span class="punctuation">,</span> upstream<span class="operator">=</span><span class="number">3000</span><span class="punctuation">,</span> downstream<span class="operator">=</span><span class="number">3000</span><span class="punctuation">)</span></span><br><span class="line">files <span class="operator">=</span> <span class="built_in">list</span><span class="punctuation">(</span>cell_1_summits <span class="operator">=</span> <span class="string">&quot;2-cell-1_summits.bed&quot;</span><span class="punctuation">,</span> cell_2_summits <span class="operator">=</span> <span class="string">&quot;2-cell-2_summits.bed&quot;</span><span class="punctuation">,</span></span><br><span class="line">             cell_4_summits <span class="operator">=</span> <span class="string">&quot;2-cell-4_summits.bed&quot;</span><span class="punctuation">,</span> cell_5_summits <span class="operator">=</span> <span class="string">&quot;2-cell-5_summits.bed&quot;</span><span class="punctuation">)</span></span><br><span class="line">peakAnnoList <span class="operator">&lt;-</span> lapply<span class="punctuation">(</span>files<span class="punctuation">,</span> annotatePeak<span class="punctuation">,</span> </span><br><span class="line">                       TxDb<span class="operator">=</span>txdb<span class="punctuation">,</span>tssRegion<span class="operator">=</span><span class="built_in">c</span><span class="punctuation">(</span><span class="operator">-</span><span class="number">3000</span><span class="punctuation">,</span> <span class="number">3000</span><span class="punctuation">)</span><span class="punctuation">)</span></span><br><span class="line">plotAnnoBar<span class="punctuation">(</span>peakAnnoList<span class="punctuation">)</span></span><br></pre></td></tr></table></figure>

<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/feature-distribution.JPG"/>


<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">plotDistToTSS(peakAnnoList, title = <span class="string">&quot;Distribution of transcription factor-binding loci\nrelative to TSS&quot;</span>)</span><br></pre></td></tr></table></figure>

<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/loci-to-TSS.JPG"/>

<p><strong>（4）查看peaks在全基因组的位置（以2-cell-1样本为例）</strong></p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">chropeaks<span class="operator">&lt;-</span>readPeakFile<span class="punctuation">(</span><span class="string">&quot;2-cell-1_peaks.narrowPeak&quot;</span><span class="punctuation">)</span></span><br><span class="line">covplot<span class="punctuation">(</span>chropeaks<span class="punctuation">,</span> weightCol<span class="operator">=</span><span class="number">5</span><span class="punctuation">,</span> title <span class="operator">=</span> <span class="string">&quot;Peaks over chromosomes&quot;</span><span class="punctuation">)</span></span><br></pre></td></tr></table></figure>

<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/peaks-over-chromsomes.JPG"/>


<p><strong>（4）重复样本间共同的peaks和注释基因</strong></p>
<p>也可以用拿到的各个样本的基因名，做韦恩图，会比这个图更美观。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">genes<span class="operator">&lt;-</span>lapply<span class="punctuation">(</span>peakAnnoList<span class="punctuation">,</span> <span class="keyword">function</span><span class="punctuation">(</span>i<span class="punctuation">)</span> as.data.frame<span class="punctuation">(</span>i<span class="punctuation">)</span><span class="operator">$</span>geneId<span class="punctuation">)</span></span><br><span class="line">vennplot<span class="punctuation">(</span>genes<span class="punctuation">)</span></span><br></pre></td></tr></table></figure>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/peaks-overlap.JPG"/>

<p>peaks注释相关参考资料</p>
<p><a target="_blank" rel="noopener" href="https://www.jianshu.com/p/c83a38915afc">https://www.jianshu.com/p/c83a38915afc</a></p>
</div><div class="article-licensing box"><div class="licensing-title"><p>ATAC-seq Analysis</p><p><a href="http://gyqgyq.com/ATAC-seq Analysis/">http://gyqgyq.com/ATAC-seq Analysis/</a></p></div><div class="licensing-meta level is-mobile"><div class="level-left"><div class="level-item is-narrow"><div><h6>Author</h6><p>Yuqi Gong</p></div></div><div class="level-item is-narrow"><div><h6>Posted on</h6><p>2020-02-08</p></div></div><div class="level-item is-narrow"><div><h6>Updated on</h6><p>2022-11-26</p></div></div><div class="level-item is-narrow"><div><h6>Licensed under</h6><p><a class="icons" rel="noopener" target="_blank" title="Creative Commons" href="https://creativecommons.org/"><i class="icon fab fa-creative-commons"></i></a><a class="icons" rel="noopener" target="_blank" title="Attribution" href="https://creativecommons.org/licenses/by/4.0/"><i class="icon fab fa-creative-commons-by"></i></a><a class="icons" rel="noopener" target="_blank" title="Noncommercial" href="https://creativecommons.org/licenses/by-nc/4.0/"><i class="icon fab fa-creative-commons-nc"></i></a></p></div></div></div></div></div><div class="article-tags is-size-7 mb-4"><span class="mr-2">#</span><a class="link-muted mr-2" rel="tag" href="/tags/NGS/">NGS</a></div><div class="a2a_kit a2a_kit_size_32 a2a_default_style"><a class="a2a_dd" target="_blank" rel="noopener" href="https://www.addtoany.com/share"></a><a class="a2a_button_facebook"></a><a class="a2a_button_twitter"></a><a class="a2a_button_telegram"></a><a class="a2a_button_whatsapp"></a><a class="a2a_button_reddit"></a></div><script src="https://static.addtoany.com/menu/page.js" defer></script></article></div><div class="card"><div class="card-content"><h3 class="menu-label has-text-centered">Like this article? Support the author with</h3><div class="buttons is-centered"><a class="button donate" data-type="alipay"><span class="icon is-small"><i class="fab fa-alipay"></i></span><span>Alipay</span><span class="qrcode"><img src="/" alt="Alipay"></span></a><a class="button donate" data-type="wechat"><span class="icon is-small"><i class="fab fa-weixin"></i></span><span>Wechat</span><span class="qrcode"><img src="/" alt="Wechat"></span></a></div></div></div><nav class="post-navigation mt-4 level is-mobile"><div class="level-start"><a class="article-nav-prev level level-item link-muted" href="/scRNA-seq%E6%95%B0%E6%8D%AE%E5%88%86%E6%9E%90%E5%AD%A6%E4%B9%A0%EF%BC%88%E4%B8%80%EF%BC%89/"><i class="level-item fas fa-chevron-left"></i><span class="level-item">scRNA-seq数据分析学习（一）</span></a></div><div class="level-end"><a class="article-nav-next level level-item link-muted" href="/LVM%E9%80%BB%E8%BE%91%E5%8D%B7%E7%AE%A1%E7%90%86%E5%99%A8/"><span class="level-item">LVM逻辑卷管理器</span><i class="level-item fas fa-chevron-right"></i></a></div></nav><!--!--></div><div class="column column-left is-4-tablet is-4-desktop is-4-widescreen  order-1"><div class="card widget" data-type="profile"><div class="card-content"><nav class="level"><div class="level-item has-text-centered flex-shrink-1"><div><figure class="image is-128x128 mx-auto mb-2"><img class="avatar" src="/img/ava.jpg" alt="Gong Yuqi"></figure><p class="title is-size-4 is-block" style="line-height:inherit;">Gong Yuqi</p><p class="is-size-6 is-flex justify-content-center"><i class="fas fa-map-marker-alt mr-1"></i><span>HangZhou, China</span></p></div></div></nav><nav class="level is-mobile"><div class="level-item has-text-centered is-marginless"><div><p class="heading">Posts</p><a href="/archives"><p class="title">38</p></a></div></div><div class="level-item has-text-centered is-marginless"><div><p class="heading">Category</p><a href="/categories"><p class="title">1</p></a></div></div><div class="level-item has-text-centered is-marginless"><div><p class="heading">Tags</p><a href="/tags"><p class="title">7</p></a></div></div></nav><div class="level"><a class="level-item button is-primary is-rounded" href="https://github.com/gongyuqiqi" target="_blank" rel="noopener">Follow</a></div><div class="level is-mobile is-multiline"><a class="level-item button is-transparent is-marginless" target="_blank" rel="noopener" title="Github" href="https://github.com/gongyuqiqi"><i class="fab fa-github"></i></a><a class="level-item button is-transparent is-marginless" target="_blank" rel="noopener" title="Facebook" href="https://facebook.com"><i class="fab fa-facebook"></i></a><a class="level-item button is-transparent is-marginless" target="_blank" rel="noopener" title="Twitter" href="https://twitter.com"><i class="fab fa-twitter"></i></a><a class="level-item button is-transparent is-marginless" target="_blank" rel="noopener" title="Dribbble" href="https://dribbble.com"><i class="fab fa-dribbble"></i></a><a class="level-item button is-transparent is-marginless" target="_blank" rel="noopener" title="RSS" href="/"><i class="fas fa-rss"></i></a></div></div></div><div class="card widget" id="toc" data-type="toc"><div class="card-content"><div class="menu"><h3 class="menu-label">Catalogue</h3><ul class="menu-list"><li><a class="level is-mobile" href="#ATAC-seq-Analysis"><span class="level-left"><span class="level-item">1</span><span class="level-item">ATAC-seq Analysis</span></span></a><ul class="menu-list"><li><a class="level is-mobile" href="#一-数据下载"><span class="level-left"><span class="level-item">1.1</span><span class="level-item">(一)数据下载</span></span></a></li><li><a class="level is-mobile" href="#（二）自定义docker容器的启动与数据共享"><span class="level-left"><span class="level-item">1.2</span><span class="level-item">（二）自定义docker容器的启动与数据共享</span></span></a></li><li><a class="level is-mobile" href="#（三）、数据过滤"><span class="level-left"><span class="level-item">1.3</span><span class="level-item">（三）、数据过滤</span></span></a></li><li><a class="level is-mobile" href="#四-、质量控制"><span class="level-left"><span class="level-item">1.4</span><span class="level-item">(四)、质量控制</span></span></a></li><li><a class="level is-mobile" href="#五-比对参考基因组"><span class="level-left"><span class="level-item">1.5</span><span class="level-item">(五)比对参考基因组</span></span></a></li><li><a class="level is-mobile" href="#（六）使用macs2找peaks"><span class="level-left"><span class="level-item">1.6</span><span class="level-item">（六）使用macs2找peaks</span></span></a></li><li><a class="level is-mobile" href="#peaks注释"><span class="level-left"><span class="level-item">1.7</span><span class="level-item">peaks注释</span></span></a></li></ul></li></ul></div></div><style>#toc .menu-list > li > a.is-active + .menu-list { display: block; }#toc .menu-list > li > a + .menu-list { display: none; }</style><script src="/js/toc.js" defer></script></div><div class="card widget" data-type="categories"><div class="card-content"><div class="menu"><h3 class="menu-label">Categories</h3><ul class="menu-list"><li><a class="level is-mobile" href="/categories/Programming-Language/"><span class="level-start"><span class="level-item">Programming Language</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></div></div></div><div class="card widget" data-type="recent-posts"><div class="card-content"><h3 class="menu-label">Recents</h3><article class="media"><div class="media-content"><p class="date"><time dateTime="2024-01-18T13:40:44.000Z">2024-01-18</time></p><p class="title"><a href="/test/">test</a></p></div></article><article class="media"><div class="media-content"><p class="date"><time dateTime="2021-07-04T05:48:30.000Z">2021-07-04</time></p><p class="title"><a href="/RNA-TNBC/">RNA-TNBC</a></p></div></article><article class="media"><div class="media-content"><p class="date"><time dateTime="2021-06-29T06:01:05.000Z">2021-06-29</time></p><p class="title"><a href="/scRNA-TNBC/">scRNA-TNBC</a></p></div></article><article class="media"><div class="media-content"><p class="date"><time dateTime="2021-06-03T04:43:57.000Z">2021-06-03</time></p><p class="title"><a href="/MeRIP-Seq/">MeRIP-Seq</a></p></div></article><article class="media"><div class="media-content"><p class="date"><time dateTime="2021-05-14T05:37:00.000Z">2021-05-14</time></p><p class="title"><a href="/WES/">WES</a></p></div></article></div></div><div class="card widget" data-type="archives"><div class="card-content"><div class="menu"><h3 class="menu-label">Archives</h3><ul class="menu-list"><li><a class="level is-mobile" href="/archives/2024/01/"><span class="level-start"><span class="level-item">January 2024</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/archives/2021/07/"><span class="level-start"><span class="level-item">July 2021</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/archives/2021/06/"><span class="level-start"><span class="level-item">June 2021</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/archives/2021/05/"><span class="level-start"><span class="level-item">May 2021</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li><li><a class="level is-mobile" href="/archives/2021/03/"><span class="level-start"><span class="level-item">March 2021</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/archives/2021/02/"><span class="level-start"><span class="level-item">February 2021</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/archives/2021/01/"><span class="level-start"><span class="level-item">January 2021</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li><li><a class="level is-mobile" href="/archives/2020/12/"><span class="level-start"><span class="level-item">December 2020</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/archives/2020/05/"><span class="level-start"><span class="level-item">May 2020</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/archives/2020/04/"><span class="level-start"><span class="level-item">April 2020</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/archives/2020/03/"><span class="level-start"><span class="level-item">March 2020</span></span><span class="level-end"><span class="level-item tag">5</span></span></a></li><li><a class="level is-mobile" href="/archives/2020/02/"><span class="level-start"><span class="level-item">February 2020</span></span><span class="level-end"><span class="level-item tag">18</span></span></a></li></ul></div></div></div><div class="card widget" data-type="tags"><div class="card-content"><div class="menu"><h3 class="menu-label">Tags</h3><div class="field is-grouped is-grouped-multiline"><div class="control"><a class="tags has-addons" href="/tags/Linux/"><span class="tag">Linux</span><span class="tag">7</span></a></div><div class="control"><a class="tags has-addons" href="/tags/MySQL/"><span class="tag">MySQL</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/NGS/"><span class="tag">NGS</span><span class="tag">18</span></a></div><div class="control"><a class="tags has-addons" href="/tags/R/"><span class="tag">R</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/System/"><span class="tag">System</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/notes/"><span class="tag">notes</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/tag1/"><span class="tag">tag1</span><span class="tag">6</span></a></div></div></div></div></div><div class="card widget" data-type="subscribe-email"><div class="card-content"><div class="menu"><h3 class="menu-label">follow.it</h3><form action="" method="post" target="_blank"><div class="field has-addons"><div class="control has-icons-left is-expanded"><input class="input" name="email" type="email" placeholder="Email"><span class="icon is-small is-left"><i class="fas fa-envelope"></i></span></div><div class="control"><input class="button" type="submit" value="Subscribe"></div></div></form></div></div></div></div><!--!--></div></div></section><footer class="footer"><div class="container"><div class="level"><div class="level-start"><a class="footer-logo is-block mb-2" href="/"><img src="/img/ava.jpg" alt="Yuqi Gong&#039;s blog" height="28"></a><p class="is-size-7"><span>&copy; 2024 Yuqi Gong</span>  Powered by Gong Yuqi <i class="fa-heart fa fa-beat" style="color: #f00;"></i><br><span id="busuanzi_container_site_uv">Visited by <span id="busuanzi_value_site_uv">0</span> users</span></p></div><div class="level-end"><div class="field has-addons"><p class="control"><a class="button is-transparent is-large" target="_blank" rel="noopener" title="Creative Commons" href="https://creativecommons.org/"><i class="fab fa-creative-commons"></i></a></p><p class="control"><a class="button is-transparent is-large" target="_blank" rel="noopener" title="Attribution 4.0 International" href="https://creativecommons.org/licenses/by/4.0/"><i class="fab fa-creative-commons-by"></i></a></p><p class="control"><a class="button is-transparent is-large" target="_blank" rel="noopener" title="Download on GitHub" href="https://github.com/gongyuqiqi/gongyuqiqi.github.io"><i class="fab fa-github"></i></a></p></div></div></div></div></footer><script src="https://cdn.jsdelivr.net/npm/jquery@3.3.1/dist/jquery.min.js"></script><script src="https://cdn.jsdelivr.net/npm/moment@2.22.2/min/moment-with-locales.min.js"></script><script src="https://cdn.jsdelivr.net/npm/clipboard@2.0.4/dist/clipboard.min.js" defer></script><script>moment.locale("en");</script><script>var IcarusThemeSettings = {
            article: {
                highlight: {
                    clipboard: true,
                    fold: 'unfolded'
                }
            }
        };</script><script src="/js/column.js"></script><script src="/js/animation.js"></script><a id="back-to-top" title="Back to top" href="javascript:;"><i class="fas fa-chevron-up"></i></a><script src="/js/back_to_top.js" defer></script><!--!--><!--!--><!--!--><!--!--><script src="https://cdn.jsdelivr.net/npm/cookieconsent@3.1.1/build/cookieconsent.min.js" defer></script><script>window.addEventListener("load", () => {
      window.cookieconsent.initialise({
        type: "info",
        theme: "edgeless",
        static: false,
        position: "bottom-left",
        content: {
          message: "This website uses cookies to improve your experience.",
          dismiss: "Got it!",
          allow: "Allow cookies",
          deny: "Decline",
          link: "Learn more",
          policy: "Cookie Policy",
          href: "https://www.cookiesandyou.com/",
        },
        palette: {
          popup: {
            background: "#edeff5",
            text: "#838391"
          },
          button: {
            background: "#4b81e8"
          },
        },
      });
    });</script><script src="https://cdn.jsdelivr.net/npm/lightgallery@1.10.0/dist/js/lightgallery.min.js" defer></script><script src="https://cdn.jsdelivr.net/npm/justifiedGallery@3.8.1/dist/js/jquery.justifiedGallery.min.js" defer></script><script>window.addEventListener("load", () => {
            if (typeof $.fn.lightGallery === 'function') {
                $('.article').lightGallery({ selector: '.gallery-item' });
            }
            if (typeof $.fn.justifiedGallery === 'function') {
                if ($('.justified-gallery > p > .gallery-item').length) {
                    $('.justified-gallery > p > .gallery-item').unwrap();
                }
                $('.justified-gallery').justifiedGallery();
            }
        });</script><!--!--><!--!--><!--!--><!--!--><!--!--><script src="/js/main.js" defer></script><div class="searchbox"><div class="searchbox-container"><div class="searchbox-header"><div class="searchbox-input-container"><input class="searchbox-input" type="text" placeholder="Type something..."></div><a class="searchbox-close" href="javascript:;">×</a></div><div class="searchbox-body"></div></div></div><script src="/js/insight.js" defer></script><script>document.addEventListener('DOMContentLoaded', function () {
            loadInsight({"contentUrl":"/content.json"}, {"hint":"Type something...","untitled":"(Untitled)","posts":"Posts","pages":"Pages","categories":"Categories","tags":"Tags"});
        });</script></body></html>