<!doctype html>
<html lang="en"><head><meta charset="utf-8"><meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1"><meta><title>2020 ATAC-seq analysis - Yuqi Gong&#039;s blog</title><link rel="manifest" href="/manifest.json"><meta name="application-name" content="Yuqi Gong&#039;s blog"><meta name="msapplication-TileImage" content="/img/ava.jpg"><meta name="apple-mobile-web-app-capable" content="yes"><meta name="apple-mobile-web-app-title" content="Yuqi Gong&#039;s blog"><meta name="apple-mobile-web-app-status-bar-style" content="default"><meta name="description" content="ATAC-seq最新处理流程"><meta property="og:type" content="blog"><meta property="og:title" content="2020 ATAC-seq analysis"><meta property="og:url" content="http://gyqgyq.com/2020%20ATAC-seq%20analysis/"><meta property="og:site_name" content="Yuqi Gong&#039;s blog"><meta property="og:description" content="ATAC-seq最新处理流程"><meta property="og:locale" content="en_US"><meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/conda environment.JPG"><meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/multiqc.JPG"><meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/mapped.JPG"><meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/NRF-PBC1-PBC2.jpeg"><meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/NRF PBC1 PBC2.JPG"><meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/SRR12692092.INSERT.PLOT.jpg"><meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/SRR12692093.INSERT.PLOT.jpg"><meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/SRR12692098.INSERT.PLOT.jpg"><meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/SRR12692099.INSERT.PLOT.jpg"><meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/BRM014-10uM_24h_wt_TSS_rainbow.png"><meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/BRM014-10uM_24h_wt_TSS_genebody_TSE.png"><meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/cc.qc.JPG"><meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/SRR12692092.cc.plot.jpg"><meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/SRR12692093.cc.plot.jpg"><meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/SRR12692098.cc.plot.jpg"><meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/SRR12692099.cc.plot.jpg"><meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/before_after_blacklist-bed.JPG"><meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/narrowPeak for idr.JPG"><meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/SRR12692092_SRR12692093_idr_command.png"><meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/SRR12692092_SRR12692093_idr.png"><meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/SRR12692098_SRR12692099_idr_command.png"><meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/SRR12692098_SRR12692099_idr.png"><meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/FRiP.JPG"><meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/Feature Distribution.jpeg"><meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/Feature Distribution TSS.jpeg"><meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/plotAnnoPie.jpeg"><meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/upsetplot(peakAnno, vennpie=TRUE).jpeg"><meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/DMSO_24h_wt_rep1.jpeg"><meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/DMSO_24h_wt_rep2.jpeg"><meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/BRM014_10uM_24h_wt_rep1.jpeg"><meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/BRM014_10uM_24h_wt_rep2.jpeg"><meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/DMSO_24h_wt.png"><meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/BRM014_10uM_24h_wt.png"><meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/DMSO_BRM014.png"><meta property="article:published_time" content="2020-12-08T05:53:55.000Z"><meta property="article:modified_time" content="2022-11-26T12:54:48.568Z"><meta property="article:author" content="Yuqi Gong"><meta property="article:tag" content="NGS"><meta property="twitter:card" content="summary"><meta property="twitter:image:src" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/conda environment.JPG"><script type="application/ld+json">{"@context":"https://schema.org","@type":"BlogPosting","mainEntityOfPage":{"@type":"WebPage","@id":"http://gyqgyq.com/2020%20ATAC-seq%20analysis/"},"headline":"2020 ATAC-seq analysis","image":["https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/SRR12692092.INSERT.PLOT.jpg","https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/SRR12692093.INSERT.PLOT.jpg","https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/SRR12692098.INSERT.PLOT.jpg","https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/SRR12692099.INSERT.PLOT.jpg","https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/BRM014-10uM_24h_wt_TSS_rainbow.png","https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/BRM014-10uM_24h_wt_TSS_genebody_TSE.png","https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/SRR12692092.cc.plot.jpg","https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/SRR12692093.cc.plot.jpg","https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/SRR12692098.cc.plot.jpg","https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/SRR12692099.cc.plot.jpg","https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/SRR12692092_SRR12692093_idr_command.png","https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/SRR12692092_SRR12692093_idr.png","https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/SRR12692098_SRR12692099_idr_command.png","https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/SRR12692098_SRR12692099_idr.png","https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/DMSO_24h_wt.png","https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/BRM014_10uM_24h_wt.png","https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/DMSO_BRM014.png"],"datePublished":"2020-12-08T05:53:55.000Z","dateModified":"2022-11-26T12:54:48.568Z","author":{"@type":"Person","name":"Yuqi Gong"},"publisher":{"@type":"Organization","name":"Yuqi Gong's blog","logo":{"@type":"ImageObject","url":"http://gyqgyq.com/img/ava.jpg"}},"description":"ATAC-seq最新处理流程"}</script><link rel="canonical" href="http://gyqgyq.com/2020%20ATAC-seq%20analysis/"><link rel="icon" href="/img/ava.jpg"><link rel="stylesheet" href="https://use.fontawesome.com/releases/v6.0.0/css/all.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/highlight.js@9.12.0/styles/atom-one-light.css"><link rel="stylesheet" href="https://fonts.googleapis.com/css2?family=Ubuntu:wght@400;600&amp;family=Source+Code+Pro"><link rel="stylesheet" href="/css/default.css"><style>body>.footer,body>.navbar,body>.section{opacity:0}</style><!--!--><!--!--><!--!--><script src="//busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js" defer></script><!--!--><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/cookieconsent@3.1.1/build/cookieconsent.min.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/lightgallery@1.10.0/dist/css/lightgallery.min.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/justifiedGallery@3.8.1/dist/css/justifiedGallery.min.css"><!--!--><!--!--><style>.pace{-webkit-pointer-events:none;pointer-events:none;-webkit-user-select:none;-moz-user-select:none;user-select:none}.pace-inactive{display:none}.pace .pace-progress{background:#3273dc;position:fixed;z-index:2000;top:0;right:100%;width:100%;height:2px}</style><script src="https://cdn.jsdelivr.net/npm/pace-js@1.2.4/pace.min.js"></script><!--!--><!--!--><!-- hexo injector head_end start --><script>
  (function () {
      function switchTab() {
          if (!location.hash) {
            return;
          }

          const $tabMenu = document.querySelector(`a[href="${location.hash}"]`);
          if (!$tabMenu) {
            return;
          }

          const $tabMenuContainer = $tabMenu.parentElement.parentElement;
          Array.from($tabMenuContainer.children).forEach($menu => $menu.classList.remove('is-active'));
          Array.from($tabMenuContainer.querySelectorAll('a'))
              .map($menu => document.getElementById($menu.getAttribute("href").substring(1)))
              .forEach($content => $content.classList.add('is-hidden'));

          if ($tabMenu) {
              $tabMenu.parentElement.classList.add('is-active');
          }
          const $activeTab = document.querySelector(location.hash);
          if ($activeTab) {
              $activeTab.classList.remove('is-hidden');
          }
      }
      switchTab();
      window.addEventListener('hashchange', switchTab, false);
  })();
  </script><!-- hexo injector head_end end --><meta name="generator" content="Hexo 6.3.0"></head><body class="is-2-column"><nav class="navbar navbar-main"><div class="container navbar-container"><div class="navbar-brand justify-content-center"><a class="navbar-item navbar-logo" href="/"><img src="/img/ava.jpg" alt="Yuqi Gong&#039;s blog" height="28"></a></div><div class="navbar-menu"><div class="navbar-start"><a class="navbar-item" href="/">Home</a><a class="navbar-item" href="/archives">Archives</a><a class="navbar-item" href="/categories">Categories</a><a class="navbar-item" href="/tags">Tags</a><a class="navbar-item" href="/about">About</a></div><div class="navbar-end"><a class="navbar-item" target="_blank" rel="noopener" title="Download on GitHub" href="https://github.com/gongyuqiqi/gongyuqiqi.github.io"><i class="fab fa-github"></i></a><a class="navbar-item is-hidden-tablet catalogue" title="Catalogue" href="javascript:;"><i class="fas fa-list-ul"></i></a><a class="navbar-item search" title="Search" href="javascript:;"><i class="fas fa-search"></i></a></div></div></div></nav><section class="section"><div class="container"><div class="columns"><div class="column order-2 column-main is-8-tablet is-8-desktop is-8-widescreen"><div class="card"><article class="card-content article" role="article"><div class="article-meta is-size-7 is-uppercase level is-mobile"><div class="level-left"><span class="level-item">Posted&nbsp;<time dateTime="2020-12-08T05:53:55.000Z" title="12/8/2020, 1:53:55 PM">2020-12-08</time></span><span class="level-item">Updated&nbsp;<time dateTime="2022-11-26T12:54:48.568Z" title="11/26/2022, 8:54:48 PM">2022-11-26</time></span><span class="level-item">31 minutes read (About 4681 words)</span><span class="level-item" id="busuanzi_container_page_pv"><span id="busuanzi_value_page_pv">0</span>&nbsp;visits</span></div></div><h1 class="title is-3 is-size-4-mobile">2020 ATAC-seq analysis</h1><div class="content"><html><head></head><body><p>ATAC-seq最新处理流程</p>
<span id="more"></span>

<h1 id="2020-ATAC-seq-analysis"><a href="#2020-ATAC-seq-analysis" class="headerlink" title="2020 ATAC-seq analysis"></a><strong>2020 ATAC-seq analysis</strong></h1><h2 id="一、ATAC-seq个性化的conda-environment的搭建"><a href="#一、ATAC-seq个性化的conda-environment的搭建" class="headerlink" title="一、ATAC-seq个性化的conda environment的搭建"></a>一、ATAC-seq个性化的conda environment的搭建</h2><h3 id="1、git-clone"><a href="#1、git-clone" class="headerlink" title="1、git clone"></a>1、git clone</h3><p>git clone文章中提到的atac-seq-pipeline (里面有你接下来分析所需要的绝大部分东西。另外一定要去看README.md，这个文档会告诉你分析ATAC-seq需要做些准备，进行些什么工作。)</p>
<figure class="highlight bash"><table><tbody><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">$ <span class="built_in">cd</span> ~/ATAC</span><br><span class="line">$ git <span class="built_in">clone</span> https://github.com/ENCODE-DCC/atac-seq-pipeline</span><br></pre></td></tr></tbody></table></figure>

<h3 id="2、环境搭建"><a href="#2、环境搭建" class="headerlink" title="2、环境搭建"></a>2、环境搭建</h3><p>（1）conda base environment搭建</p>
<figure class="highlight bash"><table><tbody><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#一路yes下去</span></span><br><span class="line">$ wget https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh</span><br><span class="line">$ bash Miniconda3-4.6.14-Linux-x86_64.sh</span><br><span class="line"><span class="comment">#最后一个yes时你同意了每次自动启动conda的base环境，显示安装完成后，一定要退出当前的终端，再重新打开一次，你就会发现在游标的最左边多了一个（base）,你的base conda environment就激活啦！你的conda环境搭建就成功了一小步(*^__^*) 嘻嘻……</span></span><br></pre></td></tr></tbody></table></figure>

<p>（2）失活掉base conda environment自动激活的设置</p>
<figure class="highlight bash"><table><tbody><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">$ conda config --<span class="built_in">set</span> auto_activate_base <span class="literal">false</span></span><br><span class="line"><span class="comment">#同样的完成上诉操作后，记得退出当前终端，再次打开一个新的终端。你就会发现游标的最左边没有（base）了</span></span><br></pre></td></tr></tbody></table></figure>

<p>（3）安装pipeline’s conda environment</p>
<p>！？你是不是充满了疑问！？下面的两个shell脚本哪里冒出来的，干什么用的。不要忘了，我们第一步git clone了这个atac-seq-pipeline呢！这两个脚本就是就是来源于这个clone下来的pipeline的。</p>
<figure class="highlight bash"><table><tbody><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 网不给力的朋友，比如说我，这一部就是整个分析流程的限速步骤！！！网络给力这一步也需要花一些时间，读一下两个shell文件就知道为什么啦。</span></span><br><span class="line">$ bash scripts/uninstall_conda_env.sh  <span class="comment"># uninstall it for clean-install</span></span><br><span class="line"><span class="comment">#下面这个shell脚本会创建两个分析ATAC-seq数据所需要的环境，一个基于python3，一个基于python2的</span></span><br><span class="line">$ bash scripts/install_conda_env.sh</span><br></pre></td></tr></tbody></table></figure>

<p>（4）激活安装的conda环境</p>
<figure class="highlight bash"><table><tbody><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">$ conda activate encode-atac-seq-pipeline</span><br><span class="line"><span class="comment">#你一定会疑问为什么是encode-atac-seq-pipeline,阅读一下scripts/install_conda_env.sh这个脚本就知道啦。因为源文档作者把这个环境名命名为encode-atac-seq-pipeline。</span></span><br></pre></td></tr></tbody></table></figure>
<p>（5）查看所有的conda环境</p>
<figure class="highlight bash"><table><tbody><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">conda <span class="built_in">env</span> list</span><br></pre></td></tr></tbody></table></figure>
<p>我们会看到有一个基于python2的环境，没有标明python版本的那个encode-atac-seq-pipeline是基于python3的。有些软件是要基于特定版本的python运行的。<br><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/conda environment.JPG"></p>
<h2 id="二、数据下载"><a href="#二、数据下载" class="headerlink" title="二、数据下载"></a>二、数据下载</h2><figure class="highlight bash"><table><tbody><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#一次性下载所有的......_1.fastq.gz和......_2.fastq.gz样本</span></span><br><span class="line"><span class="built_in">dir</span>=/home/gongyuqi/.aspera/connect/etc/asperaweb_id_dsa.openssh</span><br><span class="line">x=_1</span><br><span class="line">y=_2</span><br><span class="line"><span class="keyword">for</span> <span class="built_in">id</span> <span class="keyword">in</span> {93,98,99}</span><br><span class="line"><span class="keyword">do</span></span><br><span class="line">ascp -QT -l 300m -P33001 -i <span class="variable">$dir</span> era-fasp@fasp.sra.ebi.ac.uk:/vol1/fastq/SRR126/0<span class="variable">$id</span>/SRR126920<span class="variable">$id</span>/SRR126920$<span class="built_in">id</span><span class="variable">$x</span>.fastq.gz .</span><br><span class="line">ascp -QT -l 300m -P33001 -i <span class="variable">$dir</span> era-fasp@fasp.sra.ebi.ac.uk:/vol1/fastq/SRR126/0<span class="variable">$id</span>/SRR126920<span class="variable">$id</span>/SRR126920$<span class="built_in">id</span><span class="variable">$y</span>.fastq.gz . </span><br><span class="line"><span class="keyword">done</span></span><br></pre></td></tr></tbody></table></figure>


<h2 id="三、bowtie2比对"><a href="#三、bowtie2比对" class="headerlink" title="三、bowtie2比对"></a>三、bowtie2比对</h2><h3 id="1、-fastqc-multiqc查看原始数据的质量情况"><a href="#1、-fastqc-multiqc查看原始数据的质量情况" class="headerlink" title="1、 fastqc+multiqc查看原始数据的质量情况"></a>1、 fastqc+multiqc查看原始数据的质量情况</h3><figure class="highlight bash"><table><tbody><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">conda install -y fastqc multiqc</span><br><span class="line"><span class="built_in">cd</span> raw </span><br><span class="line">fastqc -t 8 *gz -o ./</span><br><span class="line">multiqc *.gzip -o ./</span><br></pre></td></tr></tbody></table></figure>
<p>multiqc结果表明原始数据质量已经很好了，完全没有adaptor残留,连PCR重复都是勉强过关的。per base sequence content和kmer content两项指标不合格，但是并不会对后续的比对造成多大的负面影响，同时这两项指标通过过滤软件的处理可能并不会后得到改善。这样的数据其实可以直接比对的。<br><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/multiqc.JPG"></p>
<h3 id="2、-过滤低质量的reads"><a href="#2、-过滤低质量的reads" class="headerlink" title="2、 过滤低质量的reads"></a>2、 过滤低质量的reads</h3><p>（如果原始数据需要进行过滤处理，那么走下面这个过滤流程，我们这里就不过滤了）</p>
<figure class="highlight bash"><table><tbody><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><span class="line"><span class="built_in">nohup</span> <span class="built_in">cat</span> rawdata.txt|<span class="keyword">while</span> <span class="built_in">read</span> <span class="built_in">id</span></span><br><span class="line"><span class="keyword">do</span></span><br><span class="line">arr=(<span class="variable">${id}</span>)</span><br><span class="line">fq1=<span class="variable">${arr[0]}</span></span><br><span class="line">fq2=<span class="variable">${arr[1]}</span></span><br><span class="line">trim_galore -j 35 --phred33 --length 35 -e 0.1 --stringency 4 --paired -o ../clean <span class="variable">$fq1</span> <span class="variable">$fq2</span></span><br><span class="line"><span class="keyword">done</span> &amp;</span><br><span class="line"><span class="comment">#过滤后的数据质量控制，看数据是否有得到改善，有没有去掉adaptor非常重要</span></span><br><span class="line"><span class="built_in">cd</span> clean</span><br><span class="line">fastqc -t 20 *gz -o ./</span><br><span class="line">multiqc *.gzip -o ./</span><br></pre></td></tr></tbody></table></figure>



<h3 id="3、-比对参考基因组"><a href="#3、-比对参考基因组" class="headerlink" title="3、 比对参考基因组"></a>3、 比对参考基因组</h3><ul>
<li><p>比对</p>
<figure class="highlight bash"><table><tbody><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line"><span class="built_in">nohup</span> <span class="built_in">cat</span> sample.txt | <span class="keyword">while</span> <span class="built_in">read</span> <span class="built_in">id</span></span><br><span class="line"><span class="keyword">do</span></span><br><span class="line">arr=(<span class="variable">${id}</span>)</span><br><span class="line">fq1=<span class="variable">${arr[0]}</span></span><br><span class="line">fq2=<span class="variable">${arr[1]}</span></span><br><span class="line">sample=<span class="variable">${fq1%%_*}</span>.sort.bam</span><br><span class="line">bowtie2  --very-sensitive -X2000 --mm --threads 8 -x /home/gongyuqi/ref/mm10/index/mm10 -1 <span class="variable">$fq1</span> -2 <span class="variable">$fq2</span>|samtools <span class="built_in">sort</span> -O bam -@ 8 - -o ../align/<span class="variable">$sample</span> </span><br><span class="line"><span class="keyword">done</span> &amp;</span><br></pre></td></tr></tbody></table></figure>

</li>
<li><p>比对情况统计及查看</p>
<figure class="highlight bash"><table><tbody><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line"><span class="built_in">ls</span> *.bam | <span class="keyword">while</span> <span class="built_in">read</span> <span class="built_in">id</span></span><br><span class="line"><span class="keyword">do</span></span><br><span class="line"><span class="built_in">nohup</span> samtools flagstat -@ 4 <span class="variable">$id</span>  &gt; <span class="variable">${id%%.*}</span>.<span class="built_in">stat</span> &amp;</span><br><span class="line"><span class="keyword">done</span></span><br><span class="line"><span class="built_in">ls</span> *.<span class="built_in">stat</span> | <span class="keyword">while</span> <span class="built_in">read</span> <span class="built_in">id</span>; <span class="keyword">do</span> <span class="built_in">cat</span> <span class="variable">$id</span> | grep <span class="string">"mapped ("</span>;<span class="keyword">done</span></span><br></pre></td></tr></tbody></table></figure>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/mapped.JPG">
</li>
<li><p>生成bam文件的索引文件</p>
<figure class="highlight bash"><table><tbody><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line"><span class="built_in">nohup</span> <span class="built_in">ls</span> *sort.bam|<span class="keyword">while</span> <span class="built_in">read</span> <span class="built_in">id</span>; <span class="keyword">do</span> samtools index -@ 10 <span class="variable">$id</span>;<span class="keyword">done</span> &amp;</span><br></pre></td></tr></tbody></table></figure></li>
</ul>
<h2 id="四、PCR去除重及质控制"><a href="#四、PCR去除重及质控制" class="headerlink" title="四、PCR去除重及质控制"></a>四、PCR去除重及质控制</h2><h3 id="1、初步过滤"><a href="#1、初步过滤" class="headerlink" title="1、初步过滤"></a>1、初步过滤</h3><p>#=============================<br>#Remove unmapped, mate unmapped<br>#not primary alignment, reads failing platform<br>#Only keep properly paired reads<br>#Obtain name sorted BAM file<br>#=============================================</p>
<figure class="highlight bash"><table><tbody><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#fixmate要求输入的bam文件按名称排序，所以第一步初步过滤后要按reads名称排序一下</span></span><br><span class="line"><span class="built_in">nohup</span> <span class="built_in">ls</span> *.sort.bam|<span class="keyword">while</span> <span class="built_in">read</span> <span class="built_in">id</span>; <span class="keyword">do</span> samtools view -@ 8 -F 524 -f 2 -u <span class="variable">$id</span> | samtools <span class="built_in">sort</span> -@ 8 -n /dev/stdin -o ../filter/<span class="variable">${id%%.*}</span>.filt.nmsrt.bam;<span class="keyword">done</span> &amp;</span><br><span class="line"><span class="comment">#multimapping=4是bowtie2的默认设置</span></span><br><span class="line"><span class="built_in">cd</span> ../filter</span><br><span class="line">multimapping=4 </span><br><span class="line"><span class="built_in">nohup</span> <span class="built_in">ls</span> *filt.nmsrt.bam|<span class="keyword">while</span> <span class="built_in">read</span> <span class="built_in">id</span>;<span class="keyword">do</span> samtools view -@ 8 -h <span class="variable">$id</span>|assign_multimappers.py -k <span class="variable">$multimapping</span> --paired-end |samtools fixmate -@ 8 -r /dev/stdin <span class="variable">${id%%.*}</span>.filt.nmsrt.fixmate.bam; <span class="keyword">done</span> &amp;</span><br></pre></td></tr></tbody></table></figure>
<h3 id="2、进一步"><a href="#2、进一步" class="headerlink" title="2、进一步"></a>2、进一步</h3><p>#===============================================<br>#Remove orphan reads (pair was removed)<br>#and read pairs mapping to different chromosomes<br>#Obtain position sorted BAM<br>#===============================================</p>
<figure class="highlight bash"><table><tbody><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#多个样本写循环</span></span><br><span class="line"><span class="built_in">nohup</span> <span class="built_in">ls</span> *.filt.nmsrt.fixmate.bam|<span class="keyword">while</span> <span class="built_in">read</span> <span class="built_in">id</span>; <span class="keyword">do</span> samtools view -@ 8 -F 1804 -f 2 -u <span class="variable">$id</span> | samtools <span class="built_in">sort</span> -@ 8 /dev/stdin -o <span class="variable">${id%%.*}</span>.prefilt.bam;<span class="keyword">done</span> &amp;</span><br></pre></td></tr></tbody></table></figure>

<h3 id="3、标记PCR重复"><a href="#3、标记PCR重复" class="headerlink" title="3、标记PCR重复"></a>3、标记PCR重复</h3><p>#===============<br>#Mark duplicates<br>#===============</p>
<figure class="highlight bash"><table><tbody><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">MARKDUP=/home/gongyuqi/miniconda3/envs/encode-atac-seq-pipeline/share/picard-2.20.7-0/picard.jar</span><br><span class="line"><span class="comment">#多个样本写循环</span></span><br><span class="line"><span class="built_in">nohup</span> <span class="built_in">ls</span> *.prefilt.bam|<span class="keyword">while</span> <span class="built_in">read</span> <span class="built_in">id</span>;<span class="keyword">do</span> java -Xmx16G -jar <span class="variable">$MARKDUP</span> MarkDuplicates INPUT=<span class="variable">$id</span> OUTPUT=<span class="variable">${id%%.*}</span>.prefilt.dupmark.bam METRICS_FILE=<span class="variable">${id%%.*}</span>.prefilt.dupmark.qc VALIDATION_STRINGENCY=LENIENT ASSUME_SORTED=<span class="literal">true</span> REMOVE_DUPLICATES=<span class="literal">false</span>;<span class="keyword">done</span> &amp;</span><br></pre></td></tr></tbody></table></figure>
<h3 id="4、去除PCR重复、建立索引文件、按name排序的bam文件、质控文件"><a href="#4、去除PCR重复、建立索引文件、按name排序的bam文件、质控文件" class="headerlink" title="4、去除PCR重复、建立索引文件、按name排序的bam文件、质控文件"></a>4、去除PCR重复、建立索引文件、按name排序的bam文件、质控文件</h3><p>#============================<br>#Remove duplicates<br>#Index final position sorted BAM<br>#Create final name sorted BAM<br>#============================</p>
<figure class="highlight bash"><table><tbody><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#多个样本写循环</span></span><br><span class="line"><span class="comment">#去重</span></span><br><span class="line"><span class="built_in">nohup</span> <span class="built_in">ls</span> *.prefilt.dupmark.bam|<span class="keyword">while</span> <span class="built_in">read</span> <span class="built_in">id</span>;<span class="keyword">do</span> samtools view -@ 16 -F 1804 -f 2 -b <span class="variable">$id</span> &gt; <span class="variable">${id%%.*}</span>.nodup.bam;<span class="keyword">done</span> &amp;</span><br><span class="line"><span class="comment">#构建索引文件</span></span><br><span class="line"><span class="built_in">nohup</span> <span class="built_in">ls</span> *nodup.bam|<span class="keyword">while</span> <span class="built_in">read</span> <span class="built_in">id</span>;<span class="keyword">do</span> samtools index <span class="variable">$id</span>;<span class="keyword">done</span> &amp;</span><br></pre></td></tr></tbody></table></figure>

<h3 id="5、测序文库复杂度的检验"><a href="#5、测序文库复杂度的检验" class="headerlink" title="5、测序文库复杂度的检验"></a>5、测序文库复杂度的检验</h3><p>#=============================<br>#Compute library complexity<br>#=============================<br>#Sort by name<br>#convert to bedPE and obtain fragment coordinates<br>#sort by position and strand<br>#Obtain unique count statistics<br>#================================================</p>
<figure class="highlight bash"><table><tbody><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#多个样本写循环</span></span><br><span class="line"><span class="built_in">nohup</span> <span class="built_in">ls</span> *.nodup.bam|<span class="keyword">while</span> <span class="built_in">read</span> <span class="built_in">id</span>;<span class="keyword">do</span> samtools <span class="built_in">sort</span> -@ 16 -n <span class="variable">$id</span> -o <span class="variable">${id%%.*}</span>.nodup.srt.name.bam;<span class="keyword">done</span> &amp;</span><br><span class="line"><span class="built_in">nohup</span> <span class="built_in">ls</span> *.nodup.srt.name.bam|<span class="keyword">while</span> <span class="built_in">read</span> <span class="built_in">id</span>; <span class="keyword">do</span> bedtools bamtobed -bedpe -i <span class="variable">$id</span> | awk <span class="string">'BEGIN{OFS="\t"}{print $1,$2,$4,$6,$9,$10}'</span> | grep -v <span class="string">'chrM'</span> | <span class="built_in">sort</span> | <span class="built_in">uniq</span> -c | awk <span class="string">'BEGIN{mt=0;m0=0;m1=0;m2=0} ($1==1){m1=m1+1} ($1==2){m2=m2+1} {m0=m0+1} {mt=mt+$1} END{m1_m2=-1.0; if(m2&gt;0) m1_m2=m1/m2;printf "%d\t%d\t%d\t%d\t%f\t%f\t%f\n",mt,m0,m1,m2,m0/mt,m1/m0,m1_m2}'</span> &gt; <span class="variable">${id%%.*}</span>.nodup.pbc.qc;<span class="keyword">done</span> &amp;</span><br></pre></td></tr></tbody></table></figure>

<p>Library complexity measures计算结果如下，…nodup.pbc.qc文件格式为：<br>TotalReadPairs<br>DistinctReadPairs<br>OneReadPair<br>TwoReadPairs<br>NRF=Distinct/Total<br>PBC1=OnePair/Distinct<br>PBC2=OnePair/TwoPair</p>
<p>针对NRF、PBC1、PBC2这几个指标，ENCODE官网提供的标准如下:</p>
<p><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/NRF-PBC1-PBC2.jpeg"></p>
<p>我们此次4个样本的分析结果显示这几个指标的值如下所示：</p>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/NRF PBC1 PBC2.JPG">

<p>除了第一个样本（SRR12692092），其他样本计算结果显示NRF、PBC1、PBC2的值都非常完美，说明我们进行过滤和PCR去重的bam文件质量上没有问题，可以用于后续的分析。</p>
<h3 id="6、插入片段质控检测"><a href="#6、插入片段质控检测" class="headerlink" title="6、插入片段质控检测"></a>6、插入片段质控检测</h3><table><tbody><tr><td bgcolor="GreenYellow">用于评估ATAC/Chip实验质量好坏的一个重要指标</td></tr></tbody></table>

<figure class="highlight bash"><table><tbody><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#多个样本循环</span></span><br><span class="line">picard_jar=/home/gongyuqi/miniconda3/envs/encode-atac-seq-pipeline/share/picard-2.20.7-0/picard.jar</span><br><span class="line"><span class="built_in">nohup</span> <span class="built_in">ls</span> *.nodup.srt.name.bam | <span class="keyword">while</span> <span class="built_in">read</span> <span class="built_in">id</span>; <span class="keyword">do</span> java -Xmx6G -XX:ParallelGCThreads=1 -jar <span class="variable">${picard_jar}</span> CollectInsertSizeMetrics INPUT=<span class="variable">$id</span> OUTPUT=<span class="variable">${id%%.*}</span>.INSERT.DATA H=<span class="variable">${id%%.*}</span>.INSERT.PLOT.pdf VERBOSITY=ERROR QUIET=TRUE USE_JDK_DEFLATER=TRUE USE_JDK_INFLATER=TRUE W=1000 STOP_AFTER=5000000; <span class="keyword">done</span> &amp;</span><br></pre></td></tr></tbody></table></figure>

<p>成功的ATAC-seq实验会生成具有逐级递减和周期性的峰，对应无核小体区域（Nucleosome free region, NRF,100bp）,单核区域（mono-nucleosome,200bp），双核区域（400bp），三核区域（600bp）。</p>
<p>本次结果显示样本质量OK</p>
<p><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/SRR12692092.INSERT.PLOT.jpg" width="50%"><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/SRR12692093.INSERT.PLOT.jpg" width="50%"></p>
<p><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/SRR12692098.INSERT.PLOT.jpg" width="50%"><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/SRR12692099.INSERT.PLOT.jpg" width="50%"></p>
<h3 id="7、TSS富集"><a href="#7、TSS富集" class="headerlink" title="7、TSS富集"></a>7、TSS富集</h3><p><strong>以BRM014-10uM_24h_wt样本为例，可视化peaks在基因组上的分布特征(TSS区域；TSS、gene-body、TSE区域)</strong></p>
<ul>
<li>both -R and -S can accept multiple files</li>
<li>UCSC官网下载小鼠参考基因组注释文件，详情参考：<a target="_blank" rel="noopener" href="http://www.bio-info-trainee.com/2494.html">http://www.bio-info-trainee.com/2494.html</a></li>
</ul>
<figure class="highlight bash"><table><tbody><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#生成bw文件用于后续的computeMatrix</span></span><br><span class="line"><span class="built_in">nohup</span> <span class="built_in">ls</span> *.nodup.bam |<span class="keyword">while</span> <span class="built_in">read</span> <span class="built_in">id</span></span><br><span class="line"><span class="keyword">do</span></span><br><span class="line"><span class="built_in">nohup</span> bamCoverage --normalizeUsing CPM -b <span class="variable">$id</span> -o <span class="variable">${id%%.*}</span>.nodup.bw</span><br><span class="line"><span class="keyword">done</span> &amp;</span><br><span class="line"></span><br><span class="line"><span class="comment">#注释文件绝对路劲</span></span><br><span class="line">refseq=/home/gongyuqi/ref/mm10/mouse.ucsc.refseq.bed</span><br><span class="line"></span><br><span class="line"><span class="comment">#TSS</span></span><br><span class="line"><span class="comment">##生成plotHeatmap需要的gz文件</span></span><br><span class="line"><span class="built_in">nohup</span> computeMatrix reference-point  --referencePoint TSS  -p 30 \</span><br><span class="line">-b 3000 -a 3000 \</span><br><span class="line">-R <span class="variable">$refseq</span> \</span><br><span class="line">-S SRR12692098.nodup.bw SRR12692099.nodup.bw \</span><br><span class="line">--skipZeros -o BRM014-10uM_24h_wt_TSS.gz \</span><br><span class="line">--outFileSortedRegions BRM014-10uM_24h_wt_TSS.bed &amp;</span><br><span class="line"><span class="comment">##可视化文件</span></span><br><span class="line"><span class="built_in">nohup</span> plotHeatmap -m BRM014-10uM_24h_wt_TSS.gz  -out BRM014-10uM_24h_wt_TSS_rainbow.png --colorMap rainbow &amp;</span><br><span class="line"></span><br><span class="line"><span class="comment">#TSS_gene-body_TSE</span></span><br><span class="line"><span class="comment">##生成plotHeatmap需要的gz文件</span></span><br><span class="line"><span class="built_in">nohup</span> computeMatrix scale-regions -p 30 \</span><br><span class="line">-a 3000 -b 3000 -m 5000 \</span><br><span class="line">-R <span class="variable">$refseq</span> \</span><br><span class="line">-S SRR12692098.nodup.bw SRR12692099.nodup.bw \</span><br><span class="line">--skipZeros -o BRM014-10uM_24h_wt_TSS_gene0body_TSE.gz \</span><br><span class="line">--outFileSortedRegions BRM014-10uM_24h_wt_TSS.bed &amp;</span><br><span class="line"><span class="comment">##可视化文件</span></span><br><span class="line"><span class="built_in">nohup</span> plotHeatmap -m BRM014-10uM_24h_wt_TSS_genebody_TSE.gz  -out BRM014-10uM_24h_wt_TSS_genebody_TSE.png --colorMap rainbow &amp;</span><br></pre></td></tr></tbody></table></figure>

<p><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/BRM014-10uM_24h_wt_TSS_rainbow.png" width="50%"><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/BRM014-10uM_24h_wt_TSS_genebody_TSE.png" width="50%"></p>
<h2 id="五、生成tagAlign格式文件"><a href="#五、生成tagAlign格式文件" class="headerlink" title="五、生成tagAlign格式文件"></a>五、生成tagAlign格式文件</h2><h3 id="1-Convert-PE-BAM-to-tagAlign"><a href="#1-Convert-PE-BAM-to-tagAlign" class="headerlink" title="1. Convert PE BAM to tagAlign"></a>1. Convert PE BAM to tagAlign</h3><ul>
<li>对于单端序列。直接用bed格式就可以；对于双端序列，推荐用bedpe格式。这两种格式都可以称之为tagAlign，可以作为macs的输入文件。</li>
<li>tagAligen格式相比bam，文件大小会小很多，更加方便文件的读取。在转换得到tagAlign格式之后，我们就可以很容易的将坐标进行偏移</li>
</ul>
<figure class="highlight bash"><table><tbody><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line"><span class="built_in">nohup</span> <span class="built_in">ls</span> *nodup.srt.name.bam|<span class="keyword">while</span> <span class="built_in">read</span> <span class="built_in">id</span>; <span class="keyword">do</span> bedtools bamtobed -bedpe -mate1 -i <span class="variable">$id</span>  | gzip -nc &gt; <span class="variable">${id%%.*}</span>.nodup.srt.name.bedpe.gz;<span class="keyword">done</span> &amp;</span><br><span class="line"><span class="comment">#含有chrM的染色体的TagAlign文件</span></span><br><span class="line"><span class="built_in">nohup</span> <span class="built_in">ls</span> *.nodup.srt.name.bedpe.gz | <span class="keyword">while</span> <span class="built_in">read</span> <span class="built_in">id</span>; <span class="keyword">do</span> zcat <span class="variable">$id</span> | awk <span class="string">'BEGIN{OFS="\t"}{printf "%s\t%s\t%s\tN\t1000\t%s\n%s\t%s\t%s\tN\t1000\t%s\n",$1,$2,$3,$9,$4,$5,$6,$10}'</span> | gzip -nc &gt; <span class="variable">${id%%.*}</span>.nodup.srt.name.tagAlign.gz; <span class="keyword">done</span> &amp;</span><br><span class="line"><span class="comment">#去除chrM的染色体的TagAlign文件</span></span><br><span class="line"><span class="built_in">nohup</span> <span class="built_in">ls</span> *nodup.srt.name.bedpe.gz|<span class="keyword">while</span> <span class="built_in">read</span> <span class="built_in">id</span>; <span class="keyword">do</span> zcat <span class="variable">$id</span> | grep -P -v <span class="string">"^chrM"</span> | awk <span class="string">'BEGIN{OFS="\t"}{printf "%s\t%s\t%s\tN\t1000\t%s\n%s\t%s\t%s\tN\t1000\t%s\n",$1,$2,$3,$9,$4,$5,$6,$10}'</span> | gzip -nc &gt; <span class="variable">${id%%.*}</span>.nodup.nomit.srt.name.tagAlign.gz; <span class="keyword">done</span></span><br></pre></td></tr></tbody></table></figure>

<h3 id="2-Stand-Cross-Correlation-analysis"><a href="#2-Stand-Cross-Correlation-analysis" class="headerlink" title="2. Stand Cross Correlation analysis"></a>2. Stand Cross Correlation analysis</h3><table><tbody><tr><td bgcolor="GreenYellow">用于评估ATAC/Chip实验质量好坏的一个重要指标</td></tr></tbody></table>

<figure class="highlight bash"><table><tbody><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">NREADS=25000000</span><br><span class="line"><span class="built_in">nohup</span> <span class="built_in">ls</span> *.nodup.srt.name.bedpe.gz | <span class="keyword">while</span> <span class="built_in">read</span> <span class="built_in">id</span>; <span class="keyword">do</span> zcat <span class="variable">$id</span> | grep -v “chrM” | <span class="built_in">shuf</span> -n <span class="variable">${NREADS}</span> --random-source=&lt;(openssl enc -aes-256-ctr -pass pass:$(zcat -f <span class="variable">${id%%.*}</span>.nodup.srt.name.tagAlign.gz | <span class="built_in">wc</span> -c) -nosalt &lt;/dev/zero 2&gt;/dev/null) | awk <span class="string">'BEGIN{OFS="\t"}{print $1,$2,$3,"N","1000",$9}'</span> | gzip -nc &gt; <span class="variable">${id%%.*}</span>.nodup.nomit.srt.name.$((NREADS / <span class="number">1000000</span>)).tagAlign.gz; <span class="keyword">done</span> &amp;</span><br><span class="line"><span class="comment">#命令最终会生成交叉相关质量评估文件，*.cc.qc文件中会输出包含11列的信息，重点关注9-11列的信息，cc.plot.pdf文件相当于*.cc.qc文件的可视化</span></span><br><span class="line"><span class="built_in">nohup</span> <span class="built_in">ls</span> *$((NREADS / <span class="number">1000000</span>)).tagAlign.gz | <span class="keyword">while</span> <span class="built_in">read</span> <span class="built_in">id</span>; <span class="keyword">do</span> Rscript $(<span class="built_in">which</span> run_spp.R) -c=<span class="variable">$id</span> -p=10 -filtchr=chrM -savp=<span class="variable">${id%%.*}</span>.cc.plot.pdf -out=<span class="variable">${id%%.*}</span>.cc.qc; <span class="keyword">done</span> &amp;</span><br><span class="line"><span class="comment">#质控结果查看，主要看NSC,RSC,Quality tag三个值即输出文件的第9列，第10列，第11列。</span></span><br><span class="line"><span class="built_in">ls</span> *.cc.qc|<span class="keyword">while</span> <span class="built_in">read</span> <span class="built_in">id</span>; <span class="keyword">do</span> <span class="built_in">cat</span> <span class="variable">$id</span> | awk <span class="string">'{print $9, "\t", $10, "\t", $11}'</span>;<span class="keyword">done</span> </span><br></pre></td></tr></tbody></table></figure>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/cc.qc.JPG" width="50%">
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/SRR12692092.cc.plot.jpg" width="50%"><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/SRR12692093.cc.plot.jpg" width="50%">

<p><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/SRR12692098.cc.plot.jpg" width="50%"><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/SRR12692099.cc.plot.jpg" width="50%"></p>
<ul>
<li>质控结果解读<blockquote>
<p>Normalized strand cross-correlation coefficent (NSC)：NSC是最大交叉相关值除以背景交叉相关的比率(所有可能的链转移的最小交叉相关值)。NSC值越大表明富集效果越好，NSC值低于1.1表明较弱的富集，小于1表示无富集。NSC值稍微低于1.05，有较低的信噪比或很少的峰，这肯能是生物学真实现象，比如有的因子在特定组织类型中只有很少的结合位点；也可能确实是数据质量差。</p>
<p>Relative strand cross-correlation coefficient (RSC)：RSC是片段长度相关值减去背景相关值除以phantom-peak相关值减去背景相关值。RSC的最小值可能是0，表示无信号；富集好的实验RSC值大于1；低于1表示质量低。</p>
<p>QualityTag: Quality tag based on thresholded RSC (codes: -2:veryLow,-1:Low,0:Medium,1:High,2:veryHigh)</p>
<p>查看交叉相关性质量评估结果，主要看NSC,RSC,Quality tag三个值，这三个值分别对应输出文件的第9列，第10列，第11列。</p>
</blockquote>
</li>
</ul>
<h2 id="六、Call-Peaks"><a href="#六、Call-Peaks" class="headerlink" title="六、Call Peaks"></a>六、Call Peaks</h2><h3 id="1、去除线粒体基因的TagAlign格式文件进行shift操作，输入macs2软件去callpeak"><a href="#1、去除线粒体基因的TagAlign格式文件进行shift操作，输入macs2软件去callpeak" class="headerlink" title="1、去除线粒体基因的TagAlign格式文件进行shift操作，输入macs2软件去callpeak"></a>1、去除线粒体基因的TagAlign格式文件进行shift操作，输入macs2软件去callpeak</h3><figure class="highlight bash"><table><tbody><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">smooth_window=150 <span class="comment"># default</span></span><br><span class="line">shiftsize=$(( -<span class="variable">$smooth_window</span>/<span class="number">2</span> ))</span><br><span class="line">pval_thresh=0.01</span><br><span class="line"><span class="built_in">nohup</span> <span class="built_in">ls</span> *nodup.nomit.srt.name.tagAlign.gz | <span class="keyword">while</span> <span class="built_in">read</span> <span class="built_in">id</span>; \</span><br><span class="line"><span class="keyword">do</span> macs2 callpeak \</span><br><span class="line">-t <span class="variable">$id</span> -f BED -n <span class="string">"<span class="variable">${id%%.*}</span>"</span> -g mm -p <span class="variable">$pval_thresh</span> \</span><br><span class="line">--<span class="built_in">shift</span> <span class="variable">$shiftsize</span> --extsize <span class="variable">$smooth_window</span> --nomodel -B --SPMR --keep-dup all --call-summits; \</span><br><span class="line"><span class="keyword">done</span> &amp;</span><br></pre></td></tr></tbody></table></figure>

<h3 id="2、去除ENCODE列入黑名单的区域"><a href="#2、去除ENCODE列入黑名单的区域" class="headerlink" title="2、去除ENCODE列入黑名单的区域"></a>2、去除ENCODE列入黑名单的区域</h3><ul>
<li><p>去除黑名单的bed文件，用于后续的peaks注释</p>
<figure class="highlight bash"><table><tbody><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">BLACKLIST=/home/gongyuqi/project/ATAC/mm10.blacklist.bed.gz</span><br><span class="line"><span class="comment">#*_summits.bed为macs2软件callpeak的结果文件之一</span></span><br><span class="line"><span class="built_in">nohup</span> <span class="built_in">ls</span> *_summits.bed | <span class="keyword">while</span> <span class="built_in">read</span> <span class="built_in">id</span>; <span class="keyword">do</span> bedtools intersect -a <span class="variable">$id</span> -b <span class="variable">$BLACKLIST</span> -v &gt; <span class="variable">${id%%.*}</span>_filt_blacklist.bed; <span class="keyword">done</span> &amp;</span><br><span class="line"><span class="comment">#查看过滤黑名单的区域前后的bed文件的peaks数</span></span><br><span class="line"><span class="built_in">ls</span> *summits.bed|<span class="keyword">while</span> <span class="built_in">read</span> <span class="built_in">id</span>; <span class="keyword">do</span> <span class="built_in">cat</span> <span class="variable">$id</span> |<span class="built_in">wc</span> -l &gt;&gt;summits.bed.txt;<span class="keyword">done</span></span><br><span class="line"><span class="built_in">ls</span> *summits_filt_blacklist.bed|<span class="keyword">while</span> <span class="built_in">read</span> <span class="built_in">id</span>; <span class="keyword">do</span> <span class="built_in">cat</span> <span class="variable">$id</span> |<span class="built_in">wc</span> -l &gt;&gt;summits_filt_blacklist.bed.txt;<span class="keyword">done</span></span><br><span class="line">past summits.bed.txt summits_filt_blacklist.bed.txt </span><br></pre></td></tr></tbody></table></figure>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/before_after_blacklist-bed.JPG">




</li>
<li><p>去除黑名单的narrowPeaks文件，用于后续的IDR评估</p>
<figure class="highlight bash"><table><tbody><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#使用IDR需要先对MACS2的结果文件narrowPeak根据-log10(p-value)进行排序,-log10(p-value)在第八列。</span></span><br><span class="line"><span class="comment"># Sort by Col8 in descending order and replace long peak names in Column 4 with Peak_&lt;peakRank&gt;</span></span><br><span class="line"><span class="comment">#*_peaks.narrowPeak为macs2软件callpeak的结果文件之一</span></span><br><span class="line">NPEAKS=300000 </span><br><span class="line"><span class="built_in">ls</span> *_peaks.narrowPeak | <span class="keyword">while</span> <span class="built_in">read</span> <span class="built_in">id</span>; <span class="keyword">do</span> <span class="built_in">sort</span> -k 8gr,8gr <span class="variable">$id</span> | awk <span class="string">'BEGIN{OFS="\t"}{$4="Peak_"NR ; print $0}'</span> | <span class="built_in">head</span> -n <span class="variable">${NPEAKS}</span> | gzip -nc &gt; <span class="variable">${id%%_*}</span>.narrowPeak.gz; <span class="keyword">done</span></span><br><span class="line"></span><br><span class="line">BLACKLIST=../BLACKLIST/mm10.blacklist.bed.gz</span><br><span class="line"></span><br><span class="line"><span class="comment">#生成不压缩文件</span></span><br><span class="line"><span class="built_in">ls</span> *.narrowPeak.gz | <span class="keyword">while</span> <span class="built_in">read</span> <span class="built_in">id</span>; <span class="keyword">do</span> bedtools intersect -v -a <span class="variable">$id</span> -b <span class="variable">${BLACKLIST}</span> | awk <span class="string">'BEGIN{OFS="\t"} {if ($5&gt;1000) $5=1000; print $0}'</span> | grep -P <span class="string">'chr[\dXY]+[ \t]'</span>  &gt; <span class="variable">${id%%.*}</span>.narrowPeak.filt_blacklist; <span class="keyword">done</span></span><br><span class="line"><span class="comment">#生成压缩文件</span></span><br><span class="line"><span class="comment">#ls *.narrowPeak.gz | while read id; do bedtools intersect -v -a $id -b ${BLACKLIST} | awk 'BEGIN{OFS="\t"} {if ($5&gt;1000) $5=1000; print $0}' | grep -P 'chr[\dXY]+[ \t]' | gzip -nc &gt; ${id%%.*}.narrowPeak.filt_blacklist.gz; done</span></span><br></pre></td></tr></tbody></table></figure></li>
</ul>
<h3 id="3、Irreproducibility-Discovery-Rate-IDR-评估"><a href="#3、Irreproducibility-Discovery-Rate-IDR-评估" class="headerlink" title="3、Irreproducibility Discovery Rate (IDR)评估"></a>3、Irreproducibility Discovery Rate (IDR)评估</h3><table><tbody><tr><td bgcolor="GreenYellow">用于评估重复样本间peaks一致性的重要指标</td></tr></tbody></table>
首先生成narrowPeak_sample.txt文件，方便后续循环处理，生成文件内容如下：
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/narrowPeak for idr.JPG">

<figure class="highlight bash"><table><tbody><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line"><span class="built_in">nohup</span> <span class="built_in">cat</span> narrowPeak_sample.txt | <span class="keyword">while</span> <span class="built_in">read</span> <span class="built_in">id</span></span><br><span class="line"><span class="keyword">do</span></span><br><span class="line">arr=(<span class="variable">${id}</span>)</span><br><span class="line">Rep1=<span class="variable">${arr[0]}</span></span><br><span class="line">Rep2=<span class="variable">${arr[1]}</span></span><br><span class="line">sample=<span class="variable">${Rep1%%.*}</span>_<span class="variable">${Rep2%%.*}</span>_idr</span><br><span class="line">idr --samples <span class="variable">$Rep1</span> <span class="variable">$Rep2</span> --input-file-type narrowPeak -o <span class="variable">$sample</span> --plot  </span><br><span class="line"><span class="keyword">done</span> &amp;</span><br></pre></td></tr></tbody></table></figure>

<ul>
<li>DMSO_24h_wt （样本处理情况）</li>
<li>SRR12692092.filt_blacklist.narrowPeak SRR12692093.filt_blacklist.narrowPeak</li>
<li>没有通过IDR阈值的显示为红色</li>
</ul>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/SRR12692092_SRR12692093_idr_command.png">

<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/SRR12692092_SRR12692093_idr.png">

<ul>
<li>BRM014-10uM_24h_wt （样本处理情况）</li>
<li>SRR12692098.filt_blacklist.narrowPeak SRR12692099.filt_blacklist.narrowPeak</li>
<li>没有通过IDR阈值的显示为红色</li>
</ul>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/SRR12692098_SRR12692099_idr_command.png">
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/SRR12692098_SRR12692099_idr.png">

<blockquote>
<p>IDR评估会同时考虑peaks间的overlap和富集倍数的一致性。通过IDR阈值（0.05）的占比越大，说明重复样本间peaks一致性越好。从idr的分析结果看，我们的测试数据还可以的呢。</p>
</blockquote>
<p><strong>IDR评估相关参考资料：</strong>  </p>
<p><a target="_blank" rel="noopener" href="https://www.jianshu.com/p/d8a7056b4294?utm_campaign=maleskine&amp;utm_content=note&amp;utm_medium=seo_notes&amp;utm_source=recommendation">重复样本的处理——IDR</a></p>
<h3 id="4、Fraction-of-reads-in-peaks-FRiP-评估"><a href="#4、Fraction-of-reads-in-peaks-FRiP-评估" class="headerlink" title="4、Fraction of reads in peaks (FRiP)评估"></a>4、Fraction of reads in peaks (FRiP)评估</h3><table><tbody><tr><td bgcolor="GreenYellow">反映样本富集效果好坏的评价指标</td></tr></tbody></table>

<figure class="highlight bash"><table><tbody><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#生成bed文件</span></span><br><span class="line"><span class="built_in">nohup</span> <span class="built_in">ls</span> *.nodup.bam|<span class="keyword">while</span> <span class="built_in">read</span> <span class="built_in">id</span>;<span class="keyword">do</span> (bedtools bamtobed -i <span class="variable">$id</span> &gt;<span class="variable">${id%%.*}</span>.nodup.bed) ;<span class="keyword">done</span> &amp;</span><br><span class="line"><span class="comment">#批量计算FRiP</span></span><br><span class="line"><span class="built_in">ls</span> *_summits_filt_blacklist.bed|<span class="keyword">while</span>  <span class="built_in">read</span> <span class="built_in">id</span>;</span><br><span class="line"><span class="keyword">do</span></span><br><span class="line"><span class="built_in">echo</span> <span class="variable">$id</span></span><br><span class="line">bed=<span class="variable">${id%%_*}</span>.nodup.bed </span><br><span class="line">Reads=$(bedtools intersect -a <span class="variable">$bed</span> -b <span class="variable">$id</span> |<span class="built_in">wc</span> -l|awk <span class="string">'{print $1}'</span>)</span><br><span class="line">totalReads=$(<span class="built_in">wc</span> -l <span class="variable">$bed</span>|awk <span class="string">'{print $1}'</span>)</span><br><span class="line"><span class="built_in">echo</span> <span class="variable">$Reads</span>  <span class="variable">$totalReads</span> </span><br><span class="line"><span class="built_in">echo</span> <span class="string">'==&gt; FRiP value:'</span> $(bc &lt;&lt;&lt; <span class="string">"scale=2;100*<span class="variable">$Reads</span>/<span class="variable">$totalReads</span>"</span>)<span class="string">'%'</span></span><br><span class="line"><span class="keyword">done</span></span><br></pre></td></tr></tbody></table></figure>
<p>FRiP值在5%以上算比较好的。但也不绝对，这是个软阈值，可以作为一个参考。</p>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/FRiP.JPG">

<p><strong>FRiP评估相关参考资料：</strong></p>
<p><a target="_blank" rel="noopener" href="https://www.jianshu.com/p/09e05bcd6981?utm_campaign=maleskine&amp;utm_content=note&amp;utm_medium=seo_notes&amp;utm_source=recommendation">https://www.jianshu.com/p/09e05bcd6981?utm_campaign=maleskine&amp;utm_content=note&amp;utm_medium=seo_notes&amp;utm_source=recommendation</a></p>
<h2 id="七、Peak-annotation"><a href="#七、Peak-annotation" class="headerlink" title="七、Peak annotation"></a>七、Peak annotation</h2><h3 id="1、Feature-Distribution"><a href="#1、Feature-Distribution" class="headerlink" title="1、Feature Distribution"></a>1、Feature Distribution</h3><figure class="highlight r"><table><tbody><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br></pre></td><td class="code"><pre><span class="line">setwd<span class="punctuation">(</span><span class="string">"path to bed file"</span><span class="punctuation">)</span></span><br><span class="line">library<span class="punctuation">(</span>ChIPpeakAnno<span class="punctuation">)</span></span><br><span class="line">library<span class="punctuation">(</span>TxDb.Mmusculus.UCSC.mm10.knownGene<span class="punctuation">)</span></span><br><span class="line">library<span class="punctuation">(</span>org.Mm.eg.db<span class="punctuation">)</span></span><br><span class="line">library<span class="punctuation">(</span>BiocInstaller<span class="punctuation">)</span></span><br><span class="line">library<span class="punctuation">(</span>ChIPseeker<span class="punctuation">)</span></span><br><span class="line">txdb <span class="operator">&lt;-</span> TxDb.Mmusculus.UCSC.mm10.knownGene</span><br><span class="line">promoter <span class="operator">&lt;-</span> getPromoters<span class="punctuation">(</span>TxDb<span class="operator">=</span>txdb<span class="punctuation">,</span> upstream<span class="operator">=</span><span class="number">3000</span><span class="punctuation">,</span> downstream<span class="operator">=</span><span class="number">3000</span><span class="punctuation">)</span></span><br><span class="line">files <span class="operator">=</span> <span class="built_in">list</span><span class="punctuation">(</span>DMSO_24h_wt_rep1 <span class="operator">=</span> <span class="string">"SRR12692092_summits_filt_blacklist.bed"</span><span class="punctuation">,</span></span><br><span class="line">             DMSO_24h_wt_rep2 <span class="operator">=</span> <span class="string">"SRR12692093_summits_filt_blacklist.bed"</span><span class="punctuation">,</span></span><br><span class="line">             BRM014_10uM_24h_wt_rep1 <span class="operator">=</span> <span class="string">"SRR12692098_summits_filt_blacklist.bed"</span><span class="punctuation">,</span></span><br><span class="line">             BRM014_10uM_24h_wt_rep2 <span class="operator">=</span> <span class="string">"SRR12692099_summits_filt_blacklist.bed"</span><span class="punctuation">)</span></span><br><span class="line"><span class="comment">#汇总所有样本</span></span><br><span class="line"><span class="comment">#plotAnnoBar和plotDistToTSS这两个柱状图都支持多个数据同时展示</span></span><br><span class="line">peakAnnoList <span class="operator">&lt;-</span> lapply<span class="punctuation">(</span>files<span class="punctuation">,</span> annotatePeak<span class="punctuation">,</span> </span><br><span class="line">                       TxDb<span class="operator">=</span>txdb<span class="punctuation">,</span></span><br><span class="line">                       tssRegion<span class="operator">=</span><span class="built_in">c</span><span class="punctuation">(</span><span class="operator">-</span><span class="number">3000</span><span class="punctuation">,</span> <span class="number">3000</span><span class="punctuation">)</span><span class="punctuation">)</span></span><br><span class="line">plotAnnoBar<span class="punctuation">(</span>peakAnnoList<span class="punctuation">,</span>title <span class="operator">=</span> <span class="string">"                        Feature Distribution"</span><span class="punctuation">)</span></span><br><span class="line">plotDistToTSS<span class="punctuation">(</span>peakAnnoList<span class="punctuation">,</span>title <span class="operator">=</span> <span class="string">"                 Feature Distribution relative to TSS"</span><span class="punctuation">)</span></span><br><span class="line"><span class="comment">#例举单个样本</span></span><br><span class="line">peakAnno <span class="operator">&lt;-</span> annotatePeak<span class="punctuation">(</span>files<span class="punctuation">[[</span><span class="number">1</span><span class="punctuation">]</span><span class="punctuation">]</span><span class="punctuation">,</span><span class="comment"># 分别改成2或者3或者4即可，分别对应四个文件</span></span><br><span class="line">                        tssRegion<span class="operator">=</span><span class="built_in">c</span><span class="punctuation">(</span><span class="operator">-</span><span class="number">3000</span><span class="punctuation">,</span> <span class="number">3000</span><span class="punctuation">)</span><span class="punctuation">,</span></span><br><span class="line">                        TxDb<span class="operator">=</span>txdb<span class="punctuation">,</span></span><br><span class="line">                        annoDb<span class="operator">=</span><span class="string">"org.Mm.eg.db"</span><span class="punctuation">)</span></span><br><span class="line">plotAnnoPie<span class="punctuation">(</span>peakAnnoLipeakAnnost<span class="punctuation">)</span></span><br><span class="line">upsetplot<span class="punctuation">(</span>peakAnno<span class="punctuation">,</span> vennpie<span class="operator">=</span><span class="literal">TRUE</span><span class="punctuation">)</span></span><br></pre></td></tr></tbody></table></figure>
<p><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/Feature Distribution.jpeg" width="50%"><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/Feature Distribution TSS.jpeg" width="50%"></p>
<p><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/plotAnnoPie.jpeg" width="50%"><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/upsetplot(peakAnno, vennpie=TRUE).jpeg" width="50%"></p>
<h3 id="2、查看peaks在全基因组上的分布"><a href="#2、查看peaks在全基因组上的分布" class="headerlink" title="2、查看peaks在全基因组上的分布"></a>2、查看peaks在全基因组上的分布</h3><figure class="highlight r"><table><tbody><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#输入文件的准备</span></span><br><span class="line">DMSO_24h_wt_rep1<span class="operator">&lt;-</span>read.csv<span class="punctuation">(</span><span class="string">"SRR12692092_summits_filt_blacklist.csv"</span><span class="punctuation">)</span></span><br><span class="line">DMSO_24h_wt_rep1<span class="operator">&lt;-</span>DMSO_24h_wt_rep1<span class="punctuation">[</span><span class="punctuation">,</span><span class="operator">-</span><span class="number">4</span><span class="punctuation">]</span></span><br><span class="line">DMSO_24h_wt_rep2<span class="operator">&lt;-</span>read.csv<span class="punctuation">(</span><span class="string">"SRR12692093_summits_filt_blacklist.csv"</span><span class="punctuation">)</span></span><br><span class="line">DMSO_24h_wt_rep2<span class="operator">&lt;-</span>DMSO_24h_wt_rep2<span class="punctuation">[</span><span class="punctuation">,</span><span class="operator">-</span><span class="number">4</span><span class="punctuation">]</span></span><br><span class="line">BRM014_10uM_24h_wt_rep1<span class="operator">&lt;-</span>read.csv<span class="punctuation">(</span><span class="string">"SRR12692098_summits_filt_blacklist.csv"</span><span class="punctuation">)</span></span><br><span class="line">BRM014_10uM_24h_wt_rep1<span class="operator">&lt;-</span>BRM014_10uM_24h_wt_rep1<span class="punctuation">[</span><span class="punctuation">,</span><span class="operator">-</span><span class="number">4</span><span class="punctuation">]</span></span><br><span class="line">BRM014_10uM_24h_wt_rep2<span class="operator">&lt;-</span>read.csv<span class="punctuation">(</span><span class="string">"SRR12692099_summits_filt_blacklist.csv"</span><span class="punctuation">)</span></span><br><span class="line">BRM014_10uM_24h_wt_rep2<span class="operator">&lt;-</span>BRM014_10uM_24h_wt_rep2<span class="punctuation">[</span><span class="punctuation">,</span><span class="operator">-</span><span class="number">4</span><span class="punctuation">]</span></span><br><span class="line"></span><br><span class="line"><span class="comment">#以DMSO_24h_wt_rep1为例</span></span><br><span class="line">set.seed<span class="punctuation">(</span><span class="number">123</span><span class="punctuation">)</span></span><br><span class="line">circos.initializeWithIdeogram<span class="punctuation">(</span>plotType <span class="operator">=</span> <span class="built_in">c</span><span class="punctuation">(</span><span class="string">"axis"</span><span class="punctuation">,</span> <span class="string">"labels"</span><span class="punctuation">)</span><span class="punctuation">)</span></span><br><span class="line">circos.track<span class="punctuation">(</span>ylim <span class="operator">=</span> <span class="built_in">c</span><span class="punctuation">(</span><span class="number">0</span><span class="punctuation">,</span> <span class="number">1</span><span class="punctuation">)</span><span class="punctuation">,</span> panel.fun <span class="operator">=</span> <span class="keyword">function</span><span class="punctuation">(</span>x<span class="punctuation">,</span> y<span class="punctuation">)</span> <span class="punctuation">{</span></span><br><span class="line">  chr <span class="operator">=</span> CELL_META<span class="operator">$</span>sector.index</span><br><span class="line">  xlim <span class="operator">=</span> CELL_META<span class="operator">$</span>xlim</span><br><span class="line">  ylim <span class="operator">=</span> CELL_META<span class="operator">$</span>ylim</span><br><span class="line">  circos.rect<span class="punctuation">(</span>xlim<span class="punctuation">[</span><span class="number">1</span><span class="punctuation">]</span><span class="punctuation">,</span> <span class="number">0</span><span class="punctuation">,</span> xlim<span class="punctuation">[</span><span class="number">2</span><span class="punctuation">]</span><span class="punctuation">,</span> <span class="number">1</span><span class="punctuation">)</span></span><br><span class="line"><span class="punctuation">}</span><span class="punctuation">,</span> track.height <span class="operator">=</span> <span class="number">0.15</span><span class="punctuation">,</span> bg.border <span class="operator">=</span> <span class="literal">NA</span><span class="punctuation">,</span> bg.col<span class="operator">=</span>rainbow<span class="punctuation">(</span><span class="number">24</span><span class="punctuation">)</span><span class="punctuation">)</span></span><br><span class="line">text<span class="punctuation">(</span><span class="number">0</span><span class="punctuation">,</span> <span class="number">0</span><span class="punctuation">,</span> <span class="string">"DMSO_24h_wt_rep1"</span><span class="punctuation">,</span> cex <span class="operator">=</span> <span class="number">1.5</span><span class="punctuation">)</span></span><br><span class="line">circos.genomicDensity<span class="punctuation">(</span>DMSO_24h_wt_rep1<span class="punctuation">,</span> col <span class="operator">=</span> <span class="built_in">c</span><span class="punctuation">(</span><span class="string">"#000080"</span><span class="punctuation">)</span><span class="punctuation">,</span> track.height <span class="operator">=</span> <span class="number">0.2</span><span class="punctuation">)</span></span><br><span class="line">circos.clear<span class="punctuation">(</span><span class="punctuation">)</span></span><br></pre></td></tr></tbody></table></figure>
<p>看到这样的结果，第一反应就是————为什么两种处理情况下染色体开放程度那么像！？难道我代码有问题！？经过反复检查验证（将一个样本chr1上的peaks都删掉，再次运行上述代码，就会发现显著的改变），可以确定分析上是没有问题的。这两种处理导致的差异可能不是很显著。再加上20万+的peaks放在这个小小的circos图上展示，有些差异会被掩盖掉。就如在做TSS富集分析的时候，单独看TSS前后3Kb区域，可以看到有两个峰，但在看TSS-genebody-TSE区域是，TSS处相对微弱的那个峰就被掩盖掉了。</p>
<p><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/DMSO_24h_wt_rep1.jpeg" width="50%"><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/DMSO_24h_wt_rep2.jpeg" width="50%"><br><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/BRM014_10uM_24h_wt_rep1.jpeg" width="50%"><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/BRM014_10uM_24h_wt_rep2.jpeg" width="50%"></p>
<h3 id="3、拿到每个样本中peaks对应得基因名"><a href="#3、拿到每个样本中peaks对应得基因名" class="headerlink" title="3、拿到每个样本中peaks对应得基因名"></a>3、拿到每个样本中peaks对应得基因名</h3><p>这一步非常重要，拿到基因名就可以根据课题需要进行差异分析等</p>
<figure class="highlight r"><table><tbody><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#以DMSO_24h_wt样本为例</span></span><br><span class="line"><span class="comment">#replicate 1</span></span><br><span class="line">peakAnno_DMSO_24h_wt_rep1 <span class="operator">&lt;-</span> annotatePeak<span class="punctuation">(</span>files<span class="punctuation">[[</span><span class="number">1</span><span class="punctuation">]</span><span class="punctuation">]</span><span class="punctuation">,</span></span><br><span class="line">                                          tssRegion<span class="operator">=</span><span class="built_in">c</span><span class="punctuation">(</span><span class="operator">-</span><span class="number">3000</span><span class="punctuation">,</span> <span class="number">3000</span><span class="punctuation">)</span><span class="punctuation">,</span></span><br><span class="line">                                          TxDb<span class="operator">=</span>txdb<span class="punctuation">,</span></span><br><span class="line">                                          annoDb<span class="operator">=</span><span class="string">"org.Mm.eg.db"</span><span class="punctuation">)</span></span><br><span class="line">genelist_DMSO_24h_wt_rep1_uniqe<span class="operator">&lt;-</span>as.data.frame<span class="punctuation">(</span>unique<span class="punctuation">(</span>peakAnno_DMSO_24h_wt_rep1<span class="operator">@</span>anno<span class="operator">@</span>elementMetadata<span class="operator">@</span>listData<span class="punctuation">[[</span><span class="string">"SYMBOL"</span><span class="punctuation">]</span><span class="punctuation">]</span><span class="punctuation">)</span><span class="punctuation">)</span></span><br><span class="line">colnames<span class="punctuation">(</span>genelist_DMSO_24h_wt_rep1_uniqe<span class="punctuation">)</span><span class="operator">&lt;-</span><span class="string">"symbol"</span></span><br><span class="line"><span class="comment">#replicate 2</span></span><br><span class="line">peakAnno_DMSO_24h_wt_rep2 <span class="operator">&lt;-</span> annotatePeak<span class="punctuation">(</span>files<span class="punctuation">[[</span><span class="number">2</span><span class="punctuation">]</span><span class="punctuation">]</span><span class="punctuation">,</span></span><br><span class="line">                                          tssRegion<span class="operator">=</span><span class="built_in">c</span><span class="punctuation">(</span><span class="operator">-</span><span class="number">3000</span><span class="punctuation">,</span> <span class="number">3000</span><span class="punctuation">)</span><span class="punctuation">,</span></span><br><span class="line">                                          TxDb<span class="operator">=</span>txdb<span class="punctuation">,</span></span><br><span class="line">                                          annoDb<span class="operator">=</span><span class="string">"org.Mm.eg.db"</span><span class="punctuation">)</span></span><br><span class="line">genelist_DMSO_24h_wt_rep2_uniqe<span class="operator">&lt;-</span>as.data.frame<span class="punctuation">(</span>unique<span class="punctuation">(</span>peakAnno_DMSO_24h_wt_rep2<span class="operator">@</span>anno<span class="operator">@</span>elementMetadata<span class="operator">@</span>listData<span class="punctuation">[[</span><span class="string">"SYMBOL"</span><span class="punctuation">]</span><span class="punctuation">]</span><span class="punctuation">)</span><span class="punctuation">)</span></span><br><span class="line">colnames<span class="punctuation">(</span>genelist_DMSO_24h_wt_rep2_uniqe<span class="punctuation">)</span><span class="operator">&lt;-</span><span class="string">"symbol"</span></span><br><span class="line"><span class="comment">#重复样本间共同的开放基因</span></span><br><span class="line">venn.diagram<span class="punctuation">(</span></span><br><span class="line">  x<span class="operator">=</span><span class="built_in">list</span><span class="punctuation">(</span></span><br><span class="line">    DMSO_24h_wt_rep1<span class="operator">=</span>genelist_DMSO_24h_wt_rep1_uniqe<span class="operator">$</span>symbol<span class="punctuation">,</span></span><br><span class="line">    DMSO_24h_wt_rep2<span class="operator">=</span>genelist_DMSO_24h_wt_rep2_uniqe<span class="operator">$</span>symbol</span><br><span class="line">  <span class="punctuation">)</span><span class="punctuation">,</span></span><br><span class="line">  filename <span class="operator">=</span> <span class="string">"DMSO_24h_wt.png"</span><span class="punctuation">,</span></span><br><span class="line">  lty<span class="operator">=</span><span class="string">"dotted"</span><span class="punctuation">,</span></span><br><span class="line">  lwd<span class="operator">=</span><span class="number">3</span><span class="punctuation">,</span></span><br><span class="line">  col<span class="operator">=</span><span class="string">"transparent"</span><span class="punctuation">,</span></span><br><span class="line">  fill<span class="operator">=</span><span class="built_in">c</span><span class="punctuation">(</span><span class="string">"darkorchid1"</span><span class="punctuation">,</span><span class="string">"cornflowerblue"</span><span class="punctuation">)</span><span class="punctuation">,</span></span><br><span class="line">  alpha<span class="operator">=</span><span class="number">0.5</span><span class="punctuation">,</span></span><br><span class="line">  </span><br><span class="line">  label.col<span class="operator">=</span><span class="built_in">c</span><span class="punctuation">(</span><span class="string">"darkorchid1"</span><span class="punctuation">,</span><span class="string">"white"</span><span class="punctuation">,</span><span class="string">"darkblue"</span><span class="punctuation">)</span> <span class="punctuation">,</span></span><br><span class="line">  cex<span class="operator">=</span><span class="number">1</span><span class="punctuation">,</span></span><br><span class="line">  fontfamily<span class="operator">=</span><span class="string">"serif"</span><span class="punctuation">,</span></span><br><span class="line">  fontface<span class="operator">=</span><span class="string">"bold"</span><span class="punctuation">,</span></span><br><span class="line">  cat.default.pos<span class="operator">=</span><span class="string">"text"</span><span class="punctuation">,</span></span><br><span class="line">  cat.col<span class="operator">=</span><span class="built_in">c</span><span class="punctuation">(</span><span class="string">"darkorchid1"</span><span class="punctuation">,</span><span class="string">"darkblue"</span><span class="punctuation">)</span><span class="punctuation">,</span></span><br><span class="line">  cat.cex<span class="operator">=</span><span class="number">0.6</span><span class="punctuation">,</span></span><br><span class="line">  cat.fontfamily<span class="operator">=</span><span class="string">"serif"</span><span class="punctuation">,</span></span><br><span class="line">  cat.dist<span class="operator">=</span><span class="built_in">c</span><span class="punctuation">(</span><span class="number">0.3</span><span class="punctuation">,</span><span class="number">0.3</span><span class="punctuation">)</span><span class="punctuation">,</span></span><br><span class="line">  cat.pos<span class="operator">=</span><span class="number">0</span></span><br><span class="line"><span class="punctuation">)</span></span><br><span class="line"></span><br><span class="line"><span class="comment">#查看各组内样本间的overlapping reads：DMSO_24h_wt， BRM014_10uM_24h_wt；</span></span><br><span class="line"><span class="comment">#以及组间peaks的异同情况：DMSO_24h_wt vs. BRM014_10uM_24h_wt</span></span><br><span class="line"><span class="comment">#代码类似上面的，就不一一展示了</span></span><br></pre></td></tr></tbody></table></figure>
<p>从下图可以看出，不管是组间还是组内，差异的peaks数目都不是很多了，这一点也验证了我们上面做的再全基因组范围内查看peaks的分布结果。</p>
<p><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/DMSO_24h_wt.png" width="50%"><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/BRM014_10uM_24h_wt.png" width="50%"><br><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/DMSO_BRM014.png" width="50%"></p>
</body></html></div><div class="article-licensing box"><div class="licensing-title"><p>2020 ATAC-seq analysis</p><p><a href="http://gyqgyq.com/2020 ATAC-seq analysis/">http://gyqgyq.com/2020 ATAC-seq analysis/</a></p></div><div class="licensing-meta level is-mobile"><div class="level-left"><div class="level-item is-narrow"><div><h6>Author</h6><p>Yuqi Gong</p></div></div><div class="level-item is-narrow"><div><h6>Posted on</h6><p>2020-12-08</p></div></div><div class="level-item is-narrow"><div><h6>Updated on</h6><p>2022-11-26</p></div></div><div class="level-item is-narrow"><div><h6>Licensed under</h6><p><a class="icons" rel="noopener" target="_blank" title="Creative Commons" href="https://creativecommons.org/"><i class="icon fab fa-creative-commons"></i></a><a class="icons" rel="noopener" target="_blank" title="Attribution" href="https://creativecommons.org/licenses/by/4.0/"><i class="icon fab fa-creative-commons-by"></i></a><a class="icons" rel="noopener" target="_blank" title="Noncommercial" href="https://creativecommons.org/licenses/by-nc/4.0/"><i class="icon fab fa-creative-commons-nc"></i></a></p></div></div></div></div></div><div class="article-tags is-size-7 mb-4"><span class="mr-2">#</span><a class="link-muted mr-2" rel="tag" href="/tags/NGS/">NGS</a></div><div class="a2a_kit a2a_kit_size_32 a2a_default_style"><a class="a2a_dd" target="_blank" rel="noopener" href="https://www.addtoany.com/share"></a><a class="a2a_button_facebook"></a><a class="a2a_button_twitter"></a><a class="a2a_button_telegram"></a><a class="a2a_button_whatsapp"></a><a class="a2a_button_reddit"></a></div><script src="https://static.addtoany.com/menu/page.js" defer></script></article></div><div class="card"><div class="card-content"><h3 class="menu-label has-text-centered">Like this article? Support the author with</h3><div class="buttons is-centered"><a class="button donate" data-type="alipay"><span class="icon is-small"><i class="fab fa-alipay"></i></span><span>Alipay</span><span class="qrcode"><img src="/" alt="Alipay"></span></a><a class="button donate" data-type="wechat"><span class="icon is-small"><i class="fab fa-weixin"></i></span><span>Wechat</span><span class="qrcode"><img src="/" alt="Wechat"></span></a></div></div></div><nav class="post-navigation mt-4 level is-mobile"><div class="level-start"><a class="article-nav-prev level level-item link-muted" href="/%E5%8F%AF%E5%8F%98%E5%89%AA%E5%88%87%E4%B9%8BMISO/"><i class="level-item fas fa-chevron-left"></i><span class="level-item">可变剪切分析之MISO</span></a></div><div class="level-end"><a class="article-nav-next level level-item link-muted" href="/NGS%E6%95%B0%E6%8D%AE%E5%88%86%E6%9E%90%E4%B9%8B%E8%A1%A8%E8%A7%82/"><span class="level-item">NGS数据分析之表观</span><i class="level-item fas fa-chevron-right"></i></a></div></nav><!--!--></div><div class="column column-left is-4-tablet is-4-desktop is-4-widescreen  order-1"><div class="card widget" data-type="profile"><div class="card-content"><nav class="level"><div class="level-item has-text-centered flex-shrink-1"><div><figure class="image is-128x128 mx-auto mb-2"><img class="avatar" src="/img/ava.jpg" alt="Yuqi Gong"></figure><p class="title is-size-4 is-block" style="line-height:inherit;">Yuqi Gong</p><p class="is-size-6 is-flex justify-content-center"><i class="fas fa-map-marker-alt mr-1"></i><span>HangZhou, China</span></p></div></div></nav><nav class="level is-mobile"><div class="level-item has-text-centered is-marginless"><div><p class="heading">Posts</p><a href="/archives"><p class="title">37</p></a></div></div><div class="level-item has-text-centered is-marginless"><div><p class="heading">Category</p><a href="/categories"><p class="title">1</p></a></div></div><div class="level-item has-text-centered is-marginless"><div><p class="heading">Tags</p><a href="/tags"><p class="title">7</p></a></div></div></nav><div class="level"><a class="level-item button is-primary is-rounded" href="https://github.com/gongyuqiqi" target="_blank" rel="noopener">Follow</a></div><div class="level is-mobile is-multiline"><a class="level-item button is-transparent is-marginless" target="_blank" rel="noopener" title="Github" href="https://github.com/gongyuqiqi"><i class="fab fa-github"></i></a><a class="level-item button is-transparent is-marginless" target="_blank" rel="noopener" title="Facebook" href="https://facebook.com"><i class="fab fa-facebook"></i></a><a class="level-item button is-transparent is-marginless" target="_blank" rel="noopener" title="Twitter" href="https://twitter.com"><i class="fab fa-twitter"></i></a><a class="level-item button is-transparent is-marginless" target="_blank" rel="noopener" title="Dribbble" href="https://dribbble.com"><i class="fab fa-dribbble"></i></a><a class="level-item button is-transparent is-marginless" target="_blank" rel="noopener" title="RSS" href="/"><i class="fas fa-rss"></i></a></div></div></div><div class="card widget" id="toc" data-type="toc"><div class="card-content"><div class="menu"><h3 class="menu-label">Catalogue</h3><ul class="menu-list"><li><a class="level is-mobile" href="#2020-ATAC-seq-analysis"><span class="level-left"><span class="level-item">1</span><span class="level-item">2020 ATAC-seq analysis</span></span></a><ul class="menu-list"><li><a class="level is-mobile" href="#一、ATAC-seq个性化的conda-environment的搭建"><span class="level-left"><span class="level-item">1.1</span><span class="level-item">一、ATAC-seq个性化的conda environment的搭建</span></span></a><ul class="menu-list"><li><a class="level is-mobile" href="#1、git-clone"><span class="level-left"><span class="level-item">1.1.1</span><span class="level-item">1、git clone</span></span></a></li><li><a class="level is-mobile" href="#2、环境搭建"><span class="level-left"><span class="level-item">1.1.2</span><span class="level-item">2、环境搭建</span></span></a></li></ul></li><li><a class="level is-mobile" href="#二、数据下载"><span class="level-left"><span class="level-item">1.2</span><span class="level-item">二、数据下载</span></span></a></li><li><a class="level is-mobile" href="#三、bowtie2比对"><span class="level-left"><span class="level-item">1.3</span><span class="level-item">三、bowtie2比对</span></span></a><ul class="menu-list"><li><a class="level is-mobile" href="#1、-fastqc-multiqc查看原始数据的质量情况"><span class="level-left"><span class="level-item">1.3.1</span><span class="level-item">1、 fastqc+multiqc查看原始数据的质量情况</span></span></a></li><li><a class="level is-mobile" href="#2、-过滤低质量的reads"><span class="level-left"><span class="level-item">1.3.2</span><span class="level-item">2、 过滤低质量的reads</span></span></a></li><li><a class="level is-mobile" href="#3、-比对参考基因组"><span class="level-left"><span class="level-item">1.3.3</span><span class="level-item">3、 比对参考基因组</span></span></a></li></ul></li><li><a class="level is-mobile" href="#四、PCR去除重及质控制"><span class="level-left"><span class="level-item">1.4</span><span class="level-item">四、PCR去除重及质控制</span></span></a><ul class="menu-list"><li><a class="level is-mobile" href="#1、初步过滤"><span class="level-left"><span class="level-item">1.4.1</span><span class="level-item">1、初步过滤</span></span></a></li><li><a class="level is-mobile" href="#2、进一步"><span class="level-left"><span class="level-item">1.4.2</span><span class="level-item">2、进一步</span></span></a></li><li><a class="level is-mobile" href="#3、标记PCR重复"><span class="level-left"><span class="level-item">1.4.3</span><span class="level-item">3、标记PCR重复</span></span></a></li><li><a class="level is-mobile" href="#4、去除PCR重复、建立索引文件、按name排序的bam文件、质控文件"><span class="level-left"><span class="level-item">1.4.4</span><span class="level-item">4、去除PCR重复、建立索引文件、按name排序的bam文件、质控文件</span></span></a></li><li><a class="level is-mobile" href="#5、测序文库复杂度的检验"><span class="level-left"><span class="level-item">1.4.5</span><span class="level-item">5、测序文库复杂度的检验</span></span></a></li><li><a class="level is-mobile" href="#6、插入片段质控检测"><span class="level-left"><span class="level-item">1.4.6</span><span class="level-item">6、插入片段质控检测</span></span></a></li><li><a class="level is-mobile" href="#7、TSS富集"><span class="level-left"><span class="level-item">1.4.7</span><span class="level-item">7、TSS富集</span></span></a></li></ul></li><li><a class="level is-mobile" href="#五、生成tagAlign格式文件"><span class="level-left"><span class="level-item">1.5</span><span class="level-item">五、生成tagAlign格式文件</span></span></a><ul class="menu-list"><li><a class="level is-mobile" href="#1-Convert-PE-BAM-to-tagAlign"><span class="level-left"><span class="level-item">1.5.1</span><span class="level-item">1. Convert PE BAM to tagAlign</span></span></a></li><li><a class="level is-mobile" href="#2-Stand-Cross-Correlation-analysis"><span class="level-left"><span class="level-item">1.5.2</span><span class="level-item">2. Stand Cross Correlation analysis</span></span></a></li></ul></li><li><a class="level is-mobile" href="#六、Call-Peaks"><span class="level-left"><span class="level-item">1.6</span><span class="level-item">六、Call Peaks</span></span></a><ul class="menu-list"><li><a class="level is-mobile" href="#1、去除线粒体基因的TagAlign格式文件进行shift操作，输入macs2软件去callpeak"><span class="level-left"><span class="level-item">1.6.1</span><span class="level-item">1、去除线粒体基因的TagAlign格式文件进行shift操作，输入macs2软件去callpeak</span></span></a></li><li><a class="level is-mobile" href="#2、去除ENCODE列入黑名单的区域"><span class="level-left"><span class="level-item">1.6.2</span><span class="level-item">2、去除ENCODE列入黑名单的区域</span></span></a></li><li><a class="level is-mobile" href="#3、Irreproducibility-Discovery-Rate-IDR-评估"><span class="level-left"><span class="level-item">1.6.3</span><span class="level-item">3、Irreproducibility Discovery Rate (IDR)评估</span></span></a></li><li><a class="level is-mobile" href="#4、Fraction-of-reads-in-peaks-FRiP-评估"><span class="level-left"><span class="level-item">1.6.4</span><span class="level-item">4、Fraction of reads in peaks (FRiP)评估</span></span></a></li></ul></li><li><a class="level is-mobile" href="#七、Peak-annotation"><span class="level-left"><span class="level-item">1.7</span><span class="level-item">七、Peak annotation</span></span></a><ul class="menu-list"><li><a class="level is-mobile" href="#1、Feature-Distribution"><span class="level-left"><span class="level-item">1.7.1</span><span class="level-item">1、Feature Distribution</span></span></a></li><li><a class="level is-mobile" href="#2、查看peaks在全基因组上的分布"><span class="level-left"><span class="level-item">1.7.2</span><span class="level-item">2、查看peaks在全基因组上的分布</span></span></a></li><li><a class="level is-mobile" href="#3、拿到每个样本中peaks对应得基因名"><span class="level-left"><span class="level-item">1.7.3</span><span class="level-item">3、拿到每个样本中peaks对应得基因名</span></span></a></li></ul></li></ul></li></ul></div></div><style>#toc .menu-list > li > a.is-active + .menu-list { display: block; }#toc .menu-list > li > a + .menu-list { display: none; }</style><script src="/js/toc.js" defer></script></div><div class="card widget" data-type="categories"><div class="card-content"><div class="menu"><h3 class="menu-label">Categories</h3><ul class="menu-list"><li><a class="level is-mobile" href="/categories/Programming-Language/"><span class="level-start"><span class="level-item">Programming Language</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></div></div></div><div class="card widget" data-type="recent-posts"><div class="card-content"><h3 class="menu-label">Recents</h3><article class="media"><div class="media-content"><p class="date"><time dateTime="2021-07-04T05:48:30.000Z">2021-07-04</time></p><p class="title"><a href="/RNA-TNBC/">RNA-TNBC</a></p></div></article><article class="media"><div class="media-content"><p class="date"><time dateTime="2021-06-29T06:01:05.000Z">2021-06-29</time></p><p class="title"><a href="/scRNA-TNBC/">scRNA-TNBC</a></p></div></article><article class="media"><div class="media-content"><p class="date"><time dateTime="2021-06-03T04:43:57.000Z">2021-06-03</time></p><p class="title"><a href="/MeRIP-Seq/">MeRIP-Seq</a></p></div></article><article class="media"><div class="media-content"><p class="date"><time dateTime="2021-05-14T05:37:00.000Z">2021-05-14</time></p><p class="title"><a href="/WES/">WES</a></p></div></article><article class="media"><div class="media-content"><p class="date"><time dateTime="2021-05-13T06:08:18.000Z">2021-05-13</time></p><p class="title"><a href="/scRNA-seq-%E6%95%B0%E6%8D%AE%E5%A4%8D%E7%8E%B0/">scRNA-seq_数据复现</a></p></div></article></div></div><div class="card widget" data-type="archives"><div class="card-content"><div class="menu"><h3 class="menu-label">Archives</h3><ul class="menu-list"><li><a class="level is-mobile" href="/archives/2021/07/"><span class="level-start"><span class="level-item">July 2021</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/archives/2021/06/"><span class="level-start"><span class="level-item">June 2021</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/archives/2021/05/"><span class="level-start"><span class="level-item">May 2021</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li><li><a class="level is-mobile" href="/archives/2021/03/"><span class="level-start"><span class="level-item">March 2021</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/archives/2021/02/"><span class="level-start"><span class="level-item">February 2021</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/archives/2021/01/"><span class="level-start"><span class="level-item">January 2021</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li><li><a class="level is-mobile" href="/archives/2020/12/"><span class="level-start"><span class="level-item">December 2020</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/archives/2020/05/"><span class="level-start"><span class="level-item">May 2020</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/archives/2020/04/"><span class="level-start"><span class="level-item">April 2020</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/archives/2020/03/"><span class="level-start"><span class="level-item">March 2020</span></span><span class="level-end"><span class="level-item tag">5</span></span></a></li><li><a class="level is-mobile" href="/archives/2020/02/"><span class="level-start"><span class="level-item">February 2020</span></span><span class="level-end"><span class="level-item tag">18</span></span></a></li></ul></div></div></div><div class="card widget" data-type="tags"><div class="card-content"><div class="menu"><h3 class="menu-label">Tags</h3><div class="field is-grouped is-grouped-multiline"><div class="control"><a class="tags has-addons" href="/tags/Linux/"><span class="tag">Linux</span><span class="tag">7</span></a></div><div class="control"><a class="tags has-addons" href="/tags/MySQL/"><span class="tag">MySQL</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/NGS/"><span class="tag">NGS</span><span class="tag">18</span></a></div><div class="control"><a class="tags has-addons" href="/tags/R/"><span class="tag">R</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/System/"><span class="tag">System</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/notes/"><span class="tag">notes</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/tag1/"><span class="tag">tag1</span><span class="tag">6</span></a></div></div></div></div></div><div class="card widget" data-type="subscribe-email"><div class="card-content"><div class="menu"><h3 class="menu-label">follow.it</h3><form action="" method="post" target="_blank"><div class="field has-addons"><div class="control has-icons-left is-expanded"><input class="input" name="email" type="email" placeholder="Email"><span class="icon is-small is-left"><i class="fas fa-envelope"></i></span></div><div class="control"><input class="button" type="submit" value="Subscribe"></div></div></form></div></div></div></div><!--!--></div></div></section><footer class="footer"><div class="container"><div class="level"><div class="level-start"><a class="footer-logo is-block mb-2" href="/"><img src="/img/ava.jpg" alt="Yuqi Gong&#039;s blog" height="28"></a><p class="is-size-7"><span>&copy; 2024 Yuqi Gong</span>  Powered by Gong Yuqi <i class="fa-heart fa fa-beat" style="color: #f00;"></i><br><span id="busuanzi_container_site_uv">Visited by <span id="busuanzi_value_site_uv">0</span> users</span></p></div><div class="level-end"><div class="field has-addons"><p class="control"><a class="button is-transparent is-large" target="_blank" rel="noopener" title="Creative Commons" href="https://creativecommons.org/"><i class="fab fa-creative-commons"></i></a></p><p class="control"><a class="button is-transparent is-large" target="_blank" rel="noopener" title="Attribution 4.0 International" href="https://creativecommons.org/licenses/by/4.0/"><i class="fab fa-creative-commons-by"></i></a></p><p class="control"><a class="button is-transparent is-large" target="_blank" rel="noopener" title="Download on GitHub" href="https://github.com/gongyuqiqi/gongyuqiqi.github.io"><i class="fab fa-github"></i></a></p></div></div></div></div></footer><script src="https://cdn.jsdelivr.net/npm/jquery@3.3.1/dist/jquery.min.js"></script><script src="https://cdn.jsdelivr.net/npm/moment@2.22.2/min/moment-with-locales.min.js"></script><script src="https://cdn.jsdelivr.net/npm/clipboard@2.0.4/dist/clipboard.min.js" defer></script><script>moment.locale("en");</script><script>var IcarusThemeSettings = {
            article: {
                highlight: {
                    clipboard: true,
                    fold: 'unfolded'
                }
            }
        };</script><script src="/js/column.js"></script><script src="/js/animation.js"></script><a id="back-to-top" title="Back to top" href="javascript:;"><i class="fas fa-chevron-up"></i></a><script src="/js/back_to_top.js" defer></script><!--!--><!--!--><!--!--><!--!--><script src="https://cdn.jsdelivr.net/npm/cookieconsent@3.1.1/build/cookieconsent.min.js" defer></script><script>window.addEventListener("load", () => {
      window.cookieconsent.initialise({
        type: "info",
        theme: "edgeless",
        static: false,
        position: "bottom-left",
        content: {
          message: "This website uses cookies to improve your experience.",
          dismiss: "Got it!",
          allow: "Allow cookies",
          deny: "Decline",
          link: "Learn more",
          policy: "Cookie Policy",
          href: "https://www.cookiesandyou.com/",
        },
        palette: {
          popup: {
            background: "#edeff5",
            text: "#838391"
          },
          button: {
            background: "#4b81e8"
          },
        },
      });
    });</script><script src="https://cdn.jsdelivr.net/npm/lightgallery@1.10.0/dist/js/lightgallery.min.js" defer></script><script src="https://cdn.jsdelivr.net/npm/justifiedGallery@3.8.1/dist/js/jquery.justifiedGallery.min.js" defer></script><script>window.addEventListener("load", () => {
            if (typeof $.fn.lightGallery === 'function') {
                $('.article').lightGallery({ selector: '.gallery-item' });
            }
            if (typeof $.fn.justifiedGallery === 'function') {
                if ($('.justified-gallery > p > .gallery-item').length) {
                    $('.justified-gallery > p > .gallery-item').unwrap();
                }
                $('.justified-gallery').justifiedGallery();
            }
        });</script><!--!--><!--!--><!--!--><!--!--><!--!--><script src="/js/main.js" defer></script><div class="searchbox"><div class="searchbox-container"><div class="searchbox-header"><div class="searchbox-input-container"><input class="searchbox-input" type="text" placeholder="Type something..."></div><a class="searchbox-close" href="javascript:;">×</a></div><div class="searchbox-body"></div></div></div><script src="/js/insight.js" defer></script><script>document.addEventListener('DOMContentLoaded', function () {
            loadInsight({"contentUrl":"/content.json"}, {"hint":"Type something...","untitled":"(Untitled)","posts":"Posts","pages":"Pages","categories":"Categories","tags":"Tags"});
        });</script></body></html>