<!doctype html>
<html lang="en"><head><meta charset="utf-8"><meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1"><meta><title>WES - Yuqi Gong&#039;s blog</title><link rel="manifest" href="/manifest.json"><meta name="application-name" content="Yuqi Gong&#039;s blog"><meta name="msapplication-TileImage" content="/img/ava.jpg"><meta name="apple-mobile-web-app-capable" content="yes"><meta name="apple-mobile-web-app-title" content="Yuqi Gong&#039;s blog"><meta name="apple-mobile-web-app-status-bar-style" content="default"><meta name="description" content="whole exome-sequencing analysis pipeline"><meta property="og:type" content="blog"><meta property="og:title" content="WES"><meta property="og:url" content="http://gyqgyq.com/WES/"><meta property="og:site_name" content="Yuqi Gong&#039;s blog"><meta property="og:description" content="whole exome-sequencing analysis pipeline"><meta property="og:locale" content="en_US"><meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/java_version.JPG"><meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/paper_methods.JPG"><meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/fastqc-status-check-heatmap.png"><meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/bwa_flagstat.JPG"><meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/Mutect2.JPG"><meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/GRCh38.99.JPG"><meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/GRCh38.99.content.JPG"><meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/vcf.JPG"><meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/wc_l.JPG"><meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/SnpSift_second.JPG"><meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/sed.JPG"><meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/G_to_A.JPG"><meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/CDK12_G_to_A.JPG"><meta property="article:published_time" content="2021-05-14T05:37:00.000Z"><meta property="article:modified_time" content="2022-11-26T06:08:08.861Z"><meta property="article:author" content="Yuqi Gong"><meta property="article:tag" content="tag1"><meta property="twitter:card" content="summary"><meta property="twitter:image:src" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/java_version.JPG"><script type="application/ld+json">{"@context":"https://schema.org","@type":"BlogPosting","mainEntityOfPage":{"@type":"WebPage","@id":"http://gyqgyq.com/WES/"},"headline":"WES","image":["https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/fastqc-status-check-heatmap.png"],"datePublished":"2021-05-14T05:37:00.000Z","dateModified":"2022-11-26T06:08:08.861Z","author":{"@type":"Person","name":"Yuqi Gong"},"publisher":{"@type":"Organization","name":"Yuqi Gong's blog","logo":{"@type":"ImageObject","url":"http://gyqgyq.com/img/ava.jpg"}},"description":"whole exome-sequencing analysis pipeline"}</script><link rel="canonical" href="http://gyqgyq.com/WES/"><link rel="icon" href="/img/ava.jpg"><link rel="stylesheet" href="https://use.fontawesome.com/releases/v6.0.0/css/all.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/highlight.js@9.12.0/styles/atom-one-light.css"><link rel="stylesheet" href="https://fonts.googleapis.com/css2?family=Ubuntu:wght@400;600&amp;family=Source+Code+Pro"><link rel="stylesheet" href="/css/default.css"><style>body>.footer,body>.navbar,body>.section{opacity:0}</style><!--!--><!--!--><!--!--><script src="//busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js" defer></script><!--!--><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/cookieconsent@3.1.1/build/cookieconsent.min.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/lightgallery@1.10.0/dist/css/lightgallery.min.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/justifiedGallery@3.8.1/dist/css/justifiedGallery.min.css"><!--!--><!--!--><style>.pace{-webkit-pointer-events:none;pointer-events:none;-webkit-user-select:none;-moz-user-select:none;user-select:none}.pace-inactive{display:none}.pace .pace-progress{background:#3273dc;position:fixed;z-index:2000;top:0;right:100%;width:100%;height:2px}</style><script src="https://cdn.jsdelivr.net/npm/pace-js@1.2.4/pace.min.js"></script><!--!--><!--!--><!-- hexo injector head_end start --><script>
  (function () {
      function switchTab() {
          if (!location.hash) {
            return;
          }

          const $tabMenu = document.querySelector(`a[href="${location.hash}"]`);
          if (!$tabMenu) {
            return;
          }

          const $tabMenuContainer = $tabMenu.parentElement.parentElement;
          Array.from($tabMenuContainer.children).forEach($menu => $menu.classList.remove('is-active'));
          Array.from($tabMenuContainer.querySelectorAll('a'))
              .map($menu => document.getElementById($menu.getAttribute("href").substring(1)))
              .forEach($content => $content.classList.add('is-hidden'));

          if ($tabMenu) {
              $tabMenu.parentElement.classList.add('is-active');
          }
          const $activeTab = document.querySelector(location.hash);
          if ($activeTab) {
              $activeTab.classList.remove('is-hidden');
          }
      }
      switchTab();
      window.addEventListener('hashchange', switchTab, false);
  })();
  </script><!-- hexo injector head_end end --><meta name="generator" content="Hexo 6.3.0"></head><body class="is-2-column"><nav class="navbar navbar-main"><div class="container navbar-container"><div class="navbar-brand justify-content-center"><a class="navbar-item navbar-logo" href="/"><img src="/img/ava.jpg" alt="Yuqi Gong&#039;s blog" height="28"></a></div><div class="navbar-menu"><div class="navbar-start"><a class="navbar-item" href="/">Home</a><a class="navbar-item" href="/archives">Archives</a><a class="navbar-item" href="/categories">Categories</a><a class="navbar-item" href="/tags">Tags</a><a class="navbar-item" href="/about">About</a></div><div class="navbar-end"><a class="navbar-item" target="_blank" rel="noopener" title="Download on GitHub" href="https://github.com/gongyuqiqi/gongyuqiqi.github.io"><i class="fab fa-github"></i></a><a class="navbar-item is-hidden-tablet catalogue" title="Catalogue" href="javascript:;"><i class="fas fa-list-ul"></i></a><a class="navbar-item search" title="Search" href="javascript:;"><i class="fas fa-search"></i></a></div></div></div></nav><section class="section"><div class="container"><div class="columns"><div class="column order-2 column-main is-8-tablet is-8-desktop is-8-widescreen"><div class="card"><article class="card-content article" role="article"><div class="article-meta is-size-7 is-uppercase level is-mobile"><div class="level-left"><span class="level-item">Posted&nbsp;<time dateTime="2021-05-14T05:37:00.000Z" title="5/14/2021, 1:37:00 PM">2021-05-14</time></span><span class="level-item">Updated&nbsp;<time dateTime="2022-11-26T06:08:08.861Z" title="11/26/2022, 2:08:08 PM">2022-11-26</time></span><span class="level-item">19 minutes read (About 2894 words)</span><span class="level-item" id="busuanzi_container_page_pv"><span id="busuanzi_value_page_pv">0</span>&nbsp;visits</span></div></div><h1 class="title is-3 is-size-4-mobile">WES</h1><div class="content"><p>whole exome-sequencing analysis pipeline</p>
<span id="more"></span>
<h1 id="全外显子数据分析"><a href="#全外显子数据分析" class="headerlink" title="全外显子数据分析"></a>全外显子数据分析</h1><h2 id="一-环境搭建"><a href="#一-环境搭建" class="headerlink" title="(一) 环境搭建"></a>(一) 环境搭建</h2><p><strong>1、GATK依赖Java 8&#x2F;JDK 1.8 (Oracle or OpenJDK)</strong></p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#查看一下环境中是否有java，如果有，版本是否符合要求</span></span><br><span class="line">java --version</span><br></pre></td></tr></table></figure>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/java_version.JPG"/>

<p><strong>2、下载安装GATK4</strong></p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">wget https://github.com/broadinstitute/gatk/releases/download/4.2.0.0/gatk-4.2.0.0.zip</span><br><span class="line">unzip gatk-4.2.0.0.zip</span><br><span class="line"><span class="comment">#将当前路径加到环境变量中</span></span><br><span class="line"><span class="built_in">echo</span> <span class="string">&#x27;export PATH=/home/gongyuqi/biosoft/GATK/gatk-4.2.0.0:$PATH&#x27;</span> &gt;&gt; ~/.bashrc</span><br><span class="line"><span class="built_in">source</span> ~/.bashrc</span><br></pre></td></tr></table></figure>

<p><strong>3、下载其他需要的软件</strong></p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#首先创建WES的conda环境</span></span><br><span class="line">conda create -n WES</span><br><span class="line"><span class="comment">#其次下载其他需要的软件</span></span><br><span class="line">conda install -y python=3.6.2</span><br><span class="line">conda install -y bwa sra-tools samtools bcftools snpEFF multiqc </span><br><span class="line">qualimap  </span><br><span class="line"><span class="comment">#激活环境，准备开始实战演练</span></span><br><span class="line">conda activate WES</span><br><span class="line"><span class="comment">#创建存放各阶段数据的文件夹</span></span><br><span class="line"><span class="built_in">cd</span> /home/gongyuqi/project/WES</span><br><span class="line"><span class="built_in">mkdir</span> raw qc clean mutaion</span><br><span class="line"><span class="built_in">cd</span> qc &amp;&amp; <span class="built_in">mkdir</span> raw_qc clean_qc</span><br></pre></td></tr></table></figure>

<h2 id="二-WES测试数据下载"><a href="#二-WES测试数据下载" class="headerlink" title="(二) WES测试数据下载"></a>(二) WES测试数据下载</h2><p><strong>1、数据来源<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE153707">GSE153707</a>，我们这里从<a target="_blank" rel="noopener" href="https://www.ebi.ac.uk/ena/browser/view/PRJNA643782">EBI直接下载fastq文件</a></strong></p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#将下列数据下载的脚本保存到download.sh文件中</span></span><br><span class="line"><span class="built_in">dir</span>=/home/gongyuqi/.aspera/connect/etc/asperaweb_id_dsa.openssh</span><br><span class="line">x=_1</span><br><span class="line">y=_2</span><br><span class="line"><span class="keyword">for</span> <span class="built_in">id</span> <span class="keyword">in</span> &#123;11,12&#125;</span><br><span class="line"><span class="keyword">do</span></span><br><span class="line">ascp -QT -l 300m -P33001 -i <span class="variable">$dir</span> era-fasp@fasp.sra.ebi.ac.uk:/vol1/fastq/SRR121/0<span class="variable">$id</span>/SRR121359<span class="variable">$id</span>/SRR121359$<span class="built_in">id</span><span class="variable">$x</span>.fastq.gz .</span><br><span class="line">ascp -QT -l 300m -P33001 -i <span class="variable">$dir</span> era-fasp@fasp.sra.ebi.ac.uk:/vol1/fastq/SRR121/0<span class="variable">$id</span>/SRR121359<span class="variable">$id</span>/SRR121359$<span class="built_in">id</span><span class="variable">$y</span>.fastq.gz . </span><br><span class="line"><span class="keyword">done</span></span><br><span class="line"><span class="comment">#赋予执行权限</span></span><br><span class="line"><span class="built_in">chmod</span> +x download.sh</span><br><span class="line"><span class="comment">#放后台进行数据下载</span></span><br><span class="line"><span class="built_in">nohup</span> ./download.sh &amp;</span><br></pre></td></tr></table></figure>

<p><strong>2、了解文章WES数据处理的相关步骤和参数</strong></p>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/paper_methods.JPG" width="70%">


<h2 id="三-GATK准备bam文件用于找变异"><a href="#三-GATK准备bam文件用于找变异" class="headerlink" title="(三) GATK准备bam文件用于找变异"></a>(三) GATK准备bam文件用于找变异</h2><h3 id="1、-比对GATK官网提供的参考基因组"><a href="#1、-比对GATK官网提供的参考基因组" class="headerlink" title="1、 比对GATK官网提供的参考基因组"></a>1、 比对GATK官网提供的参考基因组</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#====step 1 首先质空，查看数据是否需要进行过滤</span></span><br><span class="line"><span class="comment">#我们这里的质控结果显示reads的整体质量还不错（GC含量看着不太过分基本不影响后续的分析），可以直接用于比对</span></span><br><span class="line"><span class="built_in">ls</span> ../raw/*.gz|xargs fastqc -t 4 -o ./</span><br><span class="line">multiqc *.zip</span><br></pre></td></tr></table></figure>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/fastqc-status-check-heatmap.png"/>


<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#====step 2 比对</span></span><br><span class="line"><span class="built_in">ls</span> *1.fastq.gz &gt;fq1.txt</span><br><span class="line"><span class="built_in">ls</span> *2.fastq.gz &gt;fq2.txt</span><br><span class="line"><span class="built_in">paste</span> fq1.txt fq2.txt &gt; sample.txt</span><br><span class="line">index=/home/gongyuqi/ref/GATK/hg38/v0/Homo_sapiens_assembly38.fasta.64</span><br><span class="line"><span class="built_in">cat</span> sample.txt | <span class="keyword">while</span> <span class="built_in">read</span> <span class="built_in">id</span></span><br><span class="line"><span class="keyword">do</span> <span class="built_in">nohup</span></span><br><span class="line">arr=(<span class="variable">$&#123;id&#125;</span>)</span><br><span class="line">fq1=<span class="variable">$&#123;arr[0]&#125;</span></span><br><span class="line">fq2=<span class="variable">$&#123;arr[1]&#125;</span></span><br><span class="line">sample=<span class="variable">$&#123;fq1%%_*&#125;</span></span><br><span class="line">bwa mem -t 8 -R <span class="string">&quot;@RG\tID:<span class="variable">$sample</span>\tSM:<span class="variable">$sample</span>\tLB:WGS\tPL:Illumina&quot;</span> /<span class="variable">$index</span> <span class="variable">$fq1</span> <span class="variable">$fq2</span> |samtools <span class="built_in">sort</span> -@ 8 -o ../align/<span class="variable">$sample</span>.sort.bam - &amp;&gt;&gt;../align/<span class="variable">$sample</span>.<span class="built_in">log</span> &amp;</span><br><span class="line"><span class="keyword">done</span></span><br></pre></td></tr></table></figure>

<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># ====step 3 查看比对结果</span></span><br><span class="line"><span class="built_in">ls</span> *.bam|<span class="keyword">while</span> <span class="built_in">read</span> <span class="built_in">id</span>;<span class="keyword">do</span> samtools flagstat <span class="variable">$id</span>;<span class="keyword">done</span></span><br></pre></td></tr></table></figure>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/bwa_flagstat.JPG" width="50%">

<h3 id="2、标记PCR重复"><a href="#2、标记PCR重复" class="headerlink" title="2、标记PCR重复"></a>2、标记PCR重复</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># ====step 1 MarkDuplicates</span></span><br><span class="line"><span class="built_in">ls</span> *.bam|<span class="keyword">while</span> <span class="built_in">read</span> sample</span><br><span class="line"><span class="keyword">do</span> <span class="built_in">nohup</span> gatk --java-options <span class="string">&quot;-Xmx20G -Djava.io.tmpdir=./&quot;</span> \</span><br><span class="line">   MarkDuplicates \</span><br><span class="line">   --INPUT <span class="variable">$sample</span> \</span><br><span class="line">   --OUTPUT <span class="variable">$&#123;sample%%.*&#125;</span>_marked.bam \</span><br><span class="line">   -M <span class="variable">$&#123;sample%%.*&#125;</span>.metrics &amp;</span><br><span class="line"><span class="keyword">done</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># ====step 2 FixMateInformation</span></span><br><span class="line"><span class="built_in">ls</span> *_marked.bam|<span class="keyword">while</span> <span class="built_in">read</span> sample</span><br><span class="line"><span class="keyword">do</span> <span class="built_in">nohup</span> gatk --java-options <span class="string">&quot;-Xmx20G -Djava.io.tmpdir=./&quot;</span> \</span><br><span class="line">   FixMateInformation \</span><br><span class="line">   -I <span class="variable">$&#123;sample&#125;</span> \</span><br><span class="line">   -O <span class="variable">$&#123;sample%%_*&#125;</span>_marked_fixed.bam \</span><br><span class="line">   -SO coordinate &amp;</span><br><span class="line"><span class="keyword">done</span></span><br><span class="line"><span class="comment">#建立*_fixed.bam文件索引</span></span><br><span class="line"><span class="built_in">ls</span> *_fixed.bam|<span class="keyword">while</span> <span class="built_in">read</span> sample</span><br><span class="line"><span class="keyword">do</span> <span class="built_in">nohup</span> samtools index <span class="variable">$sample</span> &amp;</span><br><span class="line"><span class="keyword">done</span></span><br></pre></td></tr></table></figure>

<h3 id="3、碱基矫正-BaseRecalibrator"><a href="#3、碱基矫正-BaseRecalibrator" class="headerlink" title="3、碱基矫正 BaseRecalibrator"></a>3、碱基矫正 BaseRecalibrator</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br></pre></td><td class="code"><pre><span class="line">ref=/home/gongyuqi/ref/GATK/hg38/v0/Homo_sapiens_assembly38.fasta</span><br><span class="line">snp=/home/gongyuqi/ref/GATK/hg38/v0/Homo_sapiens_assembly38.dbsnp138.vcf</span><br><span class="line">indel=/home/gongyuqi/ref/GATK/hg38/v0/Mills_and_1000G_gold_standard.indels.hg38.vcf.gz</span><br><span class="line"></span><br><span class="line"><span class="built_in">ls</span> *_marked_fixed.bam|<span class="keyword">while</span> <span class="built_in">read</span> sample</span><br><span class="line"><span class="keyword">do</span> <span class="built_in">nohup</span> gatk \</span><br><span class="line">         --java-options <span class="string">&quot;-Xmx20G -Djava.io.tmpdir=./&quot;</span> \</span><br><span class="line">         BaseRecalibrator \</span><br><span class="line">         -R <span class="variable">$ref</span> \</span><br><span class="line">         -I <span class="variable">$sample</span> \</span><br><span class="line">         --known-sites <span class="variable">$snp</span> \</span><br><span class="line">         --known-sites <span class="variable">$indel</span> \</span><br><span class="line">         -O <span class="variable">$&#123;sample%%_*&#125;</span>_recal.table \</span><br><span class="line">         1&gt;<span class="variable">$&#123;sample%%_*&#125;</span>_log.recal 2&gt;&amp;1 &amp;</span><br><span class="line"><span class="keyword">done</span></span><br><span class="line"></span><br><span class="line"><span class="comment">#BWA比对时设置@RG的重要性在这一步体现出来了，算法通过@RG中的ID来识别各个独立的测序过程</span></span><br><span class="line"><span class="built_in">ls</span> *_marked_fixed.bam|<span class="keyword">while</span> <span class="built_in">read</span> sample</span><br><span class="line"><span class="keyword">do</span> <span class="built_in">nohup</span> gatk \</span><br><span class="line">         --java-options <span class="string">&quot;-Xmx20G -Djava.io.tmpdir=./&quot;</span>   ApplyBQSR \</span><br><span class="line">         -R <span class="variable">$ref</span>  \</span><br><span class="line">         -I <span class="variable">$sample</span>  \</span><br><span class="line">         -bqsr <span class="variable">$&#123;sample%%_*&#125;</span>_recal.table \</span><br><span class="line">         -O <span class="variable">$&#123;sample%%_*&#125;</span>_bqsr.bam \</span><br><span class="line">         1&gt;<span class="variable">$&#123;sample%%_*&#125;</span>_log.ApplyBQSR  2&gt;&amp;1 &amp;</span><br><span class="line"><span class="keyword">done</span></span><br><span class="line"><span class="comment"># 查看比对结果</span></span><br><span class="line"><span class="built_in">ls</span> *bqsr.bam|<span class="keyword">while</span> <span class="built_in">read</span> <span class="built_in">id</span></span><br><span class="line"><span class="keyword">do</span> <span class="built_in">nohup</span> samtools flagstat <span class="variable">$id</span> &amp;</span><br><span class="line"><span class="keyword">done</span></span><br><span class="line"></span><br><span class="line">gatk --java-options <span class="string">&quot;-Xmx20G&quot;</span> AnalyzeCovariates \</span><br><span class="line">     -before recal_data.table1 \</span><br><span class="line">     -after recal_data.table2 \</span><br><span class="line">     -plots AnalyzeCovariates.pdf</span><br></pre></td></tr></table></figure>

<h3 id="4、HaplotypeCaller"><a href="#4、HaplotypeCaller" class="headerlink" title="4、HaplotypeCaller"></a>4、HaplotypeCaller</h3><blockquote>
<p>这一步是使用GATK的HaplotypeCaller找变异，但现在GATK官网推荐GATK4+Mutect2找变异，在这里我们还是简单运行一下HaplotypeCaller。</p>
</blockquote>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line"><span class="built_in">ls</span> *bqsr.bam|<span class="keyword">while</span> <span class="built_in">read</span> sample</span><br><span class="line"><span class="keyword">do</span> <span class="built_in">nohup</span> gatk \</span><br><span class="line">         --java-options <span class="string">&quot;-Xmx20G -Djava.io.tmpdir=./&quot;</span> HaplotypeCaller \</span><br><span class="line">         -R <span class="variable">$ref</span>  \</span><br><span class="line">         -I <span class="variable">$sample</span> \</span><br><span class="line">         --dbsnp <span class="variable">$snp</span> \</span><br><span class="line">         -O ../vcf/<span class="variable">$&#123;sample%%_*&#125;</span>_raw.vcf \</span><br><span class="line">         1&gt;../vcf/<span class="variable">$&#123;sample%%_*&#125;</span>_log.HC 2&gt;&amp;1 &amp;</span><br><span class="line"><span class="keyword">done</span></span><br></pre></td></tr></table></figure>

<h2 id="三-Mutect2找变异"><a href="#三-Mutect2找变异" class="headerlink" title="(三) Mutect2找变异"></a>(三) Mutect2找变异</h2><p><strong><a target="_blank" rel="noopener" href="https://gatk.broadinstitute.org/hc/en-us/articles/360035889791?id=11136">Mutect2 v4.1.0.0的教程</a>目前已经不被GATK官方强烈推荐。官方更新推荐了<a target="_blank" rel="noopener" href="https://gatk.broadinstitute.org/hc/en-us/articles/360035531132">Mutect2 v4.1.1.0 and later版本的使用教程</a>，更新过的教程比较推荐用于正常对照样本&gt;40的WES测序数据。如果没有这么多的样本量，Mutect2 v4.1.0.0的教程也同样可以使用，只是有部分参数在新的Mutect2版本中被弃用或是将其功能整合到另外的参数去了。具体问题可以查看官网说明。</strong></p>
<h3 id="1、过滤种系突变"><a href="#1、过滤种系突变" class="headerlink" title="1、过滤种系突变"></a>1、过滤种系突变</h3><blockquote>
<p>首先需要一个germline variant sites VCF文件，去官网下载<a target="_blank" rel="noopener" href="https://console.cloud.google.com/storage/browser/gatk-best-practices/somatic-hg38">af-only-gnomad.hg38.vcf.gz</a>文件。记得一并下载对应的.tbi文件</p>
</blockquote>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br></pre></td><td class="code"><pre><span class="line">ref=/home/gongyuqi/ref/GATK/hg38/v0/Homo_sapiens_assembly38.fasta</span><br><span class="line">germine_vcf=/home/gongyuqi/ref/GATK/hg38/af-only-gnomad/af-only-gnomad.hg38.vcf.gz</span><br><span class="line"></span><br><span class="line"><span class="built_in">nohup</span> gatk --java-options <span class="string">&quot;-Xmx20g&quot;</span> Mutect2 \</span><br><span class="line">-R <span class="variable">$ref</span> \</span><br><span class="line">-I SRR12135912_bqsr.bam \</span><br><span class="line">-tumor SRR12135912 \</span><br><span class="line">-I SRR12135911_bqsr.bam \</span><br><span class="line">-normal SRR12135911 \</span><br><span class="line">--germline-resource <span class="variable">$germine_vcf</span> \</span><br><span class="line">--af-of-alleles-not-in-resource 0.0000025 \</span><br><span class="line">--disable-read-filter MateOnSameContigOrNoMappedMateReadFilter \</span><br><span class="line">--bam-output HQ461-untreated-Mutect2.bam \</span><br><span class="line">-O ../Mutect2/HQ461-untreated.Mutect2.vcf &amp;</span><br></pre></td></tr></table></figure>

<p>生成文件如下：<br><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/Mutect2.JPG"/></p>
<h3 id="2、考虑样品纯度及假阳性"><a href="#2、考虑样品纯度及假阳性" class="headerlink" title="2、考虑样品纯度及假阳性"></a>2、考虑样品纯度及假阳性</h3><blockquote>
<p>通常，病人的肿瘤样本是混合有正常组织的，这个时候我们可以选择<code>GetPileupSummaries</code>工具来计算肿瘤样本的污染程度。对于Mutect2流程而言，GATK只考虑了肿瘤样本中正常样本的污染情况，不考虑正常样本中肿瘤样本的污染情况。这里我们的处理组:HQ461,对照组untreated。，虽然我们的样本是肿瘤细胞的单克隆株，我们这里还是跑一下这个流程。</p>
</blockquote>
<h4 id="step-1、生成af-only-gnomad-hg38-SNP-biallelic-vcf-gz"><a href="#step-1、生成af-only-gnomad-hg38-SNP-biallelic-vcf-gz" class="headerlink" title="step 1、生成af-only-gnomad.hg38.SNP_biallelic.vcf.gz"></a>step 1、生成af-only-gnomad.hg38.SNP_biallelic.vcf.gz</h4><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">ref=/home/gongyuqi/ref/GATK/hg38/v0/Homo_sapiens_assembly38.fasta</span><br><span class="line">germine_vcf=/home/gongyuqi/ref/GATK/hg38/af-only-gnomad/af-only-gnomad.hg38.vcf.gz</span><br><span class="line"><span class="built_in">nohup</span> gatk SelectVariants \</span><br><span class="line">-R <span class="variable">$ref</span> \</span><br><span class="line">-V <span class="variable">$germine_vcf</span> \</span><br><span class="line">--select-type-to-include SNP \</span><br><span class="line">--restrict-alleles-to BIALLELIC \</span><br><span class="line">-O /home/gongyuqi/ref/GATK/hg38/af-only-gnomad/af-only-gnomad.hg38.SNP_biallelic.vcf.gz &amp;</span><br></pre></td></tr></table></figure>

<h4 id="step-2、GetPileupSummaries"><a href="#step-2、GetPileupSummaries" class="headerlink" title="step 2、GetPileupSummaries"></a>step 2、GetPileupSummaries</h4><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br></pre></td><td class="code"><pre><span class="line">germine_biallelic_vcf=/home/gongyuqi/ref/GATK/hg38/af-only-gnomad/af-only-gnomad.hg38.SNP_biallelic.vcf.gz</span><br><span class="line">genomic_intervals=/home/gongyuqi/ref/GATK/hg38/v0/wgs_calling_regions.hg38.interval_list</span><br><span class="line"></span><br><span class="line"><span class="built_in">nohup</span> gatk GetPileupSummaries \</span><br><span class="line">-I SRR12135911_bqsr.bam \</span><br><span class="line">-L <span class="variable">$genomic_intervals</span> \</span><br><span class="line">-V <span class="variable">$germine_biallelic_vcf</span> \</span><br><span class="line">-O ../Mutect2/SRR12135911.pileups.table &amp;</span><br><span class="line"></span><br><span class="line"><span class="built_in">nohup</span> gatk GetPileupSummaries \</span><br><span class="line">-I SRR12135912_bqsr.bam \</span><br><span class="line">-L <span class="variable">$genomic_intervals</span> \</span><br><span class="line">-V <span class="variable">$germine_biallelic_vcf</span> \</span><br><span class="line">-O ../Mutect2/SRR12135912.pileups.table &amp;</span><br></pre></td></tr></table></figure>

<h4 id="step-3、-CalculateContamination"><a href="#step-3、-CalculateContamination" class="headerlink" title="step 3、 CalculateContamination"></a>step 3、 CalculateContamination</h4><blockquote>
<p>计算污染比例，用以过滤掉somatic variant中一些可能由污染导致的假阳性突变</p>
</blockquote>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"><span class="built_in">nohup</span> gatk CalculateContamination \</span><br><span class="line">-I SRR12135912.pileups.table \</span><br><span class="line">-matched SRR12135911.pileups.table \</span><br><span class="line">-O HQ461-untreated.calculatecontamination.table &amp;</span><br></pre></td></tr></table></figure>

<h4 id="step-3、-FilterMutectCalls"><a href="#step-3、-FilterMutectCalls" class="headerlink" title="step 3、 FilterMutectCalls"></a>step 3、 FilterMutectCalls</h4><blockquote>
<p>结合CalculateContamination的评估，进行最后的过滤</p>
</blockquote>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">ref=/home/gongyuqi/ref/GATK/hg38/v0/Homo_sapiens_assembly38.fasta</span><br><span class="line"><span class="built_in">nohup</span> gatk FilterMutectCalls \</span><br><span class="line">-R <span class="variable">$ref</span> \</span><br><span class="line">-V HQ461-untreated.Mutect2.vcf \</span><br><span class="line">--contamination-table HQ461-untreated.calculatecontamination.table \</span><br><span class="line">-O HQ461-untreated.filtered.Mutect2.vcf &amp;</span><br></pre></td></tr></table></figure>

<blockquote>
<p>至此，我们找变异的步骤就差不多走完了。你会发现我们并没有使用到<code>CollectSequencingArtifactMetrics</code>和<code>FilterByOrientationBias</code>参数，那是因为我们当前版本的Mutect2已经将FilterByOrientationBias的功能整合进<code>FilterMutectCalls</code>中了，详情见<a target="_blank" rel="noopener" href="https://gatk.broadinstitute.org/hc/en-us/community/posts/360077855051-gatk-FilterByOrientationBias">GATK论坛中的一篇帖子</a>。</p>
</blockquote>
<blockquote>
<p>另外，在做WES分析时，对照组是很重要的。拿肿瘤样本举例，tumor-only的模式会得到很多很多很多的假阳性，详情见<a target="_blank" rel="noopener" href="https://gatk.broadinstitute.org/hc/en-us/community/posts/360057810051-Mutect2-somatic-variant-calling-with-without-matched-normal-sample">GATK论坛中的另一篇帖子</a>。对照组的存在会大大降低假阳性率。</p>
</blockquote>
<h2 id="四-SnpEff-amp-SnpSift注释"><a href="#四-SnpEff-amp-SnpSift注释" class="headerlink" title="(四) SnpEff &amp; SnpSift注释"></a>(四) SnpEff &amp; SnpSift注释</h2><blockquote>
<p>以下流程参考<a target="_blank" rel="noopener" href="https://pcingola.github.io/SnpEff/se_introduction/">SnpEff &amp; SnpSift官网文档</a></p>
</blockquote>
<h3 id="1、SnpEff软件及所需注释数据库的下载"><a href="#1、SnpEff软件及所需注释数据库的下载" class="headerlink" title="1、SnpEff软件及所需注释数据库的下载"></a>1、SnpEff软件及所需注释数据库的下载</h3><h4 id="step-1、下载安装SnpEff软件"><a href="#step-1、下载安装SnpEff软件" class="headerlink" title="step 1、下载安装SnpEff软件"></a>step 1、下载安装SnpEff软件</h4><p>下载<a target="_blank" rel="noopener" href="https://pcingola.github.io/SnpEff/download/">SnpEff</a>，现在SnpEff和SnpSift是绑定在一起的。</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line"><span class="built_in">cd</span> ~/biosoft/</span><br><span class="line"><span class="comment">#下载</span></span><br><span class="line">wget https://snpeff.blob.core.windows.net/versions/snpEff_latest_core.zip &amp;</span><br><span class="line"><span class="comment">#安装</span></span><br><span class="line">unzip snpEff_latest_core.zip</span><br></pre></td></tr></table></figure>


<h4 id="step-2、下载SnpEff软件需要的数据库文件"><a href="#step-2、下载SnpEff软件需要的数据库文件" class="headerlink" title="step 2、下载SnpEff软件需要的数据库文件"></a>step 2、下载SnpEff软件需要的数据库文件</h4><p>下载 SnpEff databases: 官网给的命令是java -jar snpEff.jar download GRCh38.76。但是运行这个命令会报错：找不到GRCh38.76</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#运行下面的命令查看SnpEff目前支持的database</span></span><br><span class="line">java -jar snpEff.jar databases | less</span><br><span class="line"><span class="comment">#运行下面的命令查看SnpEff支持的GRCh38 database,目前是GRCh38.99</span></span><br><span class="line">java -jar snpEff.jar databases | grep -i GRCh38</span><br><span class="line"><span class="comment">#看来SnpEff网站更新不及时呀，在网站维护和更新这一点上，要重点表扬GATK团队！！！</span></span><br></pre></td></tr></table></figure>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/GRCh38.99.JPG"/>


<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#于是下载GRCh38.99</span></span><br><span class="line"><span class="built_in">nohup</span> java -jar snpEff.jar download GRCh38.99 &amp;</span><br><span class="line"><span class="comment">#但是这个网络问题是无解的。我最后还是自己手动下载snpEff_v5_0_GRCh38.99.zip文件</span></span><br><span class="line"><span class="comment">#解压完，最后GRCh38.99文件夹绝对路径/home/gongyuqi/biosoft/snpEff/data/GRCh38.99</span></span><br></pre></td></tr></table></figure>
<p>GRCh38.99内容如下<br><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/GRCh38.99.content.JPG"/></p>
<h3 id="2、SnpEff、SnpSift使用"><a href="#2、SnpEff、SnpSift使用" class="headerlink" title="2、SnpEff、SnpSift使用"></a>2、SnpEff、SnpSift使用</h3><h4 id="step-1、SnpEff注释"><a href="#step-1、SnpEff注释" class="headerlink" title="step 1、SnpEff注释"></a>step 1、SnpEff注释</h4><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#the &quot;verbose&quot; mode (command line option -v), this makes SnpEff to show a lot of information which can be useful for debugging.</span></span><br><span class="line">java -Xmx16g -jar snpEff.jar -v GRCh38.99 /home/gongyuqi/project/WES/Mutect2/HQ461-untreated.filtered.Mutect2.vcf &gt; HQ461-untreated.filtered.ann.vcf</span><br></pre></td></tr></table></figure>


<h4 id="step-2、SnpSift过滤"><a href="#step-2、SnpSift过滤" class="headerlink" title="step 2、SnpSift过滤"></a>step 2、SnpSift过滤</h4><p>该部分同时参考官网说明文档及某优秀博客<a target="_blank" rel="noopener" href="http://www.sci666.net/40794.html">用SnpSift过滤VCF文件</a></p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#保留Filter字段为&#x27;PASS&#x27;或缺失值的记录</span></span><br><span class="line"><span class="built_in">cat</span> HQ461-untreated.filtered.ann.vcf | java -jar SnpSift.jar filter <span class="string">&quot;( na FILTER ) | (FILTER = &#x27;PASS&#x27;)&quot;</span> &gt; HQ461-untreated.filtered.ann.snpsift.vcf</span><br><span class="line"><span class="comment">#保留INFO中ANN字段的IMPACT为&#x27;HIGH&#x27;或&#x27;MODERATE&#x27;的记录</span></span><br><span class="line"><span class="built_in">cat</span> HQ461-untreated.filtered.ann.snpsift.vcf | java -jar SnpSift.jar filter <span class="string">&quot;(ANN[0].IMPACT has &#x27;HIGH&#x27;) | (ANN[0].IMPACT has &#x27;MODERATE&#x27;)&quot;</span> &gt; HQ461-untreated.filtered.ann.snpsift.second.vcf</span><br></pre></td></tr></table></figure>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#简单看一下三个vcf文件的大小</span></span><br><span class="line"><span class="built_in">ls</span> -lh *.vcf</span><br><span class="line"><span class="built_in">ls</span> *.vcf|<span class="keyword">while</span> <span class="built_in">read</span> <span class="built_in">id</span>; <span class="keyword">do</span> <span class="built_in">cat</span> <span class="variable">$id</span>|<span class="built_in">wc</span> -l;<span class="keyword">done</span></span><br></pre></td></tr></table></figure>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/vcf.JPG"/>


<h4 id="step-3、sed命令复现SnpSift的功能"><a href="#step-3、sed命令复现SnpSift的功能" class="headerlink" title="step 3、sed命令复现SnpSift的功能"></a>step 3、<code>sed</code>命令复现SnpSift的功能</h4><blockquote>
<p>这里不难发现，SnpSift其实就是对文本文件的处理。要是shell脚本够扎实，也完全可以不依赖SnpSift。重要的是，在学习NGS组学分析过程中，但凡有锻炼自己shell脚本的地方，就一定要抓住机会动手写一写💪 <strong>这里我们可以用一个简单的<code>sed</code>命令解决</strong></p>
</blockquote>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#第一个`sed`命令打印出所有标记为PASS的位点</span></span><br><span class="line"><span class="comment">#第二个`sed`命令承接上一个管道输出的内容，保留标记为HIGH或者MODERATE的位点</span></span><br><span class="line"><span class="built_in">cat</span> HQ461-untreated.filtered.ann.vcf | grep -v <span class="string">&quot;#&quot;</span>| \</span><br><span class="line">sed -n <span class="string">&#x27;/PASS/p&#x27;</span> - | \</span><br><span class="line">sed -n <span class="string">&#x27;/\(HIGH\|MODERATE\)/p&#x27;</span> - \</span><br><span class="line">&gt;test_PASS_HIGH_MODERATE.third.vcf</span><br></pre></td></tr></table></figure>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#检验一下我们`sed`脚本的正确性</span></span><br><span class="line"><span class="comment">#看看位点数目是否一致</span></span><br><span class="line"><span class="built_in">cat</span> HQ461-untreated.filtered.ann.snpsift.second.vcf|grep -v <span class="string">&quot;#&quot;</span>|<span class="built_in">wc</span> -l</span><br><span class="line"><span class="built_in">cat</span> test_PASS_HIGH_MODERATE.second.vcf|<span class="built_in">wc</span> -l</span><br></pre></td></tr></table></figure>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/wc_l.JPG"/>

<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#再具体看看内容是否一致</span></span><br><span class="line"><span class="built_in">cat</span> HQ461-untreated.filtered.ann.snpsift.second.vcf|grep -v <span class="string">&quot;#&quot;</span>|less -SN</span><br><span class="line"><span class="built_in">cat</span> test_PASS_HIGH_MODERATE.second.vcf|less -SN</span><br></pre></td></tr></table></figure>
<p>HQ461-untreated.filtered.ann.snpsift.second.vcf<br><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/SnpSift_second.JPG"/><br>test_PASS_HIGH_MODERATE.second.vcf<br><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/sed.JPG"/></p>
<h2 id="五-原文数据复现"><a href="#五-原文数据复现" class="headerlink" title="(五) 原文数据复现"></a>(五) 原文数据复现</h2><blockquote>
<p>真正接受检验的时候到了，看看我们的pipeline能不能复现文章的数据。这也是进一步检验我们pipeline的正确性的重要环节（当然也不能迷信文章的结果，但是7+文章的数据分析质量应该还是可以作为一个较好的参考）。<br><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/G_to_A.JPG"/></p>
</blockquote>
<p><strong>再看看我们过滤后的vcf文件中是否有检测到CDK12的G到A的突变</strong><br><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/CDK12_G_to_A.JPG"/></p>
<p><strong>学习WES心得体会</strong></p>
<p>1、学习某个软件的用法时，主要参考官方文档！！！可以选择性参考相关博客主博客，但切记拿来主义！！！深刻理解用到的每个参数。</p>
<p>2、WES的相关分析还有很多，我们这里只涉及到SNV。同样的，突变的注释也涉及到多种软件，我们这里只涉及到了SnpEff。学海无涯啊~~~从培养起自主学习的思维和习惯开始。</p>
<p>3、拿到最终的注释文件（.vcf），我们还可以用到maftools、ComplexHeatmap等R包将突变情况进行可视化。</p>
<p>4、不急不慢，不慌不忙，坚持下去❤</p>
</div><div class="article-licensing box"><div class="licensing-title"><p>WES</p><p><a href="http://gyqgyq.com/WES/">http://gyqgyq.com/WES/</a></p></div><div class="licensing-meta level is-mobile"><div class="level-left"><div class="level-item is-narrow"><div><h6>Author</h6><p>Yuqi Gong</p></div></div><div class="level-item is-narrow"><div><h6>Posted on</h6><p>2021-05-14</p></div></div><div class="level-item is-narrow"><div><h6>Updated on</h6><p>2022-11-26</p></div></div><div class="level-item is-narrow"><div><h6>Licensed under</h6><p><a class="icons" rel="noopener" target="_blank" title="Creative Commons" href="https://creativecommons.org/"><i class="icon fab fa-creative-commons"></i></a><a class="icons" rel="noopener" target="_blank" title="Attribution" href="https://creativecommons.org/licenses/by/4.0/"><i class="icon fab fa-creative-commons-by"></i></a><a class="icons" rel="noopener" target="_blank" title="Noncommercial" href="https://creativecommons.org/licenses/by-nc/4.0/"><i class="icon fab fa-creative-commons-nc"></i></a></p></div></div></div></div></div><div class="article-tags is-size-7 mb-4"><span class="mr-2">#</span><a class="link-muted mr-2" rel="tag" href="/tags/tag1/">tag1</a></div><div class="a2a_kit a2a_kit_size_32 a2a_default_style"><a class="a2a_dd" target="_blank" rel="noopener" href="https://www.addtoany.com/share"></a><a class="a2a_button_facebook"></a><a class="a2a_button_twitter"></a><a class="a2a_button_telegram"></a><a class="a2a_button_whatsapp"></a><a class="a2a_button_reddit"></a></div><script src="https://static.addtoany.com/menu/page.js" defer></script></article></div><div class="card"><div class="card-content"><h3 class="menu-label has-text-centered">Like this article? Support the author with</h3><div class="buttons is-centered"><a class="button donate" data-type="alipay"><span class="icon is-small"><i class="fab fa-alipay"></i></span><span>Alipay</span><span class="qrcode"><img src="/" alt="Alipay"></span></a><a class="button donate" data-type="wechat"><span class="icon is-small"><i class="fab fa-weixin"></i></span><span>Wechat</span><span class="qrcode"><img src="/" alt="Wechat"></span></a></div></div></div><nav class="post-navigation mt-4 level is-mobile"><div class="level-start"><a class="article-nav-prev level level-item link-muted" href="/MeRIP-Seq/"><i class="level-item fas fa-chevron-left"></i><span class="level-item">MeRIP-Seq</span></a></div><div class="level-end"><a class="article-nav-next level level-item link-muted" href="/scRNA-seq-%E6%95%B0%E6%8D%AE%E5%A4%8D%E7%8E%B0/"><span class="level-item">scRNA-seq_数据复现</span><i class="level-item fas fa-chevron-right"></i></a></div></nav><!--!--></div><div class="column column-left is-4-tablet is-4-desktop is-4-widescreen  order-1"><div class="card widget" data-type="profile"><div class="card-content"><nav class="level"><div class="level-item has-text-centered flex-shrink-1"><div><figure class="image is-128x128 mx-auto mb-2"><img class="avatar" src="/img/ava.jpg" alt="Yuqi Gong"></figure><p class="title is-size-4 is-block" style="line-height:inherit;">Yuqi Gong</p><p class="is-size-6 is-flex justify-content-center"><i class="fas fa-map-marker-alt mr-1"></i><span>HangZhou, China</span></p></div></div></nav><nav class="level is-mobile"><div class="level-item has-text-centered is-marginless"><div><p class="heading">Posts</p><a href="/archives"><p class="title">38</p></a></div></div><div class="level-item has-text-centered is-marginless"><div><p class="heading">Category</p><a href="/categories"><p class="title">1</p></a></div></div><div class="level-item has-text-centered is-marginless"><div><p class="heading">Tags</p><a href="/tags"><p class="title">7</p></a></div></div></nav><div class="level"><a class="level-item button is-primary is-rounded" href="https://github.com/gongyuqiqi" target="_blank" rel="noopener">Follow</a></div><div class="level is-mobile is-multiline"><a class="level-item button is-transparent is-marginless" target="_blank" rel="noopener" title="Github" href="https://github.com/gongyuqiqi"><i class="fab fa-github"></i></a><a class="level-item button is-transparent is-marginless" target="_blank" rel="noopener" title="Facebook" href="https://facebook.com"><i class="fab fa-facebook"></i></a><a class="level-item button is-transparent is-marginless" target="_blank" rel="noopener" title="Twitter" href="https://twitter.com"><i class="fab fa-twitter"></i></a><a class="level-item button is-transparent is-marginless" target="_blank" rel="noopener" title="Dribbble" href="https://dribbble.com"><i class="fab fa-dribbble"></i></a><a class="level-item button is-transparent is-marginless" target="_blank" rel="noopener" title="RSS" href="/"><i class="fas fa-rss"></i></a></div></div></div><div class="card widget" id="toc" data-type="toc"><div class="card-content"><div class="menu"><h3 class="menu-label">Catalogue</h3><ul class="menu-list"><li><a class="level is-mobile" href="#全外显子数据分析"><span class="level-left"><span class="level-item">1</span><span class="level-item">全外显子数据分析</span></span></a><ul class="menu-list"><li><a class="level is-mobile" href="#一-环境搭建"><span class="level-left"><span class="level-item">1.1</span><span class="level-item">(一) 环境搭建</span></span></a></li><li><a class="level is-mobile" href="#二-WES测试数据下载"><span class="level-left"><span class="level-item">1.2</span><span class="level-item">(二) WES测试数据下载</span></span></a></li><li><a class="level is-mobile" href="#三-GATK准备bam文件用于找变异"><span class="level-left"><span class="level-item">1.3</span><span class="level-item">(三) GATK准备bam文件用于找变异</span></span></a><ul class="menu-list"><li><a class="level is-mobile" href="#1、-比对GATK官网提供的参考基因组"><span class="level-left"><span class="level-item">1.3.1</span><span class="level-item">1、 比对GATK官网提供的参考基因组</span></span></a></li><li><a class="level is-mobile" href="#2、标记PCR重复"><span class="level-left"><span class="level-item">1.3.2</span><span class="level-item">2、标记PCR重复</span></span></a></li><li><a class="level is-mobile" href="#3、碱基矫正-BaseRecalibrator"><span class="level-left"><span class="level-item">1.3.3</span><span class="level-item">3、碱基矫正 BaseRecalibrator</span></span></a></li><li><a class="level is-mobile" href="#4、HaplotypeCaller"><span class="level-left"><span class="level-item">1.3.4</span><span class="level-item">4、HaplotypeCaller</span></span></a></li></ul></li><li><a class="level is-mobile" href="#三-Mutect2找变异"><span class="level-left"><span class="level-item">1.4</span><span class="level-item">(三) Mutect2找变异</span></span></a><ul class="menu-list"><li><a class="level is-mobile" href="#1、过滤种系突变"><span class="level-left"><span class="level-item">1.4.1</span><span class="level-item">1、过滤种系突变</span></span></a></li><li><a class="level is-mobile" href="#2、考虑样品纯度及假阳性"><span class="level-left"><span class="level-item">1.4.2</span><span class="level-item">2、考虑样品纯度及假阳性</span></span></a></li></ul></li><li><a class="level is-mobile" href="#四-SnpEff-amp-SnpSift注释"><span class="level-left"><span class="level-item">1.5</span><span class="level-item">(四) SnpEff &amp; SnpSift注释</span></span></a><ul class="menu-list"><li><a class="level is-mobile" href="#1、SnpEff软件及所需注释数据库的下载"><span class="level-left"><span class="level-item">1.5.1</span><span class="level-item">1、SnpEff软件及所需注释数据库的下载</span></span></a></li><li><a class="level is-mobile" href="#2、SnpEff、SnpSift使用"><span class="level-left"><span class="level-item">1.5.2</span><span class="level-item">2、SnpEff、SnpSift使用</span></span></a></li></ul></li><li><a class="level is-mobile" href="#五-原文数据复现"><span class="level-left"><span class="level-item">1.6</span><span class="level-item">(五) 原文数据复现</span></span></a></li></ul></li></ul></div></div><style>#toc .menu-list > li > a.is-active + .menu-list { display: block; }#toc .menu-list > li > a + .menu-list { display: none; }</style><script src="/js/toc.js" defer></script></div><div class="card widget" data-type="categories"><div class="card-content"><div class="menu"><h3 class="menu-label">Categories</h3><ul class="menu-list"><li><a class="level is-mobile" href="/categories/Programming-Language/"><span class="level-start"><span class="level-item">Programming Language</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></div></div></div><div class="card widget" data-type="recent-posts"><div class="card-content"><h3 class="menu-label">Recents</h3><article class="media"><div class="media-content"><p class="date"><time dateTime="2024-01-18T13:40:44.000Z">2024-01-18</time></p><p class="title"><a href="/test/">test</a></p></div></article><article class="media"><div class="media-content"><p class="date"><time dateTime="2021-07-04T05:48:30.000Z">2021-07-04</time></p><p class="title"><a href="/RNA-TNBC/">RNA-TNBC</a></p></div></article><article class="media"><div class="media-content"><p class="date"><time dateTime="2021-06-29T06:01:05.000Z">2021-06-29</time></p><p class="title"><a href="/scRNA-TNBC/">scRNA-TNBC</a></p></div></article><article class="media"><div class="media-content"><p class="date"><time dateTime="2021-06-03T04:43:57.000Z">2021-06-03</time></p><p class="title"><a href="/MeRIP-Seq/">MeRIP-Seq</a></p></div></article><article class="media"><div class="media-content"><p class="date"><time dateTime="2021-05-14T05:37:00.000Z">2021-05-14</time></p><p class="title"><a href="/WES/">WES</a></p></div></article></div></div><div class="card widget" data-type="archives"><div class="card-content"><div class="menu"><h3 class="menu-label">Archives</h3><ul class="menu-list"><li><a class="level is-mobile" href="/archives/2024/01/"><span class="level-start"><span class="level-item">January 2024</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/archives/2021/07/"><span class="level-start"><span class="level-item">July 2021</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/archives/2021/06/"><span class="level-start"><span class="level-item">June 2021</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/archives/2021/05/"><span class="level-start"><span class="level-item">May 2021</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li><li><a class="level is-mobile" href="/archives/2021/03/"><span class="level-start"><span class="level-item">March 2021</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/archives/2021/02/"><span class="level-start"><span class="level-item">February 2021</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/archives/2021/01/"><span class="level-start"><span class="level-item">January 2021</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li><li><a class="level is-mobile" href="/archives/2020/12/"><span class="level-start"><span class="level-item">December 2020</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/archives/2020/05/"><span class="level-start"><span class="level-item">May 2020</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/archives/2020/04/"><span class="level-start"><span class="level-item">April 2020</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/archives/2020/03/"><span class="level-start"><span class="level-item">March 2020</span></span><span class="level-end"><span class="level-item tag">5</span></span></a></li><li><a class="level is-mobile" href="/archives/2020/02/"><span class="level-start"><span class="level-item">February 2020</span></span><span class="level-end"><span class="level-item tag">18</span></span></a></li></ul></div></div></div><div class="card widget" data-type="tags"><div class="card-content"><div class="menu"><h3 class="menu-label">Tags</h3><div class="field is-grouped is-grouped-multiline"><div class="control"><a class="tags has-addons" href="/tags/Linux/"><span class="tag">Linux</span><span class="tag">7</span></a></div><div class="control"><a class="tags has-addons" href="/tags/MySQL/"><span class="tag">MySQL</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/NGS/"><span class="tag">NGS</span><span class="tag">18</span></a></div><div class="control"><a class="tags has-addons" href="/tags/R/"><span class="tag">R</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/System/"><span class="tag">System</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/notes/"><span class="tag">notes</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/tag1/"><span class="tag">tag1</span><span class="tag">6</span></a></div></div></div></div></div><div class="card widget" data-type="subscribe-email"><div class="card-content"><div class="menu"><h3 class="menu-label">follow.it</h3><form action="" method="post" target="_blank"><div class="field has-addons"><div class="control has-icons-left is-expanded"><input class="input" name="email" type="email" placeholder="Email"><span class="icon is-small is-left"><i class="fas fa-envelope"></i></span></div><div class="control"><input class="button" type="submit" value="Subscribe"></div></div></form></div></div></div></div><!--!--></div></div></section><footer class="footer"><div class="container"><div class="level"><div class="level-start"><a class="footer-logo is-block mb-2" href="/"><img src="/img/ava.jpg" alt="Yuqi Gong&#039;s blog" height="28"></a><p class="is-size-7"><span>&copy; 2024 Yuqi Gong</span>  Powered by Gong Yuqi <i class="fa-heart fa fa-beat" style="color: #f00;"></i><br><span id="busuanzi_container_site_uv">Visited by <span id="busuanzi_value_site_uv">0</span> users</span></p></div><div class="level-end"><div class="field has-addons"><p class="control"><a class="button is-transparent is-large" target="_blank" rel="noopener" title="Creative Commons" href="https://creativecommons.org/"><i class="fab fa-creative-commons"></i></a></p><p class="control"><a class="button is-transparent is-large" target="_blank" rel="noopener" title="Attribution 4.0 International" href="https://creativecommons.org/licenses/by/4.0/"><i class="fab fa-creative-commons-by"></i></a></p><p class="control"><a class="button is-transparent is-large" target="_blank" rel="noopener" title="Download on GitHub" href="https://github.com/gongyuqiqi/gongyuqiqi.github.io"><i class="fab fa-github"></i></a></p></div></div></div></div></footer><script src="https://cdn.jsdelivr.net/npm/jquery@3.3.1/dist/jquery.min.js"></script><script src="https://cdn.jsdelivr.net/npm/moment@2.22.2/min/moment-with-locales.min.js"></script><script src="https://cdn.jsdelivr.net/npm/clipboard@2.0.4/dist/clipboard.min.js" defer></script><script>moment.locale("en");</script><script>var IcarusThemeSettings = {
            article: {
                highlight: {
                    clipboard: true,
                    fold: 'unfolded'
                }
            }
        };</script><script src="/js/column.js"></script><script src="/js/animation.js"></script><a id="back-to-top" title="Back to top" href="javascript:;"><i class="fas fa-chevron-up"></i></a><script src="/js/back_to_top.js" defer></script><!--!--><!--!--><!--!--><!--!--><script src="https://cdn.jsdelivr.net/npm/cookieconsent@3.1.1/build/cookieconsent.min.js" defer></script><script>window.addEventListener("load", () => {
      window.cookieconsent.initialise({
        type: "info",
        theme: "edgeless",
        static: false,
        position: "bottom-left",
        content: {
          message: "This website uses cookies to improve your experience.",
          dismiss: "Got it!",
          allow: "Allow cookies",
          deny: "Decline",
          link: "Learn more",
          policy: "Cookie Policy",
          href: "https://www.cookiesandyou.com/",
        },
        palette: {
          popup: {
            background: "#edeff5",
            text: "#838391"
          },
          button: {
            background: "#4b81e8"
          },
        },
      });
    });</script><script src="https://cdn.jsdelivr.net/npm/lightgallery@1.10.0/dist/js/lightgallery.min.js" defer></script><script src="https://cdn.jsdelivr.net/npm/justifiedGallery@3.8.1/dist/js/jquery.justifiedGallery.min.js" defer></script><script>window.addEventListener("load", () => {
            if (typeof $.fn.lightGallery === 'function') {
                $('.article').lightGallery({ selector: '.gallery-item' });
            }
            if (typeof $.fn.justifiedGallery === 'function') {
                if ($('.justified-gallery > p > .gallery-item').length) {
                    $('.justified-gallery > p > .gallery-item').unwrap();
                }
                $('.justified-gallery').justifiedGallery();
            }
        });</script><!--!--><!--!--><!--!--><!--!--><!--!--><script src="/js/main.js" defer></script><div class="searchbox"><div class="searchbox-container"><div class="searchbox-header"><div class="searchbox-input-container"><input class="searchbox-input" type="text" placeholder="Type something..."></div><a class="searchbox-close" href="javascript:;">×</a></div><div class="searchbox-body"></div></div></div><script src="/js/insight.js" defer></script><script>document.addEventListener('DOMContentLoaded', function () {
            loadInsight({"contentUrl":"/content.json"}, {"hint":"Type something...","untitled":"(Untitled)","posts":"Posts","pages":"Pages","categories":"Categories","tags":"Tags"});
        });</script></body></html>