<!doctype html>
<html lang="en"><head><meta charset="utf-8"><meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1"><meta><title>可变剪切分析之MISO - Gong Yuqi&#039;s blog</title><link rel="manifest" href="/manifest.json"><meta name="application-name" content="Gong Yuqi&#039;s blog"><meta name="msapplication-TileImage" content="/img/ava.jpg"><meta name="apple-mobile-web-app-capable" content="yes"><meta name="apple-mobile-web-app-title" content="Gong Yuqi&#039;s blog"><meta name="apple-mobile-web-app-status-bar-style" content="default"><meta name="description" content="RNA-seq高级分析之可变剪切"><meta property="og:type" content="blog"><meta property="og:title" content="可变剪切分析之MISO"><meta property="og:url" content="http://gyqgyq.com/%E5%8F%AF%E5%8F%98%E5%89%AA%E5%88%87%E4%B9%8BMISO/"><meta property="og:site_name" content="Gong Yuqi&#039;s blog"><meta property="og:description" content="RNA-seq高级分析之可变剪切"><meta property="og:locale" content="en_US"><meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/align1.JPG"><meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/align2.JPG"><meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/align4.JPG"><meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/unzipgff.JPG"><meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/ESgff3index.JPG"><meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/miso_run_out.JPG"><meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/filter.JPG"><meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/sashimi_plot1.JPG"><meta property="article:published_time" content="2021-01-09T06:05:52.000Z"><meta property="article:modified_time" content="2022-11-26T06:08:08.861Z"><meta property="article:author" content="Gong Yuqi"><meta property="article:tag" content="NGS"><meta property="twitter:card" content="summary"><meta property="twitter:image:src" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/align1.JPG"><script type="application/ld+json">{"@context":"https://schema.org","@type":"BlogPosting","mainEntityOfPage":{"@type":"WebPage","@id":"http://gyqgyq.com/%E5%8F%AF%E5%8F%98%E5%89%AA%E5%88%87%E4%B9%8BMISO/"},"headline":"可变剪切分析之MISO","image":[],"datePublished":"2021-01-09T06:05:52.000Z","dateModified":"2022-11-26T06:08:08.861Z","author":{"@type":"Person","name":"Gong Yuqi"},"publisher":{"@type":"Organization","name":"Gong Yuqi's blog","logo":{"@type":"ImageObject","url":"http://gyqgyq.com/img/ava.jpg"}},"description":"RNA-seq高级分析之可变剪切"}</script><link rel="canonical" href="http://gyqgyq.com/%E5%8F%AF%E5%8F%98%E5%89%AA%E5%88%87%E4%B9%8BMISO/"><link rel="icon" href="/img/ava.jpg"><link rel="stylesheet" href="https://use.fontawesome.com/releases/v6.0.0/css/all.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/highlight.js@9.12.0/styles/atom-one-light.css"><link rel="stylesheet" href="https://fonts.googleapis.com/css2?family=Ubuntu:wght@400;600&amp;family=Source+Code+Pro"><link rel="stylesheet" href="/css/default.css"><style>body>.footer,body>.navbar,body>.section{opacity:0}</style><!--!--><!--!--><!--!--><script src="//busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js" defer></script><!--!--><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/cookieconsent@3.1.1/build/cookieconsent.min.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/lightgallery@1.10.0/dist/css/lightgallery.min.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/justifiedGallery@3.8.1/dist/css/justifiedGallery.min.css"><!--!--><!--!--><style>.pace{-webkit-pointer-events:none;pointer-events:none;-webkit-user-select:none;-moz-user-select:none;user-select:none}.pace-inactive{display:none}.pace .pace-progress{background:#3273dc;position:fixed;z-index:2000;top:0;right:100%;width:100%;height:2px}</style><script src="https://cdn.jsdelivr.net/npm/pace-js@1.2.4/pace.min.js"></script><!--!--><!--!--><!-- hexo injector head_end start --><script>
  (function () {
      function switchTab() {
          if (!location.hash) {
            return;
          }

          const $tabMenu = document.querySelector(`a[href="${location.hash}"]`);
          if (!$tabMenu) {
            return;
          }

          const $tabMenuContainer = $tabMenu.parentElement.parentElement;
          Array.from($tabMenuContainer.children).forEach($menu => $menu.classList.remove('is-active'));
          Array.from($tabMenuContainer.querySelectorAll('a'))
              .map($menu => document.getElementById($menu.getAttribute("href").substring(1)))
              .forEach($content => $content.classList.add('is-hidden'));

          if ($tabMenu) {
              $tabMenu.parentElement.classList.add('is-active');
          }
          const $activeTab = document.querySelector(location.hash);
          if ($activeTab) {
              $activeTab.classList.remove('is-hidden');
          }
      }
      switchTab();
      window.addEventListener('hashchange', switchTab, false);
  })();
  </script><!-- hexo injector head_end end --><meta name="generator" content="Hexo 6.3.0"></head><body class="is-2-column"><nav class="navbar navbar-main"><div class="container navbar-container"><div class="navbar-brand justify-content-center"><a class="navbar-item navbar-logo" href="/"><img src="/img/ava.jpg" alt="Gong Yuqi&#039;s blog" height="28"></a></div><div class="navbar-menu"><div class="navbar-start"><a class="navbar-item" href="/">Home</a><a class="navbar-item" href="/archives">Archives</a><a class="navbar-item" href="/categories">Categories</a><a class="navbar-item" href="/tags">Tags</a><a class="navbar-item" href="/about">About</a></div><div class="navbar-end"><a class="navbar-item" target="_blank" rel="noopener" title="Download on GitHub" href="https://github.com/gongyuqiqi/gongyuqiqi.github.io"><i class="fab fa-github"></i></a><a class="navbar-item is-hidden-tablet catalogue" title="Catalogue" href="javascript:;"><i class="fas fa-list-ul"></i></a><a class="navbar-item search" title="Search" href="javascript:;"><i class="fas fa-search"></i></a></div></div></div></nav><section class="section"><div class="container"><div class="columns"><div class="column order-2 column-main is-8-tablet is-8-desktop is-8-widescreen"><div class="card"><article class="card-content article" role="article"><div class="article-meta is-size-7 is-uppercase level is-mobile"><div class="level-left"><span class="level-item">Posted&nbsp;<time dateTime="2021-01-09T06:05:52.000Z" title="1/9/2021, 2:05:52 PM">2021-01-09</time></span><span class="level-item">Updated&nbsp;<time dateTime="2022-11-26T06:08:08.861Z" title="11/26/2022, 2:08:08 PM">2022-11-26</time></span><span class="level-item">11 minutes read (About 1665 words)</span><span class="level-item" id="busuanzi_container_page_pv"><span id="busuanzi_value_page_pv">0</span>&nbsp;visits</span></div></div><h1 class="title is-3 is-size-4-mobile">可变剪切分析之MISO</h1><div class="content"><p>RNA-seq高级分析之可变剪切</p>
<span id="more"></span>

<h1 id="可变剪切分析之MISO"><a href="#可变剪切分析之MISO" class="headerlink" title="可变剪切分析之MISO"></a>可变剪切分析之MISO</h1><p>MISO软件原文档：<a target="_blank" rel="noopener" href="https://miso.readthedocs.io/en/fastmiso/index.html#installing-fastmiso">https://miso.readthedocs.io/en/fastmiso/index.html#installing-fastmiso</a></p>
<h2 id="数据下载"><a href="#数据下载" class="headerlink" title="数据下载"></a>数据下载</h2><ul>
<li>写一个bash脚本下载数据<br>一次性下载所有的.fastq.gz文件，本次实验数据是单端测序数据<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#！/bin/bash</span></span><br><span class="line"><span class="built_in">dir</span>=/home/gongyuqi/.aspera/connect/etc/asperaweb_id_dsa.openssh</span><br><span class="line"><span class="keyword">for</span> <span class="built_in">id</span> <span class="keyword">in</span>  &#123;84,78,76,72,82,80,74,68&#125; </span><br><span class="line"><span class="keyword">do</span> </span><br><span class="line">       <span class="built_in">nohup</span> ascp -QT -l 300m -P33001 -i <span class="variable">$dsa</span> era-fasp@fasp.sra.ebi.ac.uk:/vol1/fastq/SRR974/SRR9749<span class="variable">$id</span>/SRR9749<span class="variable">$id</span>.fastq.gz . &amp;</span><br><span class="line"><span class="keyword">done</span> </span><br></pre></td></tr></table></figure></li>
<li>执行脚本</li>
</ul>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"><span class="built_in">chmod</span> +x download.sh</span><br><span class="line"><span class="built_in">nohup</span> ./download.sh &amp;</span><br></pre></td></tr></table></figure>
<h2 id="比对"><a href="#比对" class="headerlink" title="比对"></a>比对</h2><p>参考资料：<a target="_blank" rel="noopener" href="http://www.360doc.com/content/18/0714/20/19913717_770401548.shtml">http://www.360doc.com/content/18/0714/20/19913717_770401548.shtml</a></p>
<blockquote>
<ul>
<li>tophat进行转录组的比对时，输出的是bam（二进制的sam）文件。</li>
<li>tophat依赖bowtie2（bowtie1也可以），参考基因组索引用bowtie2-build进行构建。</li>
<li>如果FASTA格式的基因组与索引文件不在同一个文件夹下面，tophat会在运行中自动生成，所以记得把参考基因组和索引文件放在同一个文件夹下面，以减少系统运行时的高消耗。</li>
<li>如果GTF或者GFF格式的基因组注释文件存在， tophat会优先比对到注释文件的转录组上。最好事先准备好转录组索引以节约后续的比对时间。</li>
<li>如果没有事先准备好转录组索引，tophat会在运行过程中生成转录组索引，这样就比较耗时耗力了。</li>
</ul>
</blockquote>
<p><strong><table><tr><td bgcolor=GreenYellow><strong>参考基因组索引构建、转录组索引构建</strong></td></tr></table></strong></p>
<p>hg19参考基因组gtf文件下载链接：<a target="_blank" rel="noopener" href="https://www.gencodegenes.org/human/release_36lift37.html">https://www.gencodegenes.org/human/release_36lift37.html</a></p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#建立hg19参考基因组索引文件</span></span><br><span class="line"><span class="built_in">nohup</span> time bowtie2-build path/to/hg19.fa ./hg19 &gt; hg19.bowtie_index.log 2&gt;&amp;1 &amp;</span><br><span class="line"><span class="comment">#建立hg19转录组索引文件</span></span><br><span class="line"><span class="comment">#../index/bowtie/hg19为上一步构建的hg19参考基因组索引文件的相对路径及前缀</span></span><br><span class="line"><span class="built_in">nohup</span> tophat -G ../gencode.v36lift37.annotation.gtf --transcriptome-index=hg19.tr ../index/bowtie/hg19 &amp;</span><br></pre></td></tr></table></figure>


<p><strong><table><tr><td bgcolor=GreenYellow><strong>参考基因组的比对</strong></td></tr></table></strong> </p>
<h5 id="比对方法一"><a href="#比对方法一" class="headerlink" title="比对方法一"></a>比对方法一</h5><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#以其中一个样本为例，走一波比对转录组流程。</span></span><br><span class="line"><span class="comment">#理论上可以节约时间，但是实际跑起来非常耗时！</span></span><br><span class="line"><span class="comment">#1G左右得样本跑了4个小时都没有跑完！！！不知道问题出在那儿，但是还是果断kill掉</span></span><br><span class="line">tr_index=/home/gongyuqi/ref/hg19/transcriptome-index/hg19.<span class="built_in">tr</span></span><br><span class="line">gene_index=~/ref/hg19/index/bowtie/hg19</span><br><span class="line">tophat2 -o ../aligndata/SRR974980.out --transcriptome-index=<span class="variable">$tr</span> -p 30 --phred64-quals <span class="variable">$gene_index</span> SRR974980.fastq.gz</span><br></pre></td></tr></table></figure>

<h5 id="比对方法二"><a href="#比对方法二" class="headerlink" title="比对方法二"></a>比对方法二</h5><ul>
<li>写一个比对的bash脚本文件（align.txt），如下<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#!/bin/bash</span></span><br><span class="line"><span class="built_in">ls</span> *.gz|<span class="keyword">while</span> <span class="built_in">read</span> <span class="built_in">id</span></span><br><span class="line"><span class="keyword">do</span></span><br><span class="line"><span class="built_in">nohup</span> tophat -G ~/ref/hg19/gencode.v36lift37.annotation.gtf -p 30 -o ../aligndata/<span class="variable">$id</span>.out ~/ref/hg19/index/bowtie/hg19 <span class="variable">$id</span> &amp;</span><br><span class="line"><span class="keyword">done</span> </span><br></pre></td></tr></table></figure></li>
<li>执行这个脚本，然后去睡觉，早上醒来看看比对得结果如何~<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line"><span class="built_in">nohup</span> bash align.txt &amp;</span><br></pre></td></tr></table></figure>
运行结果<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/align1.JPG"/></li>
</ul>
<p>每个文件夹打开结果如下<br><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/align2.JPG"/></p>
<p>大概了解一下样本的运行日志</p>
<blockquote>
<p>注意reads的长度——36。miso后续分析中需要知道这个值<br>注意其中有一个error,至于为什么有这个报错，先不追究<br>注意比对结果的那个统计文件align_summary.txt,查看这个文件发现各个样本比对情况都挺好的。这就是为什么这里暂时不追究那个报错<br>最后会统计样本比对时长<br><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/align4.JPG"/></p>
</blockquote>
<p>3、bam文件排序及索引文件建立</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"><span class="built_in">ls</span> *.bam|<span class="keyword">while</span> <span class="built_in">read</span> <span class="built_in">id</span>; <span class="keyword">do</span> samtools <span class="built_in">sort</span> -@ 30 -m 8G -O bam <span class="variable">$id</span> -o <span class="variable">$&#123;id%%.*&#125;</span>.accepted_hits.sorted.bam;<span class="keyword">done</span></span><br><span class="line"><span class="built_in">ls</span> *sorted.bam|<span class="keyword">while</span> <span class="built_in">read</span> <span class="built_in">id</span>; <span class="keyword">do</span> <span class="built_in">nohup</span> samtools index -@ 30 <span class="variable">$id</span> &amp; <span class="keyword">done</span></span><br></pre></td></tr></table></figure>

<h2 id="运行MISO"><a href="#运行MISO" class="headerlink" title="运行MISO"></a>运行MISO</h2><p><strong><table><tr><td bgcolor=GreenYellow><strong>下载hg19的GFF文件并构建索引</strong></td></tr></table></strong></p>
<p>具体下载网页及详情：<a target="_blank" rel="noopener" href="https://miso.readthedocs.io/en/fastmiso/annotation.html">https://miso.readthedocs.io/en/fastmiso/annotation.html</a></p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line"><span class="built_in">cd</span> /home/gongyuqi/project/AS/annotation/human/hg19</span><br><span class="line">wget hollywood.mit.edu/burgelab/miso/annotations/ver2/miso_annotations_hg19_v2.zip</span><br><span class="line">unzip miso_annotations_hg19_v2.zip</span><br></pre></td></tr></table></figure>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/unzipgff.JPG"/>

<p>这里我们以外显子的gff3为例构建索引</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">index_gff --index SE.hg19.gff3 ./index</span><br></pre></td></tr></table></figure>
<p>结果如下<br><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/ESgff3index.JPG"/></p>
<p><strong><table><tr><td bgcolor=GreenYellow><strong>以单端测序样本为例运行miso</strong></td></tr></table></strong></p>
<p>这一步生成的结果是很关键的，会计算alternative splicing events的PSI值。而这个在最后可视化的阶段会以posterior distributions over Ψ(PSI)的形式呈现——即sashimi_plot图最右边一列的柱状图。</p>
<ul>
<li><p>写一个miso_run的脚本，内容如下</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#!/bin/bash</span></span><br><span class="line"><span class="built_in">cd</span> /home/gongyuqi/project/AS/aligndata/accpted_bam</span><br><span class="line">index_db=/home/gongyuqi/project/AS/annotation/human/hg19/index/</span><br><span class="line">out_dir=/home/gongyuqi/project/AS/miso_step1</span><br><span class="line"><span class="built_in">ls</span> *.sorted.bam|<span class="keyword">while</span> <span class="built_in">read</span> <span class="built_in">id</span></span><br><span class="line"><span class="keyword">do</span></span><br><span class="line"><span class="built_in">nohup</span> miso --run <span class="variable">$index_db</span> <span class="variable">$id</span> --output-dir <span class="variable">$out_dir</span>/<span class="variable">$&#123;id%%.*&#125;</span> --read-len 36 &amp;</span><br><span class="line"><span class="keyword">done</span></span><br></pre></td></tr></table></figure>
</li>
<li><p>运行脚本</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line"><span class="built_in">nohup</span> ./miso_run.sh &amp;</span><br></pre></td></tr></table></figure></li>
</ul>
<p>举例查看其中一个样本的运行结果</p>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/miso_run_out.JPG"/>


<table><tr><td bgcolor=GreenYellow>**样本间比较**</td></tr></table>

<p>这个软件最大的问题在于只能进行两个样本的比较，这一点就很气！！！<br>气归气，还是先运行一下，至少先把流程走完，先学起来。</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">compare_miso --compare-samples SRR974978/ SRR974982/ comparison/</span><br></pre></td></tr></table></figure>


<p><strong><table><tr><td bgcolor=GreenYellow><strong>过滤</strong></td></tr></table></strong></p>
<p>设置各种参数的阈值，目的是筛选差异显著的alternative splicing events。</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">filter_events \</span><br><span class="line">--filter  SRR974978_vs_SRR974982.miso_bf \</span><br><span class="line">--num-inc 1 \</span><br><span class="line">--num-exc 1 \</span><br><span class="line">--num-sum-inc-exc 10 \</span><br><span class="line">--delta-psi 0.20 \</span><br><span class="line">--bayes-factor 10 \</span><br><span class="line">--output-dir ./filter_out</span><br></pre></td></tr></table></figure>
<p>过滤前后.misoz_bf文件中的alternative splicing events有明显减少<br><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/filter.JPG"/></p>
<p><strong><table><tr><td bgcolor=GreenYellow><strong>可视化</strong></td></tr></table></strong></p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">index_db=/home/gongyuqi/project/AS/annotation/human/hg19/index/</span><br><span class="line"><span class="comment">#下面做演示的剪切事件是从上一步过滤后的.miso_bf.filtered文件中提取的一个</span></span><br><span class="line">sashimi_plot \</span><br><span class="line">--plot-event <span class="string">&quot;chr20:1093906:1094052:+@chr20:1099396:1099545:+@chr20:1106141:1106293:+&quot;</span> \</span><br><span class="line"><span class="variable">$index_db</span> \</span><br><span class="line">sashimi_plot_settings.txt  \</span><br><span class="line">--output-dir ./sashimi_plot</span><br></pre></td></tr></table></figure>


<p>其中sashimi_plot_settings.txt是配置文件，内容如下。当然，也可以根据自己的作图需要修改相应的参数</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br></pre></td><td class="code"><pre><span class="line">[data]</span><br><span class="line"><span class="comment"># directory where BAM files are</span></span><br><span class="line">bam_prefix = /home/gongyuqi/project/AS/aligndata/accpted_bam/</span><br><span class="line"><span class="comment"># directory where MISO output is</span></span><br><span class="line">miso_prefix = /home/gongyuqi/project/AS/miso_step1</span><br><span class="line"></span><br><span class="line">bam_files = [</span><br><span class="line">    <span class="string">&quot;SRR974978.accepted_hits.sorted.bam&quot;</span>,</span><br><span class="line">    <span class="string">&quot;SRR974984.accepted_hits.sorted.bam&quot;</span>,</span><br><span class="line">    <span class="string">&quot;SRR974968.accepted_hits.sorted.bam&quot;</span>,</span><br><span class="line">    <span class="string">&quot;SRR974972.accepted_hits.sorted.bam&quot;</span>,</span><br><span class="line">    <span class="string">&quot;SRR974982.accepted_hits.sorted.bam&quot;</span>,</span><br><span class="line">    <span class="string">&quot;SRR974980.accepted_hits.sorted.bam&quot;</span>,</span><br><span class="line">    <span class="string">&quot;SRR974974.accepted_hits.sorted.bam&quot;</span>,</span><br><span class="line">    <span class="string">&quot;SRR974976.accepted_hits.sorted.bam&quot;</span>]</span><br><span class="line"></span><br><span class="line">miso_files = [</span><br><span class="line">    <span class="string">&quot;SRR974978&quot;</span>,</span><br><span class="line">    <span class="string">&quot;SRR974984&quot;</span>,</span><br><span class="line">    <span class="string">&quot;SRR974968&quot;</span>,</span><br><span class="line">    <span class="string">&quot;SRR974972&quot;</span>,</span><br><span class="line">    <span class="string">&quot;SRR974982&quot;</span>,</span><br><span class="line">    <span class="string">&quot;SRR974980&quot;</span>,</span><br><span class="line">    <span class="string">&quot;SRR974974&quot;</span>,</span><br><span class="line">    <span class="string">&quot;SRR974976&quot;</span>]</span><br><span class="line"></span><br><span class="line">[plotting]</span><br><span class="line"><span class="comment"># Dimensions of figure to be plotted (in inches)</span></span><br><span class="line">fig_width = 4</span><br><span class="line">fig_height = 10</span><br><span class="line"><span class="comment"># Factor to scale down introns and exons by</span></span><br><span class="line">intron_scale = 30</span><br><span class="line">exon_scale = 4</span><br><span class="line"><span class="comment"># Whether to use a log scale or not when plotting</span></span><br><span class="line">logged = False</span><br><span class="line">font_size = 6</span><br><span class="line"></span><br><span class="line"><span class="comment"># Max y-axis</span></span><br><span class="line">ymax = 150</span><br><span class="line"></span><br><span class="line"><span class="comment"># Whether to plot posterior distributions inferred by MISO</span></span><br><span class="line">show_posteriors = True</span><br><span class="line"></span><br><span class="line"><span class="comment"># Whether to show posterior distributions as bar summaries</span></span><br><span class="line">bar_posteriors = False</span><br><span class="line"></span><br><span class="line"><span class="comment"># Whether to plot the number of reads in each junction</span></span><br><span class="line">number_junctions = True</span><br><span class="line"></span><br><span class="line">resolution = .5</span><br><span class="line">posterior_bins = 40</span><br><span class="line">gene_posterior_ratio = 5</span><br><span class="line"></span><br><span class="line"><span class="comment"># List of colors for read denisites of each sample</span></span><br><span class="line">colors = [<span class="string">&quot;#CC0011&quot;</span>,</span><br><span class="line">          <span class="string">&quot;#CC0011&quot;</span>,</span><br><span class="line">          <span class="string">&quot;#CC0011&quot;</span>,</span><br><span class="line">          <span class="string">&quot;#CC0011&quot;</span>,</span><br><span class="line">          <span class="string">&quot;#FF8800&quot;</span>,</span><br><span class="line">          <span class="string">&quot;#FF8800&quot;</span>,</span><br><span class="line">          <span class="string">&quot;#FF8800&quot;</span>,</span><br><span class="line">          <span class="string">&quot;#FF8800&quot;</span>]</span><br><span class="line"></span><br><span class="line"><span class="comment"># Number of mapped reads in each sample</span></span><br><span class="line"><span class="comment"># (Used to normalize the read density for RPKM calculation)</span></span><br><span class="line">coverages = [27232306,</span><br><span class="line">             16799279,</span><br><span class="line">             31298963,</span><br><span class="line">             26308550,</span><br><span class="line">             27067722,</span><br><span class="line">             17372484,</span><br><span class="line">             26095030,</span><br><span class="line">             26980460]</span><br><span class="line"></span><br><span class="line"><span class="comment"># Bar color for Bayes factor distribution</span></span><br><span class="line"><span class="comment"># plots (--plot-bf-dist)</span></span><br><span class="line"><span class="comment"># Paint them blue</span></span><br><span class="line">bar_color = <span class="string">&quot;b&quot;</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># Bayes factors thresholds to use for --plot-bf-dist</span></span><br><span class="line">bf_thresholds = [0, 1, 2, 5, 10, 20]</span><br></pre></td></tr></table></figure>

<p>可视化结果如下<br>因为我们选取的剪切事件是SRR974978和SRR974982之间的差异剪切事件，所以在多个样本间趋势不是很明显。但这里我们先学个方法。<br><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/sashimi_plot1.JPG"/></p>
</div><div class="article-licensing box"><div class="licensing-title"><p>可变剪切分析之MISO</p><p><a href="http://gyqgyq.com/可变剪切之MISO/">http://gyqgyq.com/可变剪切之MISO/</a></p></div><div class="licensing-meta level is-mobile"><div class="level-left"><div class="level-item is-narrow"><div><h6>Author</h6><p>Gong Yuqi</p></div></div><div class="level-item is-narrow"><div><h6>Posted on</h6><p>2021-01-09</p></div></div><div class="level-item is-narrow"><div><h6>Updated on</h6><p>2022-11-26</p></div></div><div class="level-item is-narrow"><div><h6>Licensed under</h6><p><a class="icons" rel="noopener" target="_blank" title="Creative Commons" href="https://creativecommons.org/"><i class="icon fab fa-creative-commons"></i></a><a class="icons" rel="noopener" target="_blank" title="Attribution" href="https://creativecommons.org/licenses/by/4.0/"><i class="icon fab fa-creative-commons-by"></i></a><a class="icons" rel="noopener" target="_blank" title="Noncommercial" href="https://creativecommons.org/licenses/by-nc/4.0/"><i class="icon fab fa-creative-commons-nc"></i></a></p></div></div></div></div></div><div class="article-tags is-size-7 mb-4"><span class="mr-2">#</span><a class="link-muted mr-2" rel="tag" href="/tags/NGS/">NGS</a></div><div class="a2a_kit a2a_kit_size_32 a2a_default_style"><a class="a2a_dd" target="_blank" rel="noopener" href="https://www.addtoany.com/share"></a><a class="a2a_button_facebook"></a><a class="a2a_button_twitter"></a><a class="a2a_button_telegram"></a><a class="a2a_button_whatsapp"></a><a class="a2a_button_reddit"></a></div><script src="https://static.addtoany.com/menu/page.js" defer></script></article></div><div class="card"><div class="card-content"><h3 class="menu-label has-text-centered">Like this article? Support the author with</h3><div class="buttons is-centered"><a class="button donate" data-type="alipay"><span class="icon is-small"><i class="fab fa-alipay"></i></span><span>Alipay</span><span class="qrcode"><img src="/" alt="Alipay"></span></a><a class="button donate" data-type="wechat"><span class="icon is-small"><i class="fab fa-weixin"></i></span><span>Wechat</span><span class="qrcode"><img src="/" alt="Wechat"></span></a></div></div></div><nav class="post-navigation mt-4 level is-mobile"><div class="level-start"><a class="article-nav-prev level level-item link-muted" href="/%E5%8F%AF%E5%8F%98%E5%89%AA%E5%88%87%E4%B9%8BMART/"><i class="level-item fas fa-chevron-left"></i><span class="level-item">可变剪切之rMART</span></a></div><div class="level-end"><a class="article-nav-next level level-item link-muted" href="/2020%20ATAC-seq%20analysis/"><span class="level-item">2020 ATAC-seq analysis</span><i class="level-item fas fa-chevron-right"></i></a></div></nav><!--!--></div><div class="column column-left is-4-tablet is-4-desktop is-4-widescreen  order-1"><div class="card widget" data-type="profile"><div class="card-content"><nav class="level"><div class="level-item has-text-centered flex-shrink-1"><div><figure class="image is-128x128 mx-auto mb-2"><img class="avatar" src="/img/ava.jpg" alt="Gong Yuqi"></figure><p class="title is-size-4 is-block" style="line-height:inherit;">Gong Yuqi</p><p class="is-size-6 is-flex justify-content-center"><i class="fas fa-map-marker-alt mr-1"></i><span>HangZhou, China</span></p></div></div></nav><nav class="level is-mobile"><div class="level-item has-text-centered is-marginless"><div><p class="heading">Posts</p><a href="/archives"><p class="title">37</p></a></div></div><div class="level-item has-text-centered is-marginless"><div><p class="heading">Category</p><a href="/categories"><p class="title">1</p></a></div></div><div class="level-item has-text-centered is-marginless"><div><p class="heading">Tags</p><a href="/tags"><p class="title">7</p></a></div></div></nav><div class="level"><a class="level-item button is-primary is-rounded" href="https://github.com/gongyuqiqi" target="_blank" rel="noopener">Follow</a></div><div class="level is-mobile is-multiline"><a class="level-item button is-transparent is-marginless" target="_blank" rel="noopener" title="Github" href="https://github.com/gongyuqiqi"><i class="fab fa-github"></i></a><a class="level-item button is-transparent is-marginless" target="_blank" rel="noopener" title="Facebook" href="https://facebook.com"><i class="fab fa-facebook"></i></a><a class="level-item button is-transparent is-marginless" target="_blank" rel="noopener" title="Twitter" href="https://twitter.com"><i class="fab fa-twitter"></i></a><a class="level-item button is-transparent is-marginless" target="_blank" rel="noopener" title="Dribbble" href="https://dribbble.com"><i class="fab fa-dribbble"></i></a><a class="level-item button is-transparent is-marginless" target="_blank" rel="noopener" title="RSS" href="/"><i class="fas fa-rss"></i></a></div></div></div><div class="card widget" id="toc" data-type="toc"><div class="card-content"><div class="menu"><h3 class="menu-label">Catalogue</h3><ul class="menu-list"><li><a class="level is-mobile" href="#可变剪切分析之MISO"><span class="level-left"><span class="level-item">1</span><span class="level-item">可变剪切分析之MISO</span></span></a><ul class="menu-list"><li><a class="level is-mobile" href="#数据下载"><span class="level-left"><span class="level-item">1.1</span><span class="level-item">数据下载</span></span></a></li><li><a class="level is-mobile" href="#比对"><span class="level-left"><span class="level-item">1.2</span><span class="level-item">比对</span></span></a><ul class="menu-list"><li><a class="level is-mobile" href="#比对方法二"><span class="level-left"><span class="level-item">1.2.1</span><span class="level-item">比对方法二</span></span></a></li></ul></li><li><a class="level is-mobile" href="#运行MISO"><span class="level-left"><span class="level-item">1.3</span><span class="level-item">运行MISO</span></span></a></li></ul></li></ul></div></div><style>#toc .menu-list > li > a.is-active + .menu-list { display: block; }#toc .menu-list > li > a + .menu-list { display: none; }</style><script src="/js/toc.js" defer></script></div><div class="card widget" data-type="categories"><div class="card-content"><div class="menu"><h3 class="menu-label">Categories</h3><ul class="menu-list"><li><a class="level is-mobile" href="/categories/Programming-Language/"><span class="level-start"><span class="level-item">Programming Language</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></div></div></div><div class="card widget" data-type="recent-posts"><div class="card-content"><h3 class="menu-label">Recents</h3><article class="media"><div class="media-content"><p class="date"><time dateTime="2021-07-04T05:48:30.000Z">2021-07-04</time></p><p class="title"><a href="/RNA-TNBC/">RNA-TNBC</a></p></div></article><article class="media"><div class="media-content"><p class="date"><time dateTime="2021-06-29T06:01:05.000Z">2021-06-29</time></p><p class="title"><a href="/scRNA-TNBC/">scRNA-TNBC</a></p></div></article><article class="media"><div class="media-content"><p class="date"><time dateTime="2021-06-03T04:43:57.000Z">2021-06-03</time></p><p class="title"><a href="/MeRIP-Seq/">MeRIP-Seq</a></p></div></article><article class="media"><div class="media-content"><p class="date"><time dateTime="2021-05-14T05:37:00.000Z">2021-05-14</time></p><p class="title"><a href="/WES/">WES</a></p></div></article><article class="media"><div class="media-content"><p class="date"><time dateTime="2021-05-13T06:08:18.000Z">2021-05-13</time></p><p class="title"><a href="/scRNA-seq-%E6%95%B0%E6%8D%AE%E5%A4%8D%E7%8E%B0/">scRNA-seq_数据复现</a></p></div></article></div></div><div class="card widget" data-type="archives"><div class="card-content"><div class="menu"><h3 class="menu-label">Archives</h3><ul class="menu-list"><li><a class="level is-mobile" href="/archives/2021/07/"><span class="level-start"><span class="level-item">July 2021</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/archives/2021/06/"><span class="level-start"><span class="level-item">June 2021</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/archives/2021/05/"><span class="level-start"><span class="level-item">May 2021</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li><li><a class="level is-mobile" href="/archives/2021/03/"><span class="level-start"><span class="level-item">March 2021</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/archives/2021/02/"><span class="level-start"><span class="level-item">February 2021</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/archives/2021/01/"><span class="level-start"><span class="level-item">January 2021</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li><li><a class="level is-mobile" href="/archives/2020/12/"><span class="level-start"><span class="level-item">December 2020</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/archives/2020/05/"><span class="level-start"><span class="level-item">May 2020</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/archives/2020/04/"><span class="level-start"><span class="level-item">April 2020</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/archives/2020/03/"><span class="level-start"><span class="level-item">March 2020</span></span><span class="level-end"><span class="level-item tag">5</span></span></a></li><li><a class="level is-mobile" href="/archives/2020/02/"><span class="level-start"><span class="level-item">February 2020</span></span><span class="level-end"><span class="level-item tag">18</span></span></a></li></ul></div></div></div><div class="card widget" data-type="tags"><div class="card-content"><div class="menu"><h3 class="menu-label">Tags</h3><div class="field is-grouped is-grouped-multiline"><div class="control"><a class="tags has-addons" href="/tags/Linux/"><span class="tag">Linux</span><span class="tag">7</span></a></div><div class="control"><a class="tags has-addons" href="/tags/MySQL/"><span class="tag">MySQL</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/NGS/"><span class="tag">NGS</span><span class="tag">18</span></a></div><div class="control"><a class="tags has-addons" href="/tags/R/"><span class="tag">R</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/System/"><span class="tag">System</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/notes/"><span class="tag">notes</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/tag1/"><span class="tag">tag1</span><span class="tag">6</span></a></div></div></div></div></div><div class="card widget" data-type="subscribe-email"><div class="card-content"><div class="menu"><h3 class="menu-label">follow.it</h3><form action="" method="post" target="_blank"><div class="field has-addons"><div class="control has-icons-left is-expanded"><input class="input" name="email" type="email" placeholder="Email"><span class="icon is-small is-left"><i class="fas fa-envelope"></i></span></div><div class="control"><input class="button" type="submit" value="Subscribe"></div></div></form></div></div></div></div><!--!--></div></div></section><footer class="footer"><div class="container"><div class="level"><div class="level-start"><a class="footer-logo is-block mb-2" href="/"><img src="/img/ava.jpg" alt="Gong Yuqi&#039;s blog" height="28"></a><p class="is-size-7"><span>&copy; 2022 Gong Yuqi</span>  Powered by Gong Yuqi <i class="fa-heart fa fa-beat" style="color: #f00;"></i><br><span id="busuanzi_container_site_uv">Visited by <span id="busuanzi_value_site_uv">0</span> users</span></p></div><div class="level-end"><div class="field has-addons"><p class="control"><a class="button is-transparent is-large" target="_blank" rel="noopener" title="Creative Commons" href="https://creativecommons.org/"><i class="fab fa-creative-commons"></i></a></p><p class="control"><a class="button is-transparent is-large" target="_blank" rel="noopener" title="Attribution 4.0 International" href="https://creativecommons.org/licenses/by/4.0/"><i class="fab fa-creative-commons-by"></i></a></p><p class="control"><a class="button is-transparent is-large" target="_blank" rel="noopener" title="Download on GitHub" href="https://github.com/gongyuqiqi/gongyuqiqi.github.io"><i class="fab fa-github"></i></a></p></div></div></div></div></footer><script src="https://cdn.jsdelivr.net/npm/jquery@3.3.1/dist/jquery.min.js"></script><script src="https://cdn.jsdelivr.net/npm/moment@2.22.2/min/moment-with-locales.min.js"></script><script src="https://cdn.jsdelivr.net/npm/clipboard@2.0.4/dist/clipboard.min.js" defer></script><script>moment.locale("en");</script><script>var IcarusThemeSettings = {
            article: {
                highlight: {
                    clipboard: true,
                    fold: 'unfolded'
                }
            }
        };</script><script src="/js/column.js"></script><script src="/js/animation.js"></script><a id="back-to-top" title="Back to top" href="javascript:;"><i class="fas fa-chevron-up"></i></a><script src="/js/back_to_top.js" defer></script><!--!--><!--!--><!--!--><!--!--><script src="https://cdn.jsdelivr.net/npm/cookieconsent@3.1.1/build/cookieconsent.min.js" defer></script><script>window.addEventListener("load", () => {
      window.cookieconsent.initialise({
        type: "info",
        theme: "edgeless",
        static: false,
        position: "bottom-left",
        content: {
          message: "This website uses cookies to improve your experience.",
          dismiss: "Got it!",
          allow: "Allow cookies",
          deny: "Decline",
          link: "Learn more",
          policy: "Cookie Policy",
          href: "https://www.cookiesandyou.com/",
        },
        palette: {
          popup: {
            background: "#edeff5",
            text: "#838391"
          },
          button: {
            background: "#4b81e8"
          },
        },
      });
    });</script><script src="https://cdn.jsdelivr.net/npm/lightgallery@1.10.0/dist/js/lightgallery.min.js" defer></script><script src="https://cdn.jsdelivr.net/npm/justifiedGallery@3.8.1/dist/js/jquery.justifiedGallery.min.js" defer></script><script>window.addEventListener("load", () => {
            if (typeof $.fn.lightGallery === 'function') {
                $('.article').lightGallery({ selector: '.gallery-item' });
            }
            if (typeof $.fn.justifiedGallery === 'function') {
                if ($('.justified-gallery > p > .gallery-item').length) {
                    $('.justified-gallery > p > .gallery-item').unwrap();
                }
                $('.justified-gallery').justifiedGallery();
            }
        });</script><!--!--><!--!--><!--!--><!--!--><!--!--><script src="/js/main.js" defer></script><div class="searchbox"><div class="searchbox-container"><div class="searchbox-header"><div class="searchbox-input-container"><input class="searchbox-input" type="text" placeholder="Type something..."></div><a class="searchbox-close" href="javascript:;">×</a></div><div class="searchbox-body"></div></div></div><script src="/js/insight.js" defer></script><script>document.addEventListener('DOMContentLoaded', function () {
            loadInsight({"contentUrl":"/content.json"}, {"hint":"Type something...","untitled":"(Untitled)","posts":"Posts","pages":"Pages","categories":"Categories","tags":"Tags"});
        });</script></body></html>