---
title: å¯å˜å‰ªåˆ‡ä¹‹rMART
date: 2021-01-14 11:05:01
tags: NGS
---

RNA-seqé«˜çº§åˆ†æä¹‹å¯å˜å‰ªåˆ‡

<!--more-->

# å¯å˜å‰ªåˆ‡ä¹‹rMART

**å®˜æ–¹å‚è€ƒèµ„æ–™**  
http://rnaseq-mats.sourceforge.net/   
**rMATS**:https://github.com/Xinglab/rmats-turbo/blob/v4.1.0/README.md  
**rmats2sashimiplot**:https://github.com/Xinglab/rmats2sashimiplot

## (ä¸€)rMATSè½¯ä»¶åŠç›¸å…³ä¾èµ–åŒ…çš„ä¸‹è½½

```bash
#å…ˆç”¨condaä¸‹è½½rMATsè½¯ä»¶éœ€è¦çš„ä¾èµ–åŒ…
#æœ‰æ—¶å€™pythonç‰ˆæœ¬é—®é¢˜æå¾—æˆ‘å¾ˆçƒ¦ï¼Œæœ‰äº›ç”Ÿä¿¡è½¯ä»¶ä¾èµ–ç‰¹å®šçš„pythonç‰ˆæœ¬ï¼Œæ‰€ä»¥åœ¨è½¯ä»¶å®‰è£…è¿‡ç¨‹ä¸­å¤šåŠ æ³¨æ„è¿™ä¸ªé—®é¢˜
conda create rMATs2.7 python=2.7
conda activate rMATs2.7
conda install -y Cython=0.29.14 numpy=1.16.5 blas lapack gsl=2.6 gcc=5.4.0 CMake=3.15.4
conda install libgfortran==1
deactivate rMATs2.7

#ä¸‹è½½è½¯ä»¶å¹¶è§£å‹rMATsè½¯ä»¶
#rMATsä¸åŒç‰ˆæœ¬ä¸‹è½½ç½‘å€ï¼šhttps://github.com/Xinglab/rmats-turbo/releases/tag/v4.1.0
cd /home/gongyuqi/project/AS/rMATS
wget https://github.com/Xinglab/rmats-turbo/releases/download/v4.1.0/rmats_turbo_v4_1_0_python_2_7.tar.gz
tar -zxvf rmats_turbo_v4_1_0_python_2_7.tar.gz
#æ°¸ä¹…æ·»åŠ è·¯å¾„åˆ°ç¯å¢ƒå˜é‡ä¸­
export PATH=/home/gongyuqi/project/AS/rMATS/rmats-turbo/:$PATH
source ~/.bashrc
```


## (äºŒ)STARæ¯”å¯¹è½¯ä»¶çš„ä½¿ç”¨

è¿™ä¹ˆå¤šæ¯”å¯¹è½¯ä»¶ï¼Œä¸ºä»€ä¹ˆé€‰STARï¼Ÿ  
å› ä¸ºå®˜ç½‘ç”¨çš„STARï¼æ®è¯´å…¶ä»–å¯ä»¥åšå¯å˜å‰ªåˆ‡çš„æ¯”å¯¹è½¯ä»¶ç”Ÿæˆçš„bamæ–‡ä»¶ï¼ŒrMATsè¿è¡Œä¸äº†~~~

**1ã€æ„å»ºå‚è€ƒåŸºå› ç»„ç´¢å¼•**

```bash
#æ¿€æ´»æ¯”å¯¹çš„ç¯å¢ƒ,è¿™ä¸ªç¯å¢ƒé‡Œé¢æˆ‘ä¸‹è½½äº†å„ç§æ¯”å¯¹è½¯ä»¶ï¼Œpythonç‰ˆæœ¬ä¸º2.7.15.
conda activate alignment2.7 
nohup STAR --runThreadN 30 --runMode genomeGenerate \
--genomeDir /home/gongyuqi/ref/hg38/index/star \
--genomeFastaFiles /home/gongyuqi/ref/hg38/Homo_sapiens.GRCh38.dna.primary_assembly.fa \
--sjdbGTFfile /home/gongyuqi/ref/hg38/Homo_sapiens.GRCh38.102.gtf &
```

**2ã€ç”¨STARè½¯ä»¶è¿›è¡Œæ¯”å¯¹**

STARçš„ä½¿ç”¨ä»¥åŠè¾“å‡ºæ–‡ä»¶è§£è¯»å‚è€ƒèµ„æ–™ï¼š  
https://blog.csdn.net/yssxswl/article/details/105703869  
https://www.jianshu.com/p/eca16bf2824e

```bash
#è¿›å…¥fastqæ–‡ä»¶ååœ¨ç›®å½•ï¼Œè¿™é‡Œæ˜¯åŸå§‹çš„fastqæ•°æ®ï¼Œè´¨æ§ç»“æœæŒºå¥½çš„ï¼Œä¸ç”¨å†è¿‡æ»¤äº†ï¼Œæ‰€ä»¥æˆ‘ç›´æ¥æ¯”å¯¹äº†
cd /home/gongyuqi/project/AS/rawdata
#å†™å¾ªç¯è¿›è¡Œæ¯”å¯¹
ls *.gz|while read id
do
nohup STAR --runThreadN 20 \
--genomeDir /home/gongyuqi/ref/hg38/index/star \
--readFilesCommand gunzip -c \
--readFilesIn ./$id \
--outSAMtype BAM SortedByCoordinate \
--outFileNamePrefix /home/gongyuqi/project/AS/aligndata_star/${id%%.*}. &
done
```

**3ã€æ¯”å¯¹æƒ…å†µç»Ÿè®¡å¹¶æŸ¥çœ‹**

(1)ç›´æ¥æŸ¥çœ‹
```bash
ls *.Log.progress.out|while read id;do sed -n '1,3p' $id;done
```
ç»“æœå¦‚ä¸‹ï¼Œæ¯”å¯¹æƒ…å†µè¿˜æ˜¯ä¸é”™çš„ï¼Œå¯ä»¥è¿›è¡Œåç»­çš„åˆ†æã€‚
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/star_align_Log.progress.out.JPG"/>

(2)flagstatç»Ÿè®¡æ¯”å¯¹æƒ…å†µæŸ¥çœ‹æ¯”å¯¹æƒ…å†µ
```bash
ls *.bam|while read id
do
nohup samtools flagstat -@ 20 $id > ${id%%.*}.flagstat.txt &
done
ls *.txt|while read id;do cat $id | grep "mapped (";done
```
è¿™ä¸ªæ¯”å¯¹ç»“æœå®åœ¨æ˜¯éš¾ä»¥ç½®ä¿¡ï¼ï¼ä½†æ˜¯ç»“åˆä¸Šé¢çš„æ¯”å¯¹ç»“æœï¼Œè¿˜æ˜¯å¯ä»¥ç›¸ä¿¡æ¯”å¯¹è¿™ä¸€æ­¥æ˜¯æ²¡æœ‰é—®é¢˜çš„ã€‚  
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/star_align_flagstat.JPG"/>



## (ä¸‰)è¿è¡ŒrMATs

```bash
#é¦–å…ˆæˆ‘ä»¬è¦å¤±æ´»æ‰æ¯”å¯¹çš„ç¯å¢ƒï¼ˆalignment2.7ï¼‰ï¼Œæ¿€æ´»rMATsç¯å¢ƒ
conda deactivate alignment2.7
conda activate rMATs2.7
#è™½ç„¶rMATSè·¯å¾„è¢«æ°¸ä¹…æ·»åŠ åˆ°ç¯å¢ƒå˜é‡ä¸­ï¼Œä½†æ˜¯PATHåªè®¤å¯æ‰§è¡Œæ–‡ä»¶ï¼Œæ‰€ä»¥è¿™ä¸€æ­¥è¿˜æ˜¯è¦åˆ‡æ¢åˆ°rMATsè·¯å¾„ä¸‹
cd /home/gongyuqi/project/AS/rMATS/rmats-turbo
input_dir=/home/gongyuqi/project/AS/aligndata_star
python rmats.py \
--b1 $input_dir/b1.txt \
--b2 $input_dir/b2.txt \
--gtf /home/gongyuqi/ref/hg38/ensembl/Homo_sapiens.GRCh38.102.gtf \
-t single --readLength 36 --nthread 20 \
--od /home/gongyuqi/project/AS/rMATS/rmats_output \
--tmp /home/gongyuqi/project/AS/rMATS/rmats_tmp
#è¿è¡Œå®Œè®°å¾—é€€å‡ºå½“å‰ç¯å¢ƒï¼Œå› ä¸ºåé¢æˆ‘ä»¬è¦ç”¨å¦å¤–ä¸€ä¸ªç¯å¢ƒ
conda deactivate rMATs2.7
```

è¿è¡Œä¸Šé¢çš„ä»£ç ï¼Œä¼šæŠ¥å¦‚ä¸‹é”™è¯¯ï¼š
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/erro_libgsl.so.25.JPG"/>

æˆ‘æŸ¥çœ‹äº†ä¸€ä¸‹ï¼Œ/home/gongyuqi/miniconda3/envs/rMATs2.7/libè·¯å¾„ä¸‹æœ‰è¿™ä¸ªæ–‡ä»¶å‘¢ï¼ä¸ºä»€ä¹ˆç”µè„‘æ‰¾ä¸åˆ°ï¼Ÿè§¦åŠåˆ°äº†æˆ‘çš„çŸ¥è¯†ç›²åŒºäº†~~~  
æŸ¥äº†ä¸€ä¸‹åˆçš„èµ„æ–™ï¼Œå‘ç°è¿™ä¸ªæŠ¥é”™æ¶‰åŠåˆ°æŸ¥æ‰¾å…±äº«åº“ï¼ˆåŠ¨æ€é“¾æ¥åº“ï¼‰é—®é¢˜ï¼Œæ ¹æ®æŸ¥åˆ°çš„ä¸œè¥¿æ“ä½œäº†ä¸€æ³¢å±…ç„¶æ²¡æœ‰è§£å†³ï¼Œæ—è¾¹çš„é›¶é£Ÿéƒ½ä¸é¦™äº†ï¼ğŸ˜«   
ç®—äº†~~~æˆ‘è¿˜æ˜¯å…ˆå»åƒæ™šé¥­å§ï¼Œå·²ç»é¥¿å¾—å¾ˆç˜¦äº†ï¼ğŸ˜«   
ã€‚ã€‚ã€‚ã€‚ã€‚ã€‚   
å¥½å•¦ï¼Œæˆ‘åƒå®Œé¥­å›æ¥äº†ã€‚æˆ‘ç»§ç»­å€’è…¾ï¼Œä¸æ”¾å¼ƒã€‚   
ã€‚ã€‚ã€‚ã€‚ã€‚ã€‚  
é¢ï¼Œæˆ‘çŸ¥é“äº†ï¼Œè¿™æ®µä»£ç è¿è¡Œè¿‡ç¨‹ä¸­è¿˜æ¶‰åŠåˆ°gccç¼–è¯‘ï¼Œå› ä¸ºå®ƒä¾èµ–åŒ…ä¸­æœ‰GCCã€‚  
è¿™ä¸¤ä¸ªé—®é¢˜å®åœ¨æ˜¯å¤ªä¸“ä¸šäº†ï¼Œä»Šå¤©åˆå­¦ä¹ åˆ°äº†ã€‚ç°åœ¨æœ‰ä¸€ç‚¹å°å°çš„å¼€å¿ƒï¼ğŸ˜Š  
è§£å†³æ–¹æ³•å‚è€ƒèµ„æ–™ï¼šhttps://www.jianshu.com/p/a62e1d327023
```bash
#Linuxé»˜è®¤åªä¼šåœ¨æŒ‡å®šçš„å‡ ä¸ªç›®å½•æ‰¾å…±äº«åº“ï¼ˆåŠ¨æ€é“¾æ¥åº“ï¼‰ï¼Œå…¶ä»–ç›®å½•çš„è¦è‡ªå·±åŠ åœ¨LD_LIBRARY_PATH
export LD_LIBRARY_PATH=/home/gongyuqi/miniconda3/envs/rMATs2.7/lib
#å› ä¸ºä¸Šè¿°å‘½ä»¤æ˜¯ä¾èµ–gccçš„ï¼Œéœ€è¦gccè¿›è¡Œç¼–è¯‘ï¼Œæ‰€ä»¥è¿˜éœ€è¦ä¸‹é¢çš„æ“ä½œ
export LIBRARY_PATH=/home/gongyuqi/miniconda3/envs/rMATs2.7/lib
#æ³¨æ„ï¼Œä¸Šè¿°éƒ½æ˜¯ä¸´æ—¶çš„ï¼Œæ²»æ ‡ä¸æ²»æœ¬ï¼Œä¸‹æ¬¡æ‰§è¡Œè¿è¡ŒrMATsæ—¶ï¼Œä¾ç„¶è¦é‡å¤ä¸Šè¿°æ“ä½œã€‚å½“ç„¶ä½ ä¹Ÿå¯ä»¥å°†å…¶æ°¸ä¹…æ€§æ·»åŠ ã€‚
```
è¿è¡Œç»“æœå¦‚ä¸‹ï¼Œå…³äºè¿™äº›ç»“æœæ–‡ä»¶çš„è§£é‡Šè¯´æ˜ï¼Œå¯ä»¥å‚è€ƒrMATSå®˜ç½‘ï¼Œä¸Šé¢æœ‰å¾ˆè¯¦ç»†çš„é˜è¿°ã€‚
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/rMATs_output.JPG"/>




## (å››)rmats2sashimiplotå¯è§†åŒ–

ç»ˆäºåˆ°äº†æ¿€åŠ¨äººå¿ƒçš„å¯è§†åŒ–é˜¶æ®µäº†ï¼Œä¸€é¡¿æ“ä½œä¹‹åï¼Œäº§ç”Ÿä¸€å¼ æ¼‚äº®çš„å›¾æ”¾PPTä¸Šè¿˜æ˜¯å¾ˆæ¬£æ…°çš„ã€‚

**1ã€ç¯å¢ƒæ­å»º**
```bash
conda create -n rmats2sashimiplot python=2.7 
conda activate rmats2sashimiplot
conda install -y numpy scipy matplotlib pysam samtools bedtools
```

**2ã€å‡†å¤‡å¯è§†åŒ–çš„txtæ–‡ä»¶**  

ä»¥Exon Skippingä¸ºä¾‹æ¼”ç¤º,è¿™é‡Œæˆ‘éšä¾¿æ‰¾ä¸€ä¸ªåŸºå› è¿›è¡Œå¯è§†åŒ–ï¼Œæ³¨æ„äº†ï¼Œè¿™ä¸ªåŸºå› ä¸ä¸€å®šæ˜¯å·®å¼‚å¯å˜å‰ªåˆ‡çš„åŸºå› å“Ÿï¼Œå› ä¸ºæ˜¯æˆ‘éšä¾¿æ‰¾çš„ä¸€ä¸ªã€‚å˜»å˜»ğŸ¤­
```bash
sed -n '2p' SE.MATS.JC.txt > SE.plot.PRMT2.txt
cat SE.plot.PRMT2.txt
```
æ–‡ä»¶å†…å®¹å¦‚ä¸‹ï¼š
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/PRMT2_rMATs.txt.JPG"/>


**3ã€å¯è§†åŒ–çš„txtæ–‡ä»¶**

###### ä¸€å¼€å§‹ï¼Œæˆ‘éå¸¸è‡ªä¿¡çš„è¿è¡Œäº†ä¸‹é¢çš„ä»£ç ~
```bash
gff3=/home/gongyuqi/ref/hg38/ensembl/Homo_sapiens.GRCh38.102.gff3
output_dir=/home/gongyuqi/project/AS/rMATS/rmats2sashimiplot/
rmats2sashimiplot \
--b1 SRR974968.Aligned.sortedByCoord.out.bam,SRR974972.Aligned.sortedByCoord.out.bam,SRR974978.Aligned.sortedByCoord.out.bam,SRR974984.Aligned.sortedByCoord.out.bam \
--b2 SRR974974.Aligned.sortedByCoord.out.bam,SRR974976.Aligned.sortedByCoord.out.bam,SRR974980.Aligned.sortedByCoord.out.bam,SRR974982.Aligned.sortedByCoord.out.bam \
-c chr21:+:46636438:46636547:$gff3 \
-e /home/gongyuqi/project/AS/rMATS/rmats_output/SE.plot.PRMT2.txt \
--l1 treatment --l2 control \
--exon_s 1 --intron_s 5 -t SE \
-o $output_dir/test
```
###### ä½†æ˜¯ï¼Œæˆ‘é‡åˆ°äº†å¦‚ä¸‹æŠ¥é”™
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/rmats2sashimiplot_chr_error.JPG"/>


###### æˆ‘å›å¤´çœ‹äº†ä¸€ä¸‹æˆ‘çš„bamæ–‡ä»¶ï¼ŒæŸ“è‰²ä½“æ˜¯ä»¥1,2...X,Yå½¢å¼ä½“ç°è€Œä¸æ˜¯chr1,chr2...chrX,chrYå½¢å¼ä½“ç°ã€‚æˆ‘ç¬¬ä¸€æ¬¡é‡åˆ°æŸ“è‰²ä½“å·çš„é—®é¢˜å‘¢ã€‚æˆ‘è¦æ€ä¹ˆè®©bamæ–‡ä»¶ä¸­æœ‰`chr`å­˜åœ¨ä¸”è®©å…¶å­˜åœ¨åœ¨æ­£ç¡®çš„ä½ç½®å‘¢ã€‚  
###### æˆ‘ç»å°½è„‘æ±çš„æƒ³äº†å¾ˆä¹…å¾ˆä¹…ğŸ˜«ï¼Œæˆ‘è¿˜è€ƒè™‘åˆ°æ˜¯ä¸æ˜¯æˆ‘å‚è€ƒåŸºå› ç»„çš„é—®é¢˜ï¼Œä¸åŒæ•°æ®åº“å‚è€ƒåŸºå› ç»„æ˜¯ä¸æ˜¯æœ‰å·®å¼‚ï¼Œæˆ‘è¦ä¸è¦æ¢ä¸€ä¸ªæ•°æ®åº“çš„å‚è€ƒåŸºå› ç»„ã€‚ç„¶åæˆ‘ä¸€é¡¿æ“ä½œï¼Œæ— æœ~~~ç®—äº†ï¼Œæˆ‘æ‰“ä¸€å±€æ¸¸æˆç¡è§‰å§ï¼Œæ˜å¤©å†æƒ³ã€‚
###### èµ·åºŠ~åƒæ—©é¥­~éª‘è½¦å»å®éªŒå®¤~æ€è€ƒ~æŸ¥èµ„æ–™~æœ‰ä¸€ç‚¹å°æƒ³æ³•äº†~æˆ‘å®è·µä¸€ä¸‹çœ‹çœ‹å¯ä¸å¯
å‚è€ƒèµ„æ–™ï¼šhttps://www.jianshu.com/p/5ceda0350d0d


###### <font color=blue>ç¬¬ä¸€æ­¥ï¼šç»™bamæ–‡ä»¶æ·»åŠ chr</font>
```bash
#æ·»åŠ chr
ls *.bam|while read id
do
nohup samtools view -h $id | sed -e '/^@SQ/s/SN\:/SN\:chr/' -e '/^[^@]/s/\t/\tchr/2'|awk -F ' ' '$7=($7=="=" || $7=="*"?$7:sprintf("chr%s",$7))' |tr " " "\t" | samtools view -h -b -@ 10 -S - > ${id%%.*}.Aligned.sortedByCoord.out.chr.bam &
done
```

###### <font color=blue>ç¬¬äºŒæ­¥ï¼šä½¿ç”¨æ·»åŠ chrçš„bamæ–‡ä»¶æ‰§è¡Œç¨‹åº</font>
```bash
gff3=/home/gongyuqi/ref/hg38/ensembl/Homo_sapiens.GRCh38.102.gff3
output_dir=/home/gongyuqi/project/AS/rMATS/rmats2sashimiplot/
rmats2sashimiplot \
--b1 SRR974968.Aligned.sortedByCoord.out.chr.bam,SRR974972.Aligned.sortedByCoord.out.chr.bam,SRR974978.Aligned.sortedByCoord.out.chr.bam,SRR974984.Aligned.sortedByCoord.out.chr.bam \
--b2 SRR974974.Aligned.sortedByCoord.out.chr.bam,SRR974976.Aligned.sortedByCoord.out.chr.bam,SRR974980.Aligned.sortedByCoord.out.chr.bam,SRR974982.Aligned.sortedByCoord.out.chr.bam \
-c chr21:+:46636438:46636547:$gff3 \
-e /home/gongyuqi/project/AS/rMATS/rmats_output/SE.plot.PRMT2.txt \
--l1 treatment --l2 control \
--exon_s 1 --intron_s 5 -t SE \
-o $output_dir/SE.plot
```

å¯è§†åŒ–ç»“æœå¦‚ä¸‹ï¼ˆä½ ä¼šå‘ç°ï¼Œå’¦~ä¸¤ç»„é—´å·®å¼‚ä¸æ˜¾è‘—å‘€ï¼Ÿå› ä¸ºæˆ‘éšä¾¿æŒ‘äº†ä¸€ä¸ªåŸºå› å¯è§†åŒ–å‘€ï¼æˆ‘æœ‰å·å·å¯è§†åŒ–æŸä¸ªåŸºå› ï¼Œå¯è§†åŒ–å‡ºæ¥çš„å·®å¼‚æ€§ä¸æˆ‘å®šé‡å’ŒåŠå®šé‡qPCRçš„ç»“æœä¸€è‡´å“Ÿï¼Œæ‰€ä»¥ä¸Šè¿°åˆ†ææµç¨‹åº”è¯¥æ˜¯OKçš„~~~å˜»å˜»å˜»ğŸ¤­ä¸ç®¡æ€æ ·ï¼Œè¿™ä¸ªåŸºå› ç¡®å®å­˜åœ¨Exon Skippingã€‚ï¼‰
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/ABC_GCB_PRMT2_rMATs.JPG"/>


### æ€»ç»“

**æŠ€æœ¯ä¸Š**  
1ã€éœ€è¦æ­å»ºä¸¤ä¸ªé‡è¦çš„ç¯å¢ƒï¼šè¿è¡ŒrMATsçš„ç¯å¢ƒï¼Œè¿è¡Œrmats2sashimiplotçš„ç¯å¢ƒã€‚  
2ã€é€‰ç”¨STARè¿›è¡Œå‚è€ƒåŸºå› ç»„çš„æ¯”å¯¹ï¼ˆæ³¨æ„äº†ï¼Œå‚è€ƒåŸºå› ç»„å’Œæ³¨é‡Šæ–‡ä»¶çš„ç‰ˆæœ¬è¦ä¸€è‡´å“Ÿï¼‰ã€‚  
3ã€åœ¨rMATsçš„ç¯å¢ƒä¸‹,ç»“åˆä¸Šè¿°ç”Ÿæˆçš„bamæ–‡ä»¶ï¼Œè¿è¡ŒrMATsè½¯ä»¶ï¼Œç”Ÿæˆå¯å˜å‰ªåˆ‡äº‹ä»¶çš„æ–‡ä»¶ã€‚  
4ã€åœ¨rmats2sashimiplotç¯å¢ƒä¸‹ï¼Œå¯è§†åŒ–ä¸Šè¿°ç”Ÿæˆçš„å¯å˜å‰ªåˆ‡äº‹ä»¶ã€‚

**å¿ƒæ€ä¸Š**  
ä¸Šåˆè§£å†³ä¸äº†å°±ä¸‹åˆè§£å†³ï¼Œä»Šå¤©è§£å†³ä¸äº†å°±æ˜å¤©è§£å†³ã€‚è¿™ä¸ªæ–¹æ³•ä¸è¡Œå°±æ¢ä¸ªæ–¹æ³•ã€‚æ€è€ƒè‰¯ä¹…ä¹‹åå¯ä»¥é€‚å½“çš„é—®é—®è€å¸ˆã€‚æœ€å…³é”®çš„è¿˜æ˜¯è¦æŒ‰æ—¶åƒé¥­æŒ‰æ—¶ç¡è§‰ã€‚





